{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 16S_procedure_3b_bdiv_australia_analysis: Here we will test the hypothesis that compartments have distinct microbial communities via beta-diversity assessment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NOTE: These anlyses utilized rep_set.tre, which was VERY poor. All resulting analyses have been move to: output/old_tree_analyses\n",
    "\n",
    "# For analysis with updated tree (constrained to Green Genes phylogeny), see: 16S_procedure_3b_bdiv_australia_analysis_gg_constrained_tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Import libraries\n",
    "from os.path import join,abspath\n",
    "from os import listdir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Set file paths\n",
    "input_dir = abspath(\"input/\")\n",
    "otu_table = abspath(\"input/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom\")\n",
    "tree_fp = abspath(\"input/rep_set.tre\")\n",
    "mapping = abspath(\"input/gcmp16S_map.txt\")\n",
    "output_dir = abspath(\"output/\")\n",
    "\n",
    "#Preference file indicating that Curtis-Brays distance should also be calculated by beta_diversity\n",
    "bc_prefs_fp = abspath(\"input/bdiv_bc_prefs.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Subsampling to 1000 loses 31 samples, but this seems reasonable.\n",
    "depth=1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.200167655029 and the largest is 17.3868973942.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on all samples\n",
    "bdiv_dir_all = join(output_dir,'bdiv_all')\n",
    "!beta_diversity_through_plots.py -i $otu_table -o $bdiv_dir_all -m $mapping -p $bc_prefs_fp --force -t $tree_fp -e $depth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Export Sceleractinian coral compartment specific biom files (i.e. coral t, m and s)\n",
    "subset_output = join(output_dir,'tissue_compartment_tables')\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_mucus.biom -m $mapping -s 'sample_type:Coral_Mucus'\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_skeleton.biom -m $mapping -s 'sample_type:Coral_Skeleton'\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_tissue.biom -m $mapping -s 'sample_type:Coral_Tissue'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Summarize BIOM to confirm filter functioned correctly\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_mucus.biom -o $subset_output/otu_table_subset_mucus_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_tissue.biom -o $subset_output/otu_table_subset_tissue_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_skeleton.biom -o $subset_output/otu_table_subset_skeleton_summary.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Summaries look good."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.103036605043 and the largest is 9.90013989806.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.212867685774 and the largest is 11.5310694731.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.107446047992 and the largest is 5.80391442338.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.298712465576 and the largest is 13.7592665485.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0914100698643 and the largest is 7.84610596964.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.251819907438 and the largest is 7.68787029476.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on rarified (1000) subset of each sample type\n",
    "tissue_otu_table_dir = join(output_dir,\"tissue_compartment_tables\")\n",
    "depth = 1000\n",
    "\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith(\".biom\"):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    !beta_diversity_through_plots.py -i $curr_otu_table -o $curr_bdiv_dir -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'log_20160426110646.txt', 'otu_table_subset_mucus_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  14     9.757 0.69696  1.6898 0.1077 0.000999\n",
      "Residuals                       196    80.840 0.41245         0.8923         \n",
      "Total                           210    90.598                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    17.422 0.87110  2.2618 0.1923 0.000999\n",
      "Residuals                       190    73.176 0.38514         0.8077         \n",
      "Total                           210    90.598                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  54    31.673 0.58654  1.5528 0.3496 0.000999\n",
      "Residuals                       156    58.925 0.37772         0.6504         \n",
      "Total                           210    90.598                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  32    20.605 0.64390  1.6375 0.22743 0.000999\n",
      "Residuals                       178    69.993 0.39322         0.77257         \n",
      "Total                           210    90.598                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  14     6.104 0.43599  1.8817 0.11848 0.000999\n",
      "Residuals                       196    45.414 0.23170         0.88152         \n",
      "Total                           210    51.517                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    11.032 0.55160  2.5887 0.21414 0.000999\n",
      "Residuals                       190    40.485 0.21308         0.78586         \n",
      "Total                           210    51.517                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  54    18.531 0.34317  1.6229 0.35971 0.000999\n",
      "Residuals                       156    32.986 0.21145         0.64029         \n",
      "Total                           210    51.517                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  32    12.715 0.39735  1.8228 0.24681 0.000999\n",
      "Residuals                       178    38.802 0.21799         0.75319         \n",
      "Total                           210    51.517                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  14     6.081 0.43434  1.2751 0.08347 0.000999\n",
      "Residuals                       196    66.766 0.34065         0.91653         \n",
      "Total                           210    72.847                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    10.469 0.52345  1.5944 0.14371 0.000999\n",
      "Residuals                       190    62.378 0.32831         0.85629         \n",
      "Total                           210    72.847                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  54    21.520 0.39851  1.2112 0.29541 0.000999\n",
      "Residuals                       156    51.328 0.32902         0.70459         \n",
      "Total                           210    72.847                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  32    13.618 0.42558   1.279 0.18694 0.000999\n",
      "Residuals                       178    59.229 0.33275         0.81306         \n",
      "Total                           210    72.847                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "['bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'log_20160426110813.txt', 'otu_table_subset_skeleton_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  14    12.738 0.90984  2.0727 0.12839 0.000999\n",
      "Residuals                       197    86.475 0.43896         0.87161         \n",
      "Total                           211    99.212                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    13.184 0.65921  1.4636 0.13289 0.000999\n",
      "Residuals                       191    86.028 0.45041         0.86711         \n",
      "Total                           211    99.212                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  56    36.740 0.65607  1.6278 0.37032 0.000999\n",
      "Residuals                       155    62.472 0.40305         0.62968         \n",
      "Total                           211    99.212                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  33    25.577 0.77505  1.8735 0.2578 0.000999\n",
      "Residuals                       178    73.636 0.41368         0.7422         \n",
      "Total                           211    99.212                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  14    12.387 0.88478  3.2465 0.18747 0.000999\n",
      "Residuals                       197    53.688 0.27253         0.81253         \n",
      "Total                           211    66.075                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    10.080 0.50401  1.7192 0.15256 0.000999\n",
      "Residuals                       191    55.995 0.29317         0.84744         \n",
      "Total                           211    66.075                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  56    29.822 0.53253  2.2768 0.45133 0.000999\n",
      "Residuals                       155    36.253 0.23389         0.54867         \n",
      "Total                           211    66.075                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  33    22.758 0.68964  2.8339 0.34443 0.000999\n",
      "Residuals                       178    43.317 0.24335         0.65557         \n",
      "Total                           211    66.075                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  14    11.027 0.78766  2.2257 0.13657 0.000999\n",
      "Residuals                       197    69.716 0.35389         0.86343         \n",
      "Total                           211    80.744                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20     9.962 0.49811  1.3441 0.12338 0.000999\n",
      "Residuals                       191    70.781 0.37058         0.87662         \n",
      "Total                           211    80.744                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  56    28.312 0.50558  1.4946 0.35064 0.000999\n",
      "Residuals                       155    52.431 0.33827         0.64936         \n",
      "Total                           211    80.744                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  33    19.725 0.59774  1.7437 0.2443 0.000999\n",
      "Residuals                       178    61.018 0.34280         0.7557         \n",
      "Total                           211    80.744                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "['bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'log_20160426111011.txt', 'otu_table_subset_tissue_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  14    16.109 1.15065  2.7345 0.16918 0.000999\n",
      "Residuals                       188    79.108 0.42079         0.83082         \n",
      "Total                           202    95.217                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    14.855 0.74276  1.6822 0.15601 0.000999\n",
      "Residuals                       182    80.362 0.44155         0.84399         \n",
      "Total                           202    95.217                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  53    40.539 0.76489  2.0844 0.42576 0.000999\n",
      "Residuals                       149    54.678 0.36696         0.57424         \n",
      "Total                           202    95.217                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  32    29.534 0.92292  2.3887 0.31017 0.000999\n",
      "Residuals                       170    65.683 0.38637         0.68983         \n",
      "Total                           202    95.217                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  14    10.642 0.76011  3.3698 0.2006 0.000999\n",
      "Residuals                       188    42.407 0.22557         0.7994         \n",
      "Total                           202    53.048                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20     9.198 0.45992  1.9089 0.1734 0.000999\n",
      "Residuals                       182    43.850 0.24093         0.8266         \n",
      "Total                           202    53.048                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  53    24.426 0.46086  2.3991 0.46044 0.000999\n",
      "Residuals                       149    28.622 0.19210         0.53956         \n",
      "Total                           202    53.048                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  32    19.125 0.59765   2.995 0.36052 0.000999\n",
      "Residuals                       170    33.923 0.19955         0.63948         \n",
      "Total                           202    53.048                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  14    10.062 0.71868  2.0278 0.1312 0.000999\n",
      "Residuals                       188    66.630 0.35441         0.8688         \n",
      "Total                           202    76.691                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20     9.783 0.48914  1.3305 0.12756 0.000999\n",
      "Residuals                       182    66.909 0.36763         0.87244         \n",
      "Total                           202    76.691                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  53    26.789 0.50545  1.5092 0.34931 0.000999\n",
      "Residuals                       149    49.902 0.33492         0.65069         \n",
      "Total                           202    76.691                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  32    18.888 0.59024  1.7359 0.24628 0.000999\n",
      "Residuals                       170    57.804 0.34002         0.75372         \n",
      "Total                           202    76.691                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n"
     ]
    }
   ],
   "source": [
    "#Check for differences among mapping categories within a given compartment\n",
    "tissue_otu_table_dir = join(output_dir,\"tissue_compartment_tables\")\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'field_host_clade_sensu_fukami,reef_name.x,field_host_name,field_host_genus_id'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(output_dir,'bdiv_%s' %c)\n",
    "    print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)   \n",
    "        for curr_cat in categories.split(\",\"):\n",
    "            print \"curr_cat: %s\" %curr_cat\n",
    "            print \"curr_metric: %s\" %metric\n",
    "            print \"curr_category: %s\" %c\n",
    "            curr_compare_cat_output_dir = join(curr_bdiv_dir,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "            !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "            the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "            !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'distance_boxplots_all_within_bray_curtis', 'distance_boxplots_all_within_unweighted_unifrac', 'distance_boxplots_all_within_weighted_unifrac', 'log_20160426110646.txt', 'otu_table_subset_mucus_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "['bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'distance_boxplots_all_within_bray_curtis', 'distance_boxplots_all_within_unweighted_unifrac', 'distance_boxplots_all_within_weighted_unifrac', 'log_20160426110813.txt', 'otu_table_subset_skeleton_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "['bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'distance_boxplots_all_within_bray_curtis', 'distance_boxplots_all_within_unweighted_unifrac', 'distance_boxplots_all_within_weighted_unifrac', 'log_20160426111011.txt', 'otu_table_subset_tissue_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries: pre\n",
    "tissue_otu_table_dir = join(output_dir,\"tissue_compartment_tables\")\n",
    "n_permutations = 1000\n",
    "categories = 'field_host_clade_sensu_fukami,reef_name.x,field_host_name,field_host_genus_id,photosynthetically_active_radiation,visibility,turf_contact_percent,n_macroalgal_contacts,cyanobacteria_percent,cyanobacteria_contact,binary_turf_contact,binary_macroalgal_contact'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(output_dir,'bdiv_%s' %c)\n",
    "    print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n"
     ]
    }
   ],
   "source": [
    "#Test for differences in all samples (i.e. not partioned by compartment). This is useful for identifying differences among compartments.\n",
    "categories = 'BiologicalMatter,tissue_compartment,field_host_clade_sensu_fukami,reef_name.x,field_host_name,field_host_genus_id,photosynthetically_active_radiation,visibility,turf_contact_percent,n_macroalgal_contacts,cyanobacteria_percent,cyanobacteria_contact,binary_turf_contact,binary_macroalgal_contact'\n",
    "for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "    curr_boxplot_output_dir = join(bdiv_dir_all,\"distance_boxplots_%s\" %metric)\n",
    "    curr_dm = join(bdiv_dir_all,\"%s_dm.txt\" %metric)\n",
    "    !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.211066508956 and the largest is 17.3499799071.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -1.24847281768 and the largest is 23.5370487213.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on all samples in dataset rarified to 1000 reads\n",
    "tissue_otu_table_dir = join(output_dir,\"bdiv_all\")\n",
    "depth = 1000\n",
    "\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith(\".biom\"):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(output_dir,'bdiv_%s' %c)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "\n",
    "!beta_diversity_through_plots.py -i $curr_otu_table -o $curr_bdiv_dir -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Convert unrarified BIOM to CSV for manipulation in PRIMER\n",
    "!biom convert -i $input_dir/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom -o $input_dir/otu_table_mc2_wtax_no_pynast_failures_no_organelles_TXT.txt --to-tsv --header-key taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Convert rarified (1000) BIOM to CSV for manipulation in PRIMER\n",
    "!biom convert -i $output_dir/bdiv_otu_table_mc2_wtax_no_pynast_failures_no_organelles_even1000.biom/otu_table_mc2_wtax_no_pynast_failures_no_organelles_even1000.biom -o $input_dir/otu_table_mc2_wtax_no_pynast_failures_no_organelles_even1000_TXT.txt --to-tsv --header-key taxonomy"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Test for significant differences in community composition among coral compartments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Test for significant differences among coral compartments\n",
    "!compare_categories.py --method adonis -i output/bdiv_all/unweighted_unifrac_dm.txt -m input/gcmp16S_map_r20.txt -c BiologicalMatter -o output/significance_testing_compartments/compare_categories_compartments_unweighted_unifrac_adonis.txt -n 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/qiime/stats.py:1247: RuntimeWarning: divide by zero encountered in double_scalars\r\n",
      "  q = 1. + (a + 1.) / (6. * n)\r\n"
     ]
    }
   ],
   "source": [
    "#Explore which OTUs differ significantly among compartments\n",
    "!group_significance.py -i input/otu_table_mc2_wtax_no_pynast_failures_no_organelles_coral_tissue_mucus_skeleton_only.biom -m input/gcmp16S_map_r20.txt -c BiologicalMatter -s g_test -o output/significance_testing_compartments/group_significance_OTU_BiologicalMatter_g_test.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Does host genus drives microbiome structure in brooders to a greater extent than spawners? We will also look at reef and season."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Set file paths\n",
    "input_dir = abspath(\"input/\")\n",
    "otu_table = abspath(\"input/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom\")\n",
    "tree_fp = abspath(\"input/rep_set.tre\")\n",
    "mapping = abspath(\"input/gcmp16S_map.txt\")\n",
    "output_dir = abspath(\"output/\")\n",
    "\n",
    "#Set file paths\n",
    "input_dir = abspath(\"input/\")\n",
    "otu_table = abspath(\"input/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom\")\n",
    "tree_fp = abspath(\"input/rep_set.tre\")\n",
    "mapping = abspath(\"input/gcmp16S_map_r20.txt\")\n",
    "output_dir = abspath(\"output/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Split OTU table into brooder- and spawner-specific biom files\n",
    "subset_output = join(output_dir,'brooder_v_spawner')\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_brooder.biom -m $mapping -s 'Mode_of_larval_development:brooder'\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_spawner.biom -m $mapping -s 'Mode_of_larval_development:spawner'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Export Sceleractinian coral compartment specific biom files (i.e. coral t, m and s) for brooders\n",
    "subset_output = join(output_dir,'brooder_v_spawner')\n",
    "brooder_otu_table = join(subset_output,'otu_table_subset_brooder.biom')\n",
    "spawner_otu_table = join(subset_output,'otu_table_subset_spawner.biom')\n",
    "#Export Sceleractinian coral compartment specific biom files (i.e. coral t, m and s) for brooders\n",
    "!filter_samples_from_otu_table.py -i $brooder_otu_table -o $subset_output/otu_table_subset_brooder_mucus.biom -m $mapping -s 'BiologicalMatter:Coral Mucus'\n",
    "!filter_samples_from_otu_table.py -i $brooder_otu_table -o $subset_output/otu_table_subset_brooder_skeleton.biom -m $mapping -s 'BiologicalMatter:Coral Skeleton'\n",
    "!filter_samples_from_otu_table.py -i $brooder_otu_table -o $subset_output/otu_table_subset_brooder_tissue.biom -m $mapping -s 'BiologicalMatter:Coral Tissue'\n",
    "#Export Sceleractinian coral compartment specific biom files (i.e. coral t, m and s) for spawners\n",
    "!filter_samples_from_otu_table.py -i $spawner_otu_table -o $subset_output/otu_table_subset_spawner_mucus.biom -m $mapping -s 'BiologicalMatter:Coral Mucus'\n",
    "!filter_samples_from_otu_table.py -i $spawner_otu_table -o $subset_output/otu_table_subset_spawner_skeleton.biom -m $mapping -s 'BiologicalMatter:Coral Skeleton'\n",
    "!filter_samples_from_otu_table.py -i $spawner_otu_table -o $subset_output/otu_table_subset_spawner_tissue.biom -m $mapping -s 'BiologicalMatter:Coral Tissue'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#Summarize BIOM to confirm filter functioned correctly\n",
    "subset_output = join(output_dir,'brooder_v_spawner')\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_brooder_mucus.biom -o $subset_output/otu_table_subset_brooder_mucus_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_brooder_skeleton.biom -o $subset_output/otu_table_subset_brooder_skeleton_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_brooder_tissue.biom -o $subset_output/otu_table_subset_brooder_tissue_summary.txt\n",
    "\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_spawner_mucus.biom -o $subset_output/otu_table_subset_spawner_mucus_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_spawner_skeleton.biom -o $subset_output/otu_table_subset_spawner_skeleton_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_spawner_tissue.biom -o $subset_output/otu_table_subset_spawner_tissue_summary.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Looks good!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0412282144692 and the largest is 3.13193392707.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0579053952104 and the largest is 1.99439774237.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00477819038945 and the largest is 1.51279918088.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0125366593789 and the largest is 1.62546157568.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00536576056968 and the largest is 0.545835871795.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.129330573672 and the largest is 10.7675230558.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.510264543752 and the largest is 14.6681864957.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0352210408835 and the largest is 6.01557564504.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.119477033685 and the largest is 6.80736188985.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0853420629271 and the largest is 3.52993370452.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.179626778374 and the largest is 7.92810281431.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0258044802679 and the largest is 4.10679319982.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.132583047246 and the largest is 5.24913276903.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on rarified (1000) subset of each sample type\n",
    "tissue_otu_table_dir = join(output_dir,\"brooder_v_spawner\")\n",
    "depth = 1000\n",
    "\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith(\".biom\"):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    print(curr_otu_table)\n",
    "    !beta_diversity_through_plots.py -i $curr_otu_table -o $curr_bdiv_dir -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    2.2918 2.29178  5.6962 0.09873 0.000999\n",
      "Residuals                       52   20.9213 0.40233         0.90127         \n",
      "Total                           53   23.2131                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    4.0682 1.01704   2.603 0.17525 0.000999\n",
      "Residuals                       49   19.1449 0.39071         0.82475         \n",
      "Total                           53   23.2131                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    3.3865 1.69323  4.3555 0.14589 0.000999\n",
      "Residuals                       51   19.8266 0.38876         0.85411         \n",
      "Total                           53   23.2131                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    3.3865 1.69323  4.3555 0.14589 0.000999\n",
      "Residuals                       51   19.8266 0.38876         0.85411         \n",
      "Total                           53   23.2131                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)   \n",
      "qiime.data$map[[opts$category]]  1    0.5117 0.51174  2.3023 0.0424 0.006993 **\n",
      "Residuals                       52   11.5581 0.22227         0.9576            \n",
      "Total                           53   12.0699                 1.0000            \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    1.5385 0.38463  1.7896 0.12747 0.000999\n",
      "Residuals                       49   10.5313 0.21493         0.87253         \n",
      "Total                           53   12.0699                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.1709 0.58543  2.7394 0.09701 0.000999\n",
      "Residuals                       51   10.8990 0.21371         0.90299         \n",
      "Total                           53   12.0699                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.1709 0.58543  2.7394 0.09701 0.000999\n",
      "Residuals                       51   10.8990 0.21371         0.90299         \n",
      "Total                           53   12.0699                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1     0.704 0.70397  2.0673 0.03823 0.000999\n",
      "Residuals                       52    17.708 0.34053         0.96177         \n",
      "Total                           53    18.412                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    1.9585 0.48963  1.4582 0.10637 0.000999\n",
      "Residuals                       49   16.4532 0.33578         0.89363         \n",
      "Total                           53   18.4117                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.2463 0.62317  1.8515 0.06769 0.000999\n",
      "Residuals                       51   17.1654 0.33658         0.93231         \n",
      "Total                           53   18.4117                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.2463 0.62317  1.8515 0.06769 0.000999\n",
      "Residuals                       51   17.1654 0.33658         0.93231         \n",
      "Total                           53   18.4117                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    1.1380 1.13797  2.9871 0.14233 0.000999\n",
      "Residuals                       18    6.8574 0.38097         0.85767         \n",
      "Total                           19    7.9953                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    2.5637 0.64092  1.7699 0.32064 0.000999\n",
      "Residuals                       15    5.4317 0.36211         0.67936         \n",
      "Total                           19    7.9953                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)   \n",
      "qiime.data$map[[opts$category]]  2    1.6047 0.80235  2.1344 0.2007 0.001998 **\n",
      "Residuals                       17    6.3906 0.37592         0.7993            \n",
      "Total                           19    7.9953                 1.0000            \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.6047 0.80235  2.1344 0.2007 0.000999\n",
      "Residuals                       17    6.3906 0.37592         0.7993         \n",
      "Total                           19    7.9953                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  1    0.3819 0.38188  1.8853 0.09481 0.04396 *\n",
      "Residuals                       18    3.6461 0.20256         0.90519          \n",
      "Total                           19    4.0279                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    1.4320 0.35801  2.0687 0.35553 0.000999\n",
      "Residuals                       15    2.5959 0.17306         0.64447         \n",
      "Total                           19    4.0279                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  2    0.5875 0.29374  1.4514 0.14585 0.06993 .\n",
      "Residuals                       17    3.4405 0.20238         0.85415          \n",
      "Total                           19    4.0279                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  2    0.5875 0.29374  1.4514 0.14585 0.07493 .\n",
      "Residuals                       17    3.4405 0.20238         0.85415          \n",
      "Total                           19    4.0279                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  1    0.4701 0.47013   1.454 0.07474 0.02198 *\n",
      "Residuals                       18    5.8202 0.32334         0.92526          \n",
      "Total                           19    6.2903                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    1.6706 0.41764  1.3561 0.26558 0.004995\n",
      "Residuals                       15    4.6197 0.30798         0.73442         \n",
      "Total                           19    6.2903                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2 Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  2    0.7668 0.38341    1.18 0.1219 0.0989 .\n",
      "Residuals                       17    5.5235 0.32491         0.8781         \n",
      "Total                           19    6.2903                 1.0000         \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    0.7668 0.38341    1.18 0.1219 0.1009\n",
      "Residuals                       17    5.5235 0.32491         0.8781       \n",
      "Total                           19    6.2903                 1.0000       \n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    0.8547 0.85475  2.2035 0.12105 0.002997\n",
      "Residuals                       16    6.2064 0.38790         0.87895         \n",
      "Total                           17    7.0612                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    2.2888  0.5722  1.5587 0.32414 0.001998\n",
      "Residuals                       13    4.7723  0.3671         0.67586         \n",
      "Total                           17    7.0612                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.8080 0.90399  2.5813 0.25605 0.000999\n",
      "Residuals                       15    5.2532 0.35021         0.74395         \n",
      "Total                           17    7.0612                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.8080 0.90399  2.5813 0.25605 0.000999\n",
      "Residuals                       15    5.2532 0.35021         0.74395         \n",
      "Total                           17    7.0612                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    0.2310 0.23101  0.9102 0.05383 0.5005\n",
      "Residuals                       16    4.0609 0.25381         0.94617       \n",
      "Total                           17    4.2919                 1.00000       \n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    1.6263 0.40656  1.9828 0.37891 0.002997\n",
      "Residuals                       13    2.6656 0.20505         0.62109         \n",
      "Total                           17    4.2919                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  2    1.0542 0.52708  2.4419 0.24561 0.01199 *\n",
      "Residuals                       15    3.2377 0.21585         0.75439          \n",
      "Total                           17    4.2919                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.0542 0.52708  2.4419 0.24561 0.008991\n",
      "Residuals                       15    3.2377 0.21585         0.75439         \n",
      "Total                           17    4.2919                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  1    0.4287 0.42867  1.3006 0.07518 0.02198 *\n",
      "Residuals                       16    5.2736 0.32960         0.92482          \n",
      "Total                           17    5.7022                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  4    1.4845 0.37113  1.1439 0.26034 0.03497 *\n",
      "Residuals                       13    4.2177 0.32444         0.73966          \n",
      "Total                           17    5.7022                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    0.8395 0.41974  1.2948 0.14722 0.001998\n",
      "Residuals                       15    4.8628 0.32418         0.85278         \n",
      "Total                           17    5.7022                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    0.8395 0.41974  1.2948 0.14722 0.002997\n",
      "Residuals                       15    4.8628 0.32418         0.85278         \n",
      "Total                           17    5.7022                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    1.0545 1.05448  3.0624 0.17948 0.002997\n",
      "Residuals                       14    4.8206 0.34433         0.82052         \n",
      "Total                           15    5.8751                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    2.8506 0.71264  2.5918 0.48519 0.001998\n",
      "Residuals                       11    3.0246 0.27496         0.51481         \n",
      "Total                           15    5.8751                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.9162 0.95810  3.1461 0.32615 0.000999\n",
      "Residuals                       13    3.9589 0.30453         0.67385         \n",
      "Total                           15    5.8751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.9162 0.95810  3.1461 0.32615 0.000999\n",
      "Residuals                       13    3.9589 0.30453         0.67385         \n",
      "Total                           15    5.8751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  1   0.23771 0.23771  1.8194 0.11501 0.05794 .\n",
      "Residuals                       14   1.82913 0.13065         0.88499          \n",
      "Total                           15   2.06685                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4   0.70004 0.17501  1.4085 0.3387 0.1329\n",
      "Residuals                       11   1.36680 0.12426         0.6613       \n",
      "Total                           15   2.06685                 1.0000       \n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2   0.55841 0.27921  2.4063 0.27018 0.004995\n",
      "Residuals                       13   1.50843 0.11603         0.72982         \n",
      "Total                           15   2.06685                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2   0.55841 0.27921  2.4063 0.27018 0.003996\n",
      "Residuals                       13   1.50843 0.11603         0.72982         \n",
      "Total                           15   2.06685                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    0.5304 0.53039  1.5603 0.10027 0.000999\n",
      "Residuals                       14    4.7590 0.33993         0.89973         \n",
      "Total                           15    5.2894                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  4    1.5642 0.39105  1.1547 0.29572 0.04595 *\n",
      "Residuals                       11    3.7252 0.33866         0.70428          \n",
      "Total                           15    5.2894                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    0.9258 0.46290  1.3791 0.17503 0.000999\n",
      "Residuals                       13    4.3636 0.33566         0.82497         \n",
      "Total                           15    5.2894                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    0.9258 0.46290  1.3791 0.17503 0.002997\n",
      "Residuals                       13    4.3636 0.33566         0.82497         \n",
      "Total                           15    5.2894                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     12.58 1.57249  3.5141 0.07115 0.000999\n",
      "Residuals                       367    164.23 0.44748         0.92885         \n",
      "Total                           375    176.81                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    18.255 0.96081  2.1574 0.10325 0.000999\n",
      "Residuals                       356   158.551 0.44537         0.89675         \n",
      "Total                           375   176.806                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26    27.644  1.0632  2.4877 0.15635 0.000999\n",
      "Residuals                       349   149.162  0.4274         0.84365         \n",
      "Total                           375   176.806                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18    22.482 1.24899  2.8893 0.12715 0.000999\n",
      "Residuals                       357   154.324 0.43228         0.87285         \n",
      "Total                           375   176.806                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8    10.977 1.37216  4.9533 0.09745 0.000999\n",
      "Residuals                       367   101.665 0.27702         0.90255         \n",
      "Total                           375   112.643                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    11.644 0.61283  2.1601 0.10337 0.000999\n",
      "Residuals                       356   100.999 0.28370         0.89663         \n",
      "Total                           375   112.643                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26    21.257 0.81758  3.1223 0.18871 0.000999\n",
      "Residuals                       349    91.385 0.26185         0.81129         \n",
      "Total                           375   112.643                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18    18.266 1.01475  3.8385 0.16215 0.000999\n",
      "Residuals                       357    94.377 0.26436         0.83785         \n",
      "Total                           375   112.643                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     9.104 1.13796  3.1229 0.06374 0.000999\n",
      "Residuals                       367   133.730 0.36439         0.93626         \n",
      "Total                           375   142.833                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    10.729 0.56469  1.5218 0.07512 0.000999\n",
      "Residuals                       356   132.104 0.37108         0.92488         \n",
      "Total                           375   142.833                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26    17.758 0.68300  1.9058 0.12433 0.000999\n",
      "Residuals                       349   125.076 0.35838         0.87567         \n",
      "Total                           375   142.833                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18     14.53 0.80720   2.246 0.10172 0.000999\n",
      "Residuals                       357    128.30 0.35939         0.89828         \n",
      "Total                           375    142.83                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     5.885 0.73561  1.8393 0.11257 0.000999\n",
      "Residuals                       116    46.394 0.39995         0.88743         \n",
      "Total                           124    52.279                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    13.840 0.72841  1.9897 0.26473 0.000999\n",
      "Residuals                       105    38.439 0.36608         0.73527         \n",
      "Total                           124    52.279                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24    13.965 0.58189  1.5188 0.26713 0.000999\n",
      "Residuals                       100    38.313 0.38313         0.73287         \n",
      "Total                           124    52.279                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17    10.813 0.63603  1.6412 0.20682 0.000999\n",
      "Residuals                       107    41.466 0.38753         0.79318         \n",
      "Total                           124    52.279                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8    3.9254 0.49067  2.1188 0.12749 0.000999\n",
      "Residuals                       116   26.8636 0.23158         0.87251         \n",
      "Total                           124   30.7890                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    8.7082 0.45832  2.1795 0.28283 0.000999\n",
      "Residuals                       105   22.0808 0.21029         0.71717         \n",
      "Total                           124   30.7890                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24    8.8741 0.36975  1.6872 0.28822 0.000999\n",
      "Residuals                       100   21.9149 0.21915         0.71178         \n",
      "Total                           124   30.7890                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17    7.0808 0.41652  1.8798 0.22998 0.000999\n",
      "Residuals                       107   23.7082 0.22157         0.77002         \n",
      "Total                           124   30.7890                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     3.398 0.42471  1.2519 0.07948 0.000999\n",
      "Residuals                       116    39.353 0.33925         0.92052         \n",
      "Total                           124    42.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     8.409 0.44256  1.3531 0.19669 0.000999\n",
      "Residuals                       105    34.342 0.32707         0.80331         \n",
      "Total                           124    42.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24     9.401 0.39172  1.1746 0.21991 0.000999\n",
      "Residuals                       100    33.350 0.33350         0.78009         \n",
      "Total                           124    42.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17     7.111 0.41830  1.2559 0.16634 0.000999\n",
      "Residuals                       107    35.640 0.33308         0.83366         \n",
      "Total                           124    42.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8      7.73 0.96630  2.2032 0.12807 0.000999\n",
      "Residuals                       120     52.63 0.43858         0.87193         \n",
      "Total                           128     60.36                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    11.834 0.62283   1.399 0.19605 0.000999\n",
      "Residuals                       109    48.526 0.44520         0.80395         \n",
      "Total                           128    60.360                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26     18.46 0.71001  1.7284 0.30584 0.000999\n",
      "Residuals                       102     41.90 0.41078         0.69416         \n",
      "Total                           128     60.36                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18    14.609 0.81163  1.9514 0.24204 0.000999\n",
      "Residuals                       110    45.751 0.41592         0.75796         \n",
      "Total                           128    60.360                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     8.532 1.06646  4.0227 0.21147 0.000999\n",
      "Residuals                       120    31.814 0.26511         0.78853         \n",
      "Total                           128    40.345                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     8.357 0.43982  1.4987 0.20713 0.000999\n",
      "Residuals                       109    31.989 0.29347         0.79287         \n",
      "Total                           128    40.345                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26    16.520 0.63538  2.7201 0.40946 0.000999\n",
      "Residuals                       102    23.825 0.23358         0.59054         \n",
      "Total                           128    40.345                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18    14.046 0.78034  3.2639 0.34815 0.000999\n",
      "Residuals                       110    26.299 0.23908         0.65185         \n",
      "Total                           128    40.345                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model    R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     7.355 0.91933  2.6056 0.148 0.000999\n",
      "Residuals                       120    42.338 0.35282         0.852         \n",
      "Total                           128    49.693                 1.000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     8.857 0.46613  1.2442 0.17822 0.000999\n",
      "Residuals                       109    40.837 0.37465         0.82178         \n",
      "Total                           128    49.693                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26    14.646  0.5633  1.6394 0.29472 0.000999\n",
      "Residuals                       102    35.047  0.3436         0.70528         \n",
      "Total                           128    49.693                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18    11.688 0.64934  1.8794 0.2352 0.000999\n",
      "Residuals                       110    38.005 0.34550         0.7648         \n",
      "Total                           128    49.693                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     9.638 1.20478  2.8782 0.16927 0.000999\n",
      "Residuals                       113    47.300 0.41859         0.83073         \n",
      "Total                           121    56.938                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    12.690 0.66790  1.5396 0.22288 0.000999\n",
      "Residuals                       102    44.248 0.43381         0.77712         \n",
      "Total                           121    56.938                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24    20.264 0.84433  2.2332 0.35589 0.000999\n",
      "Residuals                        97    36.675 0.37809         0.64411         \n",
      "Total                           121    56.938                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17    16.616 0.97743   2.521 0.29183 0.000999\n",
      "Residuals                       104    40.322 0.38771         0.70817         \n",
      "Total                           121    56.938                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     7.292 0.91154  3.8984 0.2163 0.000999\n",
      "Residuals                       113    26.422 0.23382         0.7837         \n",
      "Total                           121    33.715                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     7.827 0.41194  1.6231 0.23215 0.000999\n",
      "Residuals                       102    25.888 0.25380         0.76785         \n",
      "Total                           121    33.715                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24    14.273 0.59472  2.9672 0.42335 0.000999\n",
      "Residuals                        97    19.441 0.20043         0.57665         \n",
      "Total                           121    33.715                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17    12.391 0.72888  3.5549 0.36753 0.000999\n",
      "Residuals                       104    21.324 0.20503         0.63247         \n",
      "Total                           121    33.715                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     6.526 0.81572   2.363 0.14332 0.000999\n",
      "Residuals                       113    39.008 0.34520         0.85668         \n",
      "Total                           121    45.533                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     8.522 0.44850   1.236 0.18715 0.000999\n",
      "Residuals                       102    37.012 0.36286         0.81285         \n",
      "Total                           121    45.533                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24    13.647 0.56861  1.7297 0.29971 0.000999\n",
      "Residuals                        97    31.887 0.32873         0.70029         \n",
      "Total                           121    45.533                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17    10.948 0.64400  1.9365 0.24044 0.000999\n",
      "Residuals                       104    34.585 0.33255         0.75956         \n",
      "Total                           121    45.533                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n"
     ]
    }
   ],
   "source": [
    "#Check for differences among mapping categories within a given compartment\n",
    "tissue_otu_table_dir = join(output_dir,\"brooder_v_spawner\")\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'field_host_clade_sensu_fukami,reef_name.x,field_host_name,field_host_genus_id'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    " #   print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)   \n",
    "        for curr_cat in categories.split(\",\"):\n",
    "            print \"curr_cat: %s\" %curr_cat\n",
    "            print \"curr_metric: %s\" %metric\n",
    "            print \"curr_category: %s\" %c\n",
    "            curr_compare_cat_output_dir = join(curr_bdiv_dir,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "            !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "            the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "            !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# We see significant correlations between alpha diversity and 1) latitude and 2) proportion of max colony size. These effects are most pronounced in stress-tolerant corals. Here, we will see if overall community structure changes 1) along latitudinal gradient and 2) as corals grow larger. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Set file paths\n",
    "input_dir = abspath(\"input/\")\n",
    "otu_table = abspath(\"input/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom\")\n",
    "tree_fp = abspath(\"input/rep_set.tre\")\n",
    "mapping = abspath(\"input/gcmp16S_map_r20.txt\")\n",
    "output_dir = abspath(\"output/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Split OTU table into different functional groups\n",
    "subset_output = join(output_dir,'functional_groups')\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_generalist.biom -m $mapping -s 'functional_group_sensu_darling:Generalist'\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_competitive.biom -m $mapping -s 'functional_group_sensu_darling:Competitive'\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_stress_tolerant.biom -m $mapping -s 'functional_group_sensu_darling:Stress-tolerant'\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_weedy.biom -m $mapping -s 'functional_group_sensu_darling:Weedy'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#Summarize BIOM to confirm filter functioned correctly\n",
    "subset_output = join(output_dir,'functional_groups')\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_generalist.biom -o $subset_output/otu_table_subset_generalist.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_competitive.biom -o $subset_output/otu_table_subset_competitive.txt\n",
    "!biom summarize-table -i $subset_output/otu_t able_subset_stress_tolerant.biom -o $subset_output/otu_table_subset_stress_tolerant.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_weedy.biom -o $subset_output/otu_table_subset_weedy.txt\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Looks good!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00741759659045 and the largest is 3.42543681733.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0534624286624 and the largest is 3.60220386107.\n",
      "  RuntimeWarning\n",
      "/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00426119158986 and the largest is 2.52518339985.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0279330580674 and the largest is 3.08720378615.\n",
      "  RuntimeWarning\n",
      "/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0943930219986 and the largest is 7.68100544672.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.35954414476 and the largest is 14.1307473814.\n",
      "  RuntimeWarning\n",
      "/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0417159752951 and the largest is 5.5945087651.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0748657297364 and the largest is 2.68518614475.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on rarified (1000) subset of each sample type\n",
    "tissue_otu_table_dir = join(output_dir,'functional_groups')\n",
    "depth = 1000\n",
    "\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    print(curr_otu_table)\n",
    "    !beta_diversity_through_plots.py -i $curr_otu_table -o $curr_bdiv_dir -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]] 23    11.243 0.48881   1.113 0.32989 0.01499 *\n",
      "Residuals                       52    22.837 0.43917         0.67011          \n",
      "Total                           75    34.079                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12     8.065 0.67208  1.6276 0.23665 0.000999\n",
      "Residuals                       63    26.015 0.41293         0.76335         \n",
      "Total                           75    34.079                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12     8.065 0.67208  1.6276 0.23665 0.000999\n",
      "Residuals                       63    26.015 0.41293         0.76335         \n",
      "Total                           75    34.079                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]] 23    5.3402 0.23218   1.019 0.31068 0.4106\n",
      "Residuals                       52   11.8484 0.22785         0.68932       \n",
      "Total                           75   17.1886                 1.00000       \n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12    3.7498 0.31248  1.4649 0.21815 0.002997\n",
      "Residuals                       63   13.4389 0.21332         0.78185         \n",
      "Total                           75   17.1886                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12    3.7498 0.31248  1.4649 0.21815 0.000999\n",
      "Residuals                       63   13.4389 0.21332         0.78185         \n",
      "Total                           75   17.1886                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]] 23    8.3365 0.36246  0.9942 0.30543 0.5155\n",
      "Residuals                       52   18.9578 0.36457         0.69457       \n",
      "Total                           75   27.2943                 1.00000       \n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)   \n",
      "qiime.data$map[[opts$category]] 12    4.9295 0.41079  1.1572 0.1806 0.001998 **\n",
      "Residuals                       63   22.3649 0.35500         0.8194            \n",
      "Total                           75   27.2943                 1.0000            \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)   \n",
      "qiime.data$map[[opts$category]] 12    4.9295 0.41079  1.1572 0.1806 0.001998 **\n",
      "Residuals                       63   22.3649 0.35500         0.8194            \n",
      "Total                           75   27.2943                 1.0000            \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12    6.9065 0.57554  1.3728 0.32638 0.000999\n",
      "Residuals                       34   14.2541 0.41924         0.67362         \n",
      "Total                           46   21.1606                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    3.4111 0.68223  1.5759 0.1612 0.000999\n",
      "Residuals                       41   17.7494 0.43291         0.8388         \n",
      "Total                           46   21.1606                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    3.4111 0.68223  1.5759 0.1612 0.000999\n",
      "Residuals                       41   17.7494 0.43291         0.8388         \n",
      "Total                           46   21.1606                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12    5.2877 0.44064  1.7682 0.38427 0.000999\n",
      "Residuals                       34    8.4728 0.24920         0.61573         \n",
      "Total                           46   13.7605                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    2.2542 0.45085  1.6065 0.16382 0.005994\n",
      "Residuals                       41   11.5062 0.28064         0.83618         \n",
      "Total                           46   13.7605                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    2.2542 0.45085  1.6065 0.16382 0.003996\n",
      "Residuals                       41   11.5062 0.28064         0.83618         \n",
      "Total                           46   13.7605                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12    4.7712 0.39760  1.1477 0.28829 0.003996\n",
      "Residuals                       34   11.7790 0.34644         0.71171         \n",
      "Total                           46   16.5501                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    2.2925 0.45849  1.3185 0.13852 0.000999\n",
      "Residuals                       41   14.2577 0.34775         0.86148         \n",
      "Total                           46   16.5501                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    2.2925 0.45849  1.3185 0.13852 0.000999\n",
      "Residuals                       41   14.2577 0.34775         0.86148         \n",
      "Total                           46   16.5501                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  62    34.758 0.56062   1.271 0.28572 0.000999\n",
      "Residuals                       197    86.893 0.44108         0.71428         \n",
      "Total                           259   121.651                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     16.63 0.87526  2.0002 0.1367 0.000999\n",
      "Residuals                       240    105.02 0.43759         0.8633         \n",
      "Total                           259    121.65                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     16.63 0.87526  2.0002 0.1367 0.000999\n",
      "Residuals                       240    105.02 0.43759         0.8633         \n",
      "Total                           259    121.65                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  62    21.846 0.35235  1.1784 0.27053 0.000999\n",
      "Residuals                       197    58.906 0.29901         0.72947         \n",
      "Total                           259    80.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    11.376 0.59874  2.0713 0.14088 0.000999\n",
      "Residuals                       240    69.375 0.28906         0.85912         \n",
      "Total                           259    80.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    11.376 0.59874  2.0713 0.14088 0.000999\n",
      "Residuals                       240    69.375 0.28906         0.85912         \n",
      "Total                           259    80.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  62    25.438 0.41028   1.118 0.26028 0.001998\n",
      "Residuals                       197    72.293 0.36697         0.73972         \n",
      "Total                           259    97.730                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    10.688 0.56251   1.551 0.10936 0.000999\n",
      "Residuals                       240    87.043 0.36268         0.89064         \n",
      "Total                           259    97.730                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    10.688 0.56251   1.551 0.10936 0.000999\n",
      "Residuals                       240    87.043 0.36268         0.89064         \n",
      "Total                           259    97.730                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 18    11.446 0.63588   1.814 0.40485 0.000999\n",
      "Residuals                       48    16.826 0.35054         0.59515         \n",
      "Total                           66    28.272                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7    6.6039 0.94342  2.5689 0.23359 0.000999\n",
      "Residuals                       59   21.6678 0.36725         0.76641         \n",
      "Total                           66   28.2717                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7    6.6039 0.94342  2.5689 0.23359 0.000999\n",
      "Residuals                       59   21.6678 0.36725         0.76641         \n",
      "Total                           66   28.2717                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]] 18    4.8226 0.26792  1.3098 0.32938 0.01399 *\n",
      "Residuals                       48    9.8188 0.20456         0.67062          \n",
      "Total                           66   14.6414                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7    2.7217 0.38882  1.9246 0.18589 0.001998\n",
      "Residuals                       59   11.9197 0.20203         0.81411         \n",
      "Total                           66   14.6414                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7    2.7217 0.38882  1.9246 0.18589 0.000999\n",
      "Residuals                       59   11.9197 0.20203         0.81411         \n",
      "Total                           66   14.6414                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 18    7.4096 0.41165  1.2081 0.31178 0.000999\n",
      "Residuals                       48   16.3558 0.34075         0.68822         \n",
      "Total                           66   23.7654                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7     3.713 0.53042  1.5607 0.15623 0.000999\n",
      "Residuals                       59    20.052 0.33987         0.84377         \n",
      "Total                           66    23.765                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7     3.713 0.53042  1.5607 0.15623 0.000999\n",
      "Residuals                       59    20.052 0.33987         0.84377         \n",
      "Total                           66    23.765                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_prop_Colony_maximum_diameter_universal_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_latitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_longitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_prop_Colony_maximum_diameter_universal_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_latitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_longitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_prop_Colony_maximum_diameter_universal_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_latitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_longitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_prop_Colony_maximum_diameter_universal_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_latitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_longitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_prop_Colony_maximum_diameter_universal_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_latitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_longitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_prop_Colony_maximum_diameter_universal_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_latitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_longitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_prop_Colony_maximum_diameter_universal_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_latitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_longitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_prop_Colony_maximum_diameter_universal_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_latitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_longitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_prop_Colony_maximum_diameter_universal_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_latitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_longitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_prop_Colony_maximum_diameter_universal_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_latitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_longitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_prop_Colony_maximum_diameter_universal_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_latitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_longitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_prop_Colony_maximum_diameter_universal_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_latitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_longitude_unweighted_unifrac/adonis_results.txt: Not a directory\n"
     ]
    }
   ],
   "source": [
    "#Check for differences among mapping categories within a given functional group\n",
    "tissue_otu_table_dir = join(output_dir,\"functional_groups\")\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'prop_Colony_maximum_diameter_universal,latitude,longitude'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)   \n",
    "        for curr_cat in categories.split(\",\"):\n",
    "            print \"curr_cat: %s\" %curr_cat\n",
    "            print \"curr_metric: %s\" %metric\n",
    "            print \"curr_category: %s\" %c\n",
    "            curr_compare_cat_output_dir = join(curr_bdiv_dir,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "            !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "            the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "            !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries:\n",
    "tissue_otu_table_dir = join(output_dir,\"functional_groups\")\n",
    "n_permutations = 1000\n",
    "categories = 'prop_Colony_maximum_diameter_universal,latitude,longitude'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Use adonis to explore how much variability in coral microbiome is explained by compartment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Export Sceleractinian coral  specific biom file\n",
    "subset_output = join(output_dir,'coral_only')\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_coral_only.biom -m $mapping --sample_id_fp input/coral_only.txt --output_mapping_fp $subset_output/otu_table_coral_only_map.txt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Summarize BIOM to confirm filter functioned correctly\n",
    "subset_output = join(output_dir,'coral_only')\n",
    "!biom summarize-table -i $subset_output/otu_table_coral_only.biom -o $subset_output/otu_table_coral_only_summary.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Summaries look good."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.192225882455 and the largest is 17.2822267266.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -1.19476909769 and the largest is 22.2380806245.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on rarified (1000) subset of each sample type\n",
    "tissue_otu_table_dir = join(output_dir,\"coral_only\")\n",
    "depth = 1000\n",
    "\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith(\".biom\"):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    !beta_diversity_through_plots.py -i $curr_otu_table -o $curr_bdiv_dir -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "curr_cat: BiologicalMatter\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_coral_only.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2    10.469  5.2344  11.443 0.03538 0.000999\n",
      "Residuals                       624   285.434  0.4574         0.96462         \n",
      "Total                           626   295.903                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_coral_only.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2    11.185  5.5925  20.442 0.06149 0.000999\n",
      "Residuals                       624   170.716  0.2736         0.93851         \n",
      "Total                           626   181.901                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_coral_only.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     7.248  3.6241  9.8508 0.03061 0.000999\n",
      "Residuals                       624   229.568  0.3679         0.96939         \n",
      "Total                           626   236.816                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_coral_only.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/compare_categories_BiologicalMatter_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_coral_only.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/compare_categories_BiologicalMatter_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_coral_only.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/compare_categories_BiologicalMatter_unweighted_unifrac/adonis_results.txt: Not a directory\n"
     ]
    }
   ],
   "source": [
    "#Check for differences among mapping categories within a given functional group\n",
    "tissue_otu_table_dir = join(output_dir,\"coral_only\")\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'BiologicalMatter'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)   \n",
    "        for curr_cat in categories.split(\",\"):\n",
    "            print \"curr_cat: %s\" %curr_cat\n",
    "            print \"curr_metric: %s\" %metric\n",
    "            print \"curr_category: %s\" %c\n",
    "            curr_compare_cat_output_dir = join(curr_bdiv_dir,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "            !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "            the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "            !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Boxplot summaries:\n",
    "tissue_otu_table_dir = join(output_dir,\"coral_only\")\n",
    "n_permutations = 1000\n",
    "categories = 'BiologicalMatter'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Explore differences in traunches of latitude (latitude_traunch) and prop_Colony_maximum_diameter_universal (prop_Colony_maximum_diameter_universal_traunch)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Trauches added in \"input/gcmp16S_map_r20_lat_size_traunches.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in all corals:\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"coral_only\")\n",
    "n_permutations = 1000\n",
    "categories = 'latitude_traunch,prop_Colony_maximum_diameter_universal_traunch'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in tissue compartment:\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"tissue_compartment_tables\")\n",
    "n_permutations = 1000\n",
    "categories = 'latitude_traunch,prop_Colony_maximum_diameter_universal_traunch'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in functional groups:\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"functional_groups\")\n",
    "n_permutations = 1000\n",
    "categories = 'latitude_traunch,prop_Colony_maximum_diameter_universal_traunch'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in all corals for compartment differences:\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"coral_only\")\n",
    "n_permutations = 1000\n",
    "categories = 'BiologicalMatter'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in brooders and spawners:\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"functional_groups\")\n",
    "n_permutations = 1000\n",
    "categories = 'latitude_traunch,prop_Colony_maximum_diameter_universal_traunch'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Explore differences in traunches of prop_Colony_maximum_diameter_universal (< 20% vs > 20% max size) w/in compartments..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Set file paths\n",
    "input_dir = abspath(\"input/\")\n",
    "otu_table = abspath(\"input/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom\")\n",
    "tree_fp = abspath(\"input/rep_set.tre\")\n",
    "mapping = abspath(\"input/gcmp16S_map_r20.txt\")\n",
    "output_dir = abspath(\"output/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue_1000.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in tissue compartment (with raw data save):\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"tissue_compartment_tables\")\n",
    "n_permutations = 1000\n",
    "categories = 'latitude_traunch,prop_Colony_maximum_diameter_universal_traunch'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "#        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_all_within --suppress_all_between -n $n_permutations --sort median --save_raw_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# With continuous data (i.e. prop_Colony_maximum_diameter_universal,latitude,longitude) it might be better to bin before box plots"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Could explore any interesting differences deeper by looking within compartments of specific functional groups."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Num samples: 692\r\n",
      "Num observations: 129305\r\n",
      "Total count: 9441738\r\n",
      "Table density (fraction of non-zero values): 0.006\r\n",
      "\r\n",
      "Counts/sample summary:\r\n",
      " Min: 2.0\r\n",
      " Max: 38523.0\r\n",
      " Median: 14010.000\r\n",
      " Mean: 13644.130\r\n",
      " Std. dev.: 7564.457\r\n",
      " Sample Metadata Categories: None provided\r\n",
      " Observation Metadata Categories: taxonomy\r\n",
      "\r\n",
      "Counts/sample detail:\r\n",
      " E7.17.Pla.sine.1.20150622.M: 2.0\r\n",
      " E3.17.Ech.mamm.1.20150127.S: 3.0\r\n",
      " E10.17.Fav.pali.2.20150630.T: 5.0\r\n",
      " E7.17.Fav.spB.1.20150623.T: 7.0\r\n",
      " E3.17.Ech.horr.1.20150123.M: 8.0\r\n",
      " E3.15.Dip.heli.2.20150120.T: 9.0\r\n",
      " E10.Cyp.sera.1.20150628.M: 17.0\r\n",
      " E10.17.Fav.pali.1.20150630.T: 17.0\r\n",
      " E3.Out.Mil.sp.2.20150125.M: 24.0\r\n",
      " E10.17.Fav.pali.1.20150630.M: 36.0\r\n",
      " E10.17.Fav.pali.2.20150630.S: 41.0\r\n",
      " E10.Cyp.sera.2.20150628.T: 42.0\r\n",
      " E1.6.Acr.sp.1.20140726.T: 42.0\r\n",
      " E3.5.Gal.astr.2.20150127.T: 53.0\r\n",
      " E10.Ser.cali.2.20150627.T: 66.0\r\n",
      " E10.Alv.alli.1.20150628.S: 96.0\r\n",
      " E10.Ser.cali.1.20150627.T: 102.0\r\n",
      " E10.17.Fav.pali.2.20150630.M: 127.0\r\n",
      " E10.17.Fav.abdi.2.20150630.T: 165.0\r\n",
      " E7.17.Gon.edwa.2.20150623.S: 197.0\r\n",
      " E10.17.Fav.pali.3.20150630.T: 200.0\r\n",
      " E10.17.Fav.pali.3.20150630.S: 252.0\r\n",
      " E10.17.Fav.pali.3.20150630.M: 276.0\r\n",
      " E3.6.Acr.hyac.3.20150106.S: 470.0\r\n",
      " E10.Ser.cali.3.20150627.T: 522.0\r\n",
      " E10.10.Sty.pist.2.20150630.S: 566.0\r\n",
      " E3.6.Acr.hyac.1.20150120.T: 576.0\r\n",
      " E3.2.Tur.reni.2.20150124.S: 627.0\r\n",
      " E7.7.Pav.vari.1.20150622.T: 690.0\r\n",
      " E3.2.Tur.reni.2.20150124.T: 788.0\r\n",
      " E3.6.Acr.hyac.4.20150106.M: 941.0\r\n",
      " E10.Ser.cali.3.20150627.S: 1004.0\r\n",
      " E3.10.Sty.pist.2.20150124.S: 1005.0\r\n",
      " E1.6.Acr.sp.1.20140726.M: 1009.0\r\n",
      " E7.10.Poc.dami.2.20150621.T: 1033.0\r\n",
      " E7.10.Poc.dami.1.20150621.T: 1114.0\r\n",
      " E3.2.Tur.reni.1.20150124.S: 1138.0\r\n",
      " E1.10.Poc.dami.1.20140728.S: 1176.0\r\n",
      " E10.10.Sty.pist.3.20150630.T: 1256.0\r\n",
      " E10.10.Sty.pist.3.20150630.S: 1270.0\r\n",
      " E3.10.Sty.pist.1.20150124.S: 1358.0\r\n",
      " E10.Ser.cali.1.20150627.S: 1390.0\r\n",
      " E3.11.Fun.sp.2.20150125.S: 1415.0\r\n",
      " Negative.Control.2: 1442.0\r\n",
      " E7.17.Mer.ampl.3.20150621.S: 1468.0\r\n",
      " E3.10.Sty.pist.1.20150127.S: 1490.0\r\n",
      " E10.Ser.cali.2.20150627.S: 1547.0\r\n",
      " E3.15.Dip.heli.2.20150123.T: 1558.0\r\n",
      " E7.17.Cyp.micr.1.20150621.T: 1563.0\r\n",
      " E7.17.Mer.ampl.1.20150621.S: 1595.0\r\n",
      " E7.17.Mer.ampl.2.20150621.S: 1640.0\r\n",
      " E3.2.Tur.reni.1.20150124.T: 1673.0\r\n",
      " E3.17.Ech.mamm.1.20150129.T: 1678.0\r\n",
      " E3.10.Sty.pist.1.20150124.T: 1700.0\r\n",
      " E3.6.Acr.hyac.2.20150121.T: 1763.0\r\n",
      " Negative.Control.1: 1771.0\r\n",
      " E1.17.Ech.mamm.1.20140815.S: 1993.0\r\n",
      " E1.14.Phy.lich.1.20140801.M: 2027.0\r\n",
      " E3.10.Sty.pist.1.20150127.T: 2059.0\r\n",
      " E10.10.Sty.pist.1.20150630.T: 2109.0\r\n",
      " Negative.Control.3: 2120.0\r\n",
      " E1.2.Tur.reni.1.20140814.S: 2194.0\r\n",
      " E3.10.Sty.pist.2.20150124.T: 2232.0\r\n",
      " E7.10.Ser.cali.3.20150622.T: 2268.0\r\n",
      " E7.7.Pav.vari.1.20150622.S: 2309.0\r\n",
      " E3.15.Dip.heli.1.20150119.T: 2325.0\r\n",
      " E3.4.Pac.spec.1.20150128.S: 2332.0\r\n",
      " E3.14.Phy.lich.2.20150125.T: 2353.0\r\n",
      " E10.Por.hero.1.20150628.S: 2355.0\r\n",
      " Negative.Control.4: 2493.0\r\n",
      " E7.10.Poc.dami.3.20150621.T: 2544.0\r\n",
      " E1.14.Phy.lich.1.20140801.T: 2568.0\r\n",
      " E10.Cyp.micr.1.20150627.S: 2638.0\r\n",
      " E3.14.Phy.lich.1.20150125.T: 2679.0\r\n",
      " E3.15.Dip.heli.3.20150106.M: 2719.0\r\n",
      " E7.10.Ser.cali.2.20150622.S: 3296.0\r\n",
      " E1.17.Ech.mamm.1.20140728.T: 3523.0\r\n",
      " E10.Aca.hill.2.20150629.T: 3617.0\r\n",
      " E7.17.Fav.abdi.2.20150623.S: 3626.0\r\n",
      " E10.10.Sty.pist.1.20150630.S: 3678.0\r\n",
      " E7.10.Ser.cali.1.20150622.T: 3837.0\r\n",
      " E1.6.Mon.hisp.1.20140807.T: 3857.0\r\n",
      " E3.19.Sym.sp.2.20150125.T: 3902.0\r\n",
      " E1.4.Pac.spec.1.20140801.M: 3954.0\r\n",
      " E3.11.Fun.sp.1.20150127.S: 3963.0\r\n",
      " E10.10.Sty.pist.2.20150630.T: 3977.0\r\n",
      " E10.Aca.hill.2.20150629.S: 3979.0\r\n",
      " E7.3.Por.loba.2.20150619.S: 4098.0\r\n",
      " E3.15.Dip.heli.1.20150121.T: 4127.0\r\n",
      " E10.Poc.dami.1.20150628.S: 4135.0\r\n",
      " E1.10.Poc.dami.1.20140905.S: 4139.0\r\n",
      " E7.10.Poc.dami.1.20150621.S: 4148.0\r\n",
      " E7.10.Ser.cali.2.20150622.T: 4274.0\r\n",
      " E3.19.Sym.sp.1.20150127.T: 4368.0\r\n",
      " E3.6.Acr.hyac.1.20150105.T: 4432.0\r\n",
      " E10.Aca.hill.1.20150629.T: 4436.0\r\n",
      " E1.10.Poc.dami.1.20140905.T: 4444.0\r\n",
      " E10.Aca.hill.3.20150629.T: 4458.0\r\n",
      " E3.15.Dip.heli.1.20150120.T: 4509.0\r\n",
      " E1.10.Poc.dami.1.20140728.T: 4523.0\r\n",
      " E1.15.Dip.heli.1.20140726.M: 4568.0\r\n",
      " E10.17.Fav.abdi.1.20150630.S: 4588.0\r\n",
      " E3.17.Ech.horr.1.20150123.T: 4609.0\r\n",
      " E1.19.Sym.sp.1.20140724.T: 4623.0\r\n",
      " E3.15.Dip.heli.2.20150121.T: 4681.0\r\n",
      " E7.17.Gon.edwa.2.20150623.T: 4685.0\r\n",
      " E3.19.Sym.sp.1.20150125.T: 4712.0\r\n",
      " E7.6.Mon.aequ.2.20150620.S: 4727.0\r\n",
      " E3.15.Dip.heli.1.20150118.S: 4736.0\r\n",
      " E1.10.Sty.pist.1.20140815.S: 4810.0\r\n",
      " E3.11.Fun.sp.1.20150127.M: 4878.0\r\n",
      " E1.5.Gal.astr.1.20140807.T: 4889.0\r\n",
      " E10.Poc.dami.2.20150628.M: 4954.0\r\n",
      " E7.17.Fav.spC.1.20150623.T: 4958.0\r\n",
      " E1.7.Pav.vari.1.20140805.M: 4965.0\r\n",
      " E1.3.Por.loba.1.20140724.S: 5152.0\r\n",
      " E10.Ser.cali.2.20150627.M: 5153.0\r\n",
      " E3.11.Fun.sp.1.20150125.S: 5204.0\r\n",
      " E1.6.Acr.humi.1.20140731.T: 5281.0\r\n",
      " E1.4.Pac.spec.1.20140801.T: 5320.0\r\n",
      " E1.Out.Lob.sp.1.20140803.S: 5325.0\r\n",
      " E3.15.Dip.heli.1.20150123.T: 5652.0\r\n",
      " E3.6.Acr.hyac.3.20150120.M: 5696.0\r\n",
      " E7.3.Por.loba.3.20150619.S: 5724.0\r\n",
      " E7.10.Poc.dami.2.20150621.S: 5754.0\r\n",
      " E7.17.Fav.spC.1.20150623.M: 5793.0\r\n",
      " E1.6.Mon.hisp.1.20140807.M: 5835.0\r\n",
      " E1.3.Por.loba.1.20140724.T: 5878.0\r\n",
      " E3.7.Pav.vari.2.20150124.T: 5887.0\r\n",
      " E3.6.Acr.hyac.1.20150118.M: 6035.0\r\n",
      " E10.Cyp.sera.2.20150628.S: 6081.0\r\n",
      " E1.10.Sty.pist.1.20140807.S: 6086.0\r\n",
      " E7.17.Hyd.exes.1.20150622.S: 6132.0\r\n",
      " E1.17.Ech.mamm.1.20140728.S: 6196.0\r\n",
      " E1.19.Sym.sp.1.20140724.M: 6234.0\r\n",
      " E3.6.Acr.hyac.4.20150106.T: 6245.0\r\n",
      " E3.19.Sym.sp.1.20150127.S: 6346.0\r\n",
      " E7.3.Por.loba.1.20150619.S: 6360.0\r\n",
      " E3.14.Phy.lich.1.20150125.S: 6382.0\r\n",
      " E7.7.Pav.vari.1.20150623.S: 6384.0\r\n",
      " E1.3.Por.loba.1.20140724.M: 6450.0\r\n",
      " E7.10.Poc.dami.3.20150621.S: 6471.0\r\n",
      " E1.2.Tur.stel.1.20140805.M: 6489.0\r\n",
      " E3.15.Dip.heli.3.20150120.T: 6496.0\r\n",
      " E7.17.Ech.lame.1.20150620.S: 6515.0\r\n",
      " E3.17.Ech.mamm.1.20150127.T: 6562.0\r\n",
      " E3.7.Pav.vari.1.20150124.T: 6573.0\r\n",
      " E3.6.Acr.hyac.1.20150118.T: 6694.0\r\n",
      " E3.15.Dip.heli.1.20150119.S: 6721.0\r\n",
      " E7.7.Pav.vari.2.20150622.T: 6791.0\r\n",
      " E10.Poc.dami.1.20150628.T: 6834.0\r\n",
      " E7.17.Fav.spA.2.20150623.T: 6850.0\r\n",
      " E3.15.Dip.heli.1.20150105.T: 6878.0\r\n",
      " E10.Mon.curt.2.20150629.S: 6910.0\r\n",
      " E10.Poc.dami.2.20150628.T: 6942.0\r\n",
      " E3.15.Dip.heli.3.20150121.T: 7078.0\r\n",
      " E3.2.Tur.reni.1.20150127.T: 7111.0\r\n",
      " E3.15.Dip.heli.1.20150106.M: 7172.0\r\n",
      " E3.6.Acr.hyac.4.20150106.S: 7236.0\r\n",
      " E1.7.Pav.veno.1.20140811.M: 7261.0\r\n",
      " E1.4.Pac.rugo.1.20140728.M: 7284.0\r\n",
      " E10.17.Fav.abdi.2.20150630.S: 7421.0\r\n",
      " E3.6.Acr.hyac.3.20150106.T: 7437.0\r\n",
      " E1.4.Pac.spec.1.20140801.S: 7482.0\r\n",
      " E10.Aca.hill.3.20150629.S: 7517.0\r\n",
      " E3.5.Gal.astr.2.20150127.S: 7623.0\r\n",
      " E10.Por.hero.2.20150628.S: 7626.0\r\n",
      " E7.5.Gal.fasc.1.20150622.S: 7667.0\r\n",
      " E1.Out.Mil.sp.1.20140731.S: 7697.0\r\n",
      " E1.17.Ech.mamm.1.20140815.T: 7698.0\r\n",
      " E10.Fav.abdi.1.20150629.S: 7702.0\r\n",
      " E1.17.Cau.furc.1.20140801.S: 7706.0\r\n",
      " E7.6.Acr.hyac.3.20150622.S: 7720.0\r\n",
      " E1.Out.Mil.sp.1.20140731.M: 7725.0\r\n",
      " E1.7.Pav.vari.1.20140805.T: 7751.0\r\n",
      " E3.15.Dip.heli.1.20150118.T: 7804.0\r\n",
      " E3.7.Pav.vari.2.20150124.S: 7866.0\r\n",
      " E7.10.Ser.cali.1.20150622.S: 7932.0\r\n",
      " E3.14.Phy.lich.1.20150129.S: 7970.0\r\n",
      " E1.19.Sym.sp.1.20140814.M: 7998.0\r\n",
      " E1.6.Acr.sp.1.20140726.S: 8071.0\r\n",
      " E3.4.Pac.rugo.1.20150128.S: 8078.0\r\n",
      " E3.7.Pav.vari.1.20150127.T: 8078.0\r\n",
      " E1.4.Pac.rugo.1.20140728.S: 8081.0\r\n",
      " E1.19.Lob.sp.1.20140730.S: 8089.0\r\n",
      " E3.3.Por.cyli.2.20150123.T: 8118.0\r\n",
      " E3.6.Acr.hyac.2.20150120.S: 8179.0\r\n",
      " E1.Out.Lob.sp.1.20140803.T: 8310.0\r\n",
      " E10.10.Poc.dami.1.20150630.S: 8322.0\r\n",
      " E1.10.Sty.pist.1.20140807.T: 8333.0\r\n",
      " E3.6.Acr.hyac.2.20150120.T: 8446.0\r\n",
      " E10.Sty.pist.1.20150628.S: 8462.0\r\n",
      " E3.15.Dip.heli.2.20150105.M: 8662.0\r\n",
      " E3.10.Sty.pist.2.20150124.M: 8700.0\r\n",
      " E3.5.Gal.astr.1.20150124.M: 8700.0\r\n",
      " E3.15.Dip.heli.3.20150123.T: 8711.0\r\n",
      " E10.Aca.hill.1.20150629.S: 8733.0\r\n",
      " E7.17.Fav.abdi.1.20150623.T: 8825.0\r\n",
      " E3.4.Pac.rugo.1.20150127.T: 8844.0\r\n",
      " E1.11.Lep.tran.1.20140731.M: 8850.0\r\n",
      " E1.15.Dip.heli.1.20140809.M: 8871.0\r\n",
      " E7.17.Fav.spC.1.20150623.S: 8946.0\r\n",
      " E1.6.Acr.humi.1.20140731.S: 8959.0\r\n",
      " E3.15.Dip.heli.2.20150106.T: 9032.0\r\n",
      " E3.17.Ech.mamm.1.20150129.S: 9038.0\r\n",
      " E1.10.Poc.dami.1.20140728.M: 9100.0\r\n",
      " E1.7.Pav.vari.1.20140815.M: 9121.0\r\n",
      " E1.15.Dip.heli.1.20140726.T: 9140.0\r\n",
      " E3.6.Acr.hyac.3.20150121.T: 9149.0\r\n",
      " E1.7.Pav.veno.1.20140809.M: 9250.0\r\n",
      " E3.17.Ech.mamm.1.20150123.T: 9287.0\r\n",
      " E1.15.Dip.heli.1.20140815.M: 9297.0\r\n",
      " E3.4.Pac.spec.1.20150128.T: 9325.0\r\n",
      " E10.Acr.pali.3.20150627.S: 9469.0\r\n",
      " E7.10.Ser.cali.3.20150622.S: 9472.0\r\n",
      " E3.6.Acr.hyac.4.20150105.M: 9494.0\r\n",
      " E3.14.Phy.lich.1.20150129.T: 9510.0\r\n",
      " E7.17.Fav.abdi.2.20150623.T: 9523.0\r\n",
      " E7.17.Hyd.exes.2.20150622.S: 9524.0\r\n",
      " E1.10.Sty.pist.1.20140815.M: 9623.0\r\n",
      " E3.2.Tur.reni.1.20150127.S: 9661.0\r\n",
      " E10.17.Fav.pali.1.20150630.S: 9706.0\r\n",
      " E3.17.Ech.horr.1.20150123.S: 9716.0\r\n",
      " E3.10.Sty.pist.1.20150124.M: 9732.0\r\n",
      " E7.17.Pla.sine.1.20150622.S: 9760.0\r\n",
      " E3.Out.Mil.sp.1.20150125.M: 9796.0\r\n",
      " E7.17.Ech.lame.3.20150620.S: 9799.0\r\n",
      " E3.15.Dip.heli.3.20150106.T: 9846.0\r\n",
      " E3.6.Acr.hyac.1.20150120.S: 9941.0\r\n",
      " E3.3.Por.cyli.1.20150127.S: 9966.0\r\n",
      " E1.2.Tur.reni.1.20140730.M: 9999.0\r\n",
      " E7.3.Por.loba.3.20150619.T: 10007.0\r\n",
      " E1.5.Gal.astr.1.20140807.M: 10133.0\r\n",
      " E7.17.Fav.spA.1.20150623.T: 10201.0\r\n",
      " E10.17.Gon.favu.1.20150629.S: 10212.0\r\n",
      " E10.10.Sty.pist.3.20150630.M: 10212.0\r\n",
      " E3.11.Fun.sp.2.20150125.T: 10291.0\r\n",
      " E1.17.Ech.mamm.1.20140815.M: 10295.0\r\n",
      " E3.15.Dip.heli.2.20150106.M: 10296.0\r\n",
      " E7.6.Acr.hyac.1.20150622.T: 10332.0\r\n",
      " E1.17.Fav.sp.1.20140805.M: 10355.0\r\n",
      " E1.11.Psa.cont.1.20140730.M: 10387.0\r\n",
      " E3.6.Acr.hyac.1.20150121.S: 10419.0\r\n",
      " E7.6.Mon.aequ.2.20150620.T: 10431.0\r\n",
      " E3.14.Phy.lich.2.20150125.S: 10481.0\r\n",
      " E3.6.Acr.hyac.3.20150120.S: 10502.0\r\n",
      " E10.Por.hero.3.20150628.S: 10512.0\r\n",
      " E10.Acr.pali.2.20150627.S: 10513.0\r\n",
      " E3.4.Pac.rugo.1.20150127.S: 10549.0\r\n",
      " E1.17.Ech.mamm.1.20140728.M: 10591.0\r\n",
      " E3.15.Dip.heli.2.20150119.T: 10595.0\r\n",
      " E3.6.Acr.hyac.3.20150121.M: 10706.0\r\n",
      " E1.7.Pav.sp.1.20140905.T: 10731.0\r\n",
      " E3.14.Phy.lich.1.20150129.M: 10734.0\r\n",
      " E3.7.Pav.vari.1.20150124.S: 10839.0\r\n",
      " E3.7.Pav.vari.1.20150127.S: 10953.0\r\n",
      " E3.4.Pac.spec.1.20150129.T: 10967.0\r\n",
      " E1.Out.Hel.coer.1.20140811.M: 10974.0\r\n",
      " E10.Ser.cali.3.20150627.M: 11005.0\r\n",
      " E1.5.Gal.astr.1.20140728.M: 11074.0\r\n",
      " E10.Sty.pist.1.20150628.M: 11224.0\r\n",
      " E1.2.Tur.stel.1.20140805.T: 11275.0\r\n",
      " E3.15.Dip.heli.4.20150106.T: 11328.0\r\n",
      " E1.11.Lep.tran.1.20140731.T: 11360.0\r\n",
      " E3.15.Dip.heli.1.20150118.M: 11439.0\r\n",
      " E1.3.Por.loba.1.20140726.T: 11460.0\r\n",
      " E1.6.Acr.humi.1.20140731.M: 11485.0\r\n",
      " E1.15.Dip.heli.1.20140726.S: 11540.0\r\n",
      " E7.Sed.1.20150621: 11605.0\r\n",
      " E7.17.Mer.ampl.1.20150621.T: 11605.0\r\n",
      " E1.4.Pac.rugo.1.20140728.T: 11640.0\r\n",
      " E1.17.Hyd.exes.1.20140803.M: 11669.0\r\n",
      " E1.2.Tur.reni.1.20140814.M: 11675.0\r\n",
      " E1.7.Pav.veno.1.20140809.T: 11682.0\r\n",
      " E3.6.Acr.hyac.1.20150106.S: 11700.0\r\n",
      " E3.Out.Pal.caes.1.20150125.S: 11757.0\r\n",
      " E1.17.Fav.sp.2.20140805.M: 11768.0\r\n",
      " E1.5.Gal.astr.1.20140807.S: 11801.0\r\n",
      " E10.Poc.dami.2.20150628.S: 11816.0\r\n",
      " E10.Por.hero.3.20150628.T: 11850.0\r\n",
      " E1.10.Poc.dami.1.20140731.M: 11857.0\r\n",
      " E3.17.Ech.mamm.1.20150123.S: 11859.0\r\n",
      " E7.19.Lob.hemp.2.20150619.T: 11937.0\r\n",
      " E1.11.Lep.tran.1.20140731.S: 11958.0\r\n",
      " E7.Sed.1.20150623: 12037.0\r\n",
      " E1.Out.Pal.caes.1.20140809.M: 12073.0\r\n",
      " E7.6.Acr.hyac.3.20150622.T: 12131.0\r\n",
      " E1.15.Dip.heli.1.20140905.M: 12132.0\r\n",
      " E7.17.Pla.sine.1.20150622.T: 12177.0\r\n",
      " E7.17.Myc.elep.1.20150622.S: 12216.0\r\n",
      " E1.6.Acr.sp.1.20140905.S: 12219.0\r\n",
      " E10.Alv.alli.2.20150628.S: 12252.0\r\n",
      " E1.11.Fun.sp.1.20140730.T: 12254.0\r\n",
      " E1.11.Fun.sp.1.20140730.M: 12261.0\r\n",
      " E10.17.Gon.favu.1.20150629.T: 12292.0\r\n",
      " E1.10.Poc.dami.1.20140731.T: 12309.0\r\n",
      " E1.19.Lob.sp.1.20140730.T: 12327.0\r\n",
      " E10.Acr.cyth.1.20150627.T: 12366.0\r\n",
      " E1.5.Gal.astr.1.20140728.S: 12471.0\r\n",
      " E1.Out.Lob.sp.1.20140803.M: 12475.0\r\n",
      " E1.17.Cau.furc.1.20140801.T: 12518.0\r\n",
      " E10.Sed.1.20150630: 12638.0\r\n",
      " E1.3.Por.loba.1.20140726.M: 12685.0\r\n",
      " E1.7.Pav.vari.1.20140815.S: 12686.0\r\n",
      " E3.6.Acr.hyac.1.20150118.S: 12699.0\r\n",
      " E10.17.Mon.curt.3.20150629.S: 12727.0\r\n",
      " E7.17.Fav.spA.1.20150623.M: 12747.0\r\n",
      " E1.2.Tur.pelt.1.20140814.T: 12762.0\r\n",
      " E7.5.Gal.fasc.1.20150622.T: 12834.0\r\n",
      " E3.Out.Pal.caes.1.20150125.T: 12848.0\r\n",
      " E3.15.Dip.heli.1.20150120.M: 12866.0\r\n",
      " E3.15.Dip.heli.1.20150106.T: 12870.0\r\n",
      " E7.17.Gon.edwa.1.20150623.S: 13000.0\r\n",
      " E7.17.Fav.spA.2.20150623.M: 13033.0\r\n",
      " E1.19.Sym.sp.1.20140905.M: 13096.0\r\n",
      " E3.6.Acr.hyac.1.20150123.S: 13103.0\r\n",
      " E7.6.Mon.aequ.4.20150620.T: 13151.0\r\n",
      " E1.17.Fav.sp.2.20140805.T: 13158.0\r\n",
      " E7.17.Pla.sine.2.20150622.T: 13281.0\r\n",
      " E10.17.Fav.abdi.1.20150630.T: 13351.0\r\n",
      " E1.17.Cau.furc.1.20140801.DS: 13371.0\r\n",
      " E10.Alv.alli.1.20150628.T: 13379.0\r\n",
      " E10.10.Sty.pist.2.20150630.M: 13456.0\r\n",
      " E3.3.Por.cyli.1.20150123.S: 13460.0\r\n",
      " E7.7.Pav.vari.2.20150622.S: 13468.0\r\n",
      " E3.5.Gal.astr.1.20150124.S: 13520.0\r\n",
      " E1.2.Tur.pelt.1.20140814.M: 13525.0\r\n",
      " E1.3.Por.loba.1.20140905.T: 13564.0\r\n",
      " E3.3.Por.cyli.1.20150123.T: 13625.0\r\n",
      " E1.10.Poc.dami.1.20140731.S: 13668.0\r\n",
      " E3.10.Sty.pist.1.20150127.M: 13701.0\r\n",
      " E1.19.Sym.sp.1.20140814.S: 13719.0\r\n",
      " E1.19.Lob.sp.1.20140730.M: 13733.0\r\n",
      " E3.Out.Pal.caes.2.20150125.T: 13758.0\r\n",
      " E10.Alv.alli.2.20150628.T: 13766.0\r\n",
      " E1.Out.Pal.caes.1.20140809.T: 13770.0\r\n",
      " E1.2.Tur.reni.1.20140730.T: 13771.0\r\n",
      " E3.3.Por.cyli.2.20150123.S: 13787.0\r\n",
      " E3.6.Acr.hyac.3.20150121.S: 13804.0\r\n",
      " E1.19.Sym.sp.1.20140814.T: 13818.0\r\n",
      " E1.2.Tur.reni.1.20140730.S: 13874.0\r\n",
      " E10.Alv.alli.3.20150628.S: 13879.0\r\n",
      " E1.3.Por.loba.1.20140905.M: 13894.0\r\n",
      " E3.11.Fun.sp.1.20150127.T: 13942.0\r\n",
      " E1.15.Dip.heli.1.20140815.S: 13966.0\r\n",
      " E7.17.Pla.sine.3.20150622.T: 14004.0\r\n",
      " E1.4.Pac.spec.1.20140811.T: 14016.0\r\n",
      " E1.Out.Pal.caes.1.20140809.S: 14058.0\r\n",
      " E1.4.Pac.spec.1.20140811.M: 14104.0\r\n",
      " E1.19.Sym.sp.1.20140905.S: 14166.0\r\n",
      " E1.3.Por.cyli.1.20140815.T: 14202.0\r\n",
      " E1.14.Phy.lich.1.20140801.S: 14225.0\r\n",
      " E7.3.Por.loba.4.20150619.S: 14240.0\r\n",
      " E7.19.Lob.hemp.2.20150619.S: 14261.0\r\n",
      " E7.Sed.1.20150622: 14269.0\r\n",
      " E7.17.Cyp.micr.2.20150621.S: 14282.0\r\n",
      " E10.Alv.alli.3.20150628.M: 14382.0\r\n",
      " E3.14.Phy.lich.2.20150125.M: 14440.0\r\n",
      " E1.14.Phy.lich.1.20140811.M: 14481.0\r\n",
      " E1.10.Sty.pist.1.20140807.M: 14553.0\r\n",
      " E1.17.Fav.sp.1.20140805.T: 14566.0\r\n",
      " E3.6.Acr.hyac.3.20150123.S: 14573.0\r\n",
      " E3.4.Pac.rugo.1.20150128.T: 14621.0\r\n",
      " E7.6.Acr.hyac.2.20150622.S: 14626.0\r\n",
      " E3.15.Dip.heli.4.20150105.T: 14639.0\r\n",
      " E7.19.Lob.hemp.1.20150619.T: 14654.0\r\n",
      " E7.17.Cyp.micr.3.20150621.S: 14686.0\r\n",
      " E7.17.Ech.lame.3.20150620.T: 14704.0\r\n",
      " E7.7.Pav.vari.1.20150623.M: 14704.0\r\n",
      " E7.17.Fav.spB.1.20150623.S: 14721.0\r\n",
      " E3.19.Sym.sp.1.20150125.S: 14724.0\r\n",
      " E7.Sed.1.20150620: 14765.0\r\n",
      " E1.5.Gal.astr.1.20140728.T: 14799.0\r\n",
      " E3.6.Acr.hyac.1.20150121.T: 14804.0\r\n",
      " E1.3.Por.loba.1.20140726.S: 14856.0\r\n",
      " E1.7.Pav.vari.1.20140805.S: 14858.0\r\n",
      " E10.Por.hero.1.20150628.T: 14863.0\r\n",
      " E3.15.Dip.heli.2.20150119.M: 14877.0\r\n",
      " E7.3.Por.loba.1.20150619.T: 14958.0\r\n",
      " E3.6.Acr.hyac.3.20150120.T: 14973.0\r\n",
      " E7.3.Por.loba.4.20150619.T: 14995.0\r\n",
      " E7.Sed.1.20150619: 15134.0\r\n",
      " E10.Aca.hill.3.20150629.M: 15140.0\r\n",
      " E3.6.Acr.hyac.2.20150105.T: 15221.0\r\n",
      " E10.17.Gon.favu.2.20150630.S: 15284.0\r\n",
      " E10.17.Gon.favu.1.20150630.M: 15322.0\r\n",
      " E10.10.Sty.pist.1.20150630.M: 15352.0\r\n",
      " E1.4.Pac.rugo.1.20140814.M: 15371.0\r\n",
      " E7.7.Pav.vari.1.20150623.T: 15420.0\r\n",
      " E1.7.Pav.veno.1.20140811.T: 15431.0\r\n",
      " E7.17.Hyd.exce.1.20150620.S: 15457.0\r\n",
      " E3.6.Acr.hyac.1.20150121.M: 15535.0\r\n",
      " E3.Out.Pal.caes.2.20150125.S: 15618.0\r\n",
      " E3.4.Pac.spec.1.20150128.M: 15656.0\r\n",
      " E3.15.Dip.heli.2.20150120.S: 15660.0\r\n",
      " E7.10.Poc.dami.1.20150621.M: 15705.0\r\n",
      " E3.6.Acr.hyac.2.20150121.S: 15725.0\r\n",
      " E10.17.Mon.curt.3.20150629.M: 15728.0\r\n",
      " E10.Acr.pali.2.20150627.T: 15797.0\r\n",
      " E7.6.Mon.aequ.1.20150620.S: 15819.0\r\n",
      " E3.15.Dip.heli.2.20150105.T: 15879.0\r\n",
      " E1.6.Mon.hisp.1.20140807.S: 15883.0\r\n",
      " E1.19.Sym.sp.1.20140815.T: 15883.0\r\n",
      " E7.6.Mon.aequ.4.20150620.S: 15948.0\r\n",
      " E10.Por.hero.3.20150628.M: 15976.0\r\n",
      " E7.7.Pav.vari.2.20150622.M: 15983.0\r\n",
      " E3.6.Acr.hyac.3.20150123.T: 15998.0\r\n",
      " E10.Cyp.sera.2.20150628.M: 16001.0\r\n",
      " E7.17.Myc.elep.1.20150622.T: 16042.0\r\n",
      " E7.6.Acr.hyac.1.20150622.S: 16048.0\r\n",
      " E1.7.Pav.sp.1.20140905.M: 16110.0\r\n",
      " E10.Acr.cyth.3.20150627.T: 16112.0\r\n",
      " E7.5.Gal.fasc.1.20150621.S: 16132.0\r\n",
      " E3.11.Fun.sp.1.20150125.T: 16167.0\r\n",
      " E1.2.Tur.pelt.1.20140814.S: 16191.0\r\n",
      " E3.5.Gal.astr.2.20150124.T: 16244.0\r\n",
      " E1.7.Pav.vari.1.20140815.T: 16254.0\r\n",
      " E7.17.Mer.ampl.2.20150621.T: 16316.0\r\n",
      " E1.19.Sym.sp.1.20140905.T: 16446.0\r\n",
      " E1.6.Iso.pali.1.20140803.M: 16469.0\r\n",
      " E1.17.Hyd.exes.1.20140803.S: 16484.0\r\n",
      " E10.Por.hero.2.20150628.T: 16494.0\r\n",
      " E3.6.Acr.hyac.4.20150105.S: 16508.0\r\n",
      " E1.17.Fav.sp.1.20140805.S: 16542.0\r\n",
      " E3.6.Acr.hyac.2.20150106.M: 16605.0\r\n",
      " E1.3.Por.cyli.1.20140801.S: 16629.0\r\n",
      " E7.17.Ech.lame.2.20150620.T: 16650.0\r\n",
      " E10.Fav.abdi.1.20150629.T: 16677.0\r\n",
      " E3.15.Dip.heli.2.20150123.M: 16697.0\r\n",
      " E3.15.Dip.heli.2.20150119.S: 16707.0\r\n",
      " E3.3.Por.cyli.1.20150123.M: 16743.0\r\n",
      " E10.Sty.pist.1.20150628.T: 16760.0\r\n",
      " E10.10.Poc.dami.1.20150630.T: 16791.0\r\n",
      " E1.7.Pav.veno.1.20140811.S: 16884.0\r\n",
      " E3.17.Ech.mamm.1.20150129.M: 16913.0\r\n",
      " E1.11.Psa.cont.1.20140730.S: 16932.0\r\n",
      " E1.17.Ech.lame.1.20140905.T: 16973.0\r\n",
      " E7.17.Fav.abdi.1.20150623.M: 16982.0\r\n",
      " E1.5.Gal.astr.1.20140814.S: 16996.0\r\n",
      " E3.15.Dip.heli.2.20150121.M: 17017.0\r\n",
      " E7.19.Lob.hemp.1.20150621.T: 17030.0\r\n",
      " E10.17.Gon.favu.1.20150630.T: 17038.0\r\n",
      " E3.15.Dip.heli.3.20150105.T: 17105.0\r\n",
      " E10.Water.1.20150629: 17107.0\r\n",
      " E10.Mon.curt.1.20150629.S: 17126.0\r\n",
      " E3.4.Pac.spec.1.20150129.S: 17135.0\r\n",
      " E7.3.Por.loba.3.20150619.M: 17158.0\r\n",
      " E7.17.Pla.sine.2.20150622.S: 17182.0\r\n",
      " E1.17.Cau.furc.1.20140801.M: 17185.0\r\n",
      " E1.17.Fav.sp.1.20140811.S: 17261.0\r\n",
      " E3.6.Acr.hyac.2.20150105.S: 17279.0\r\n",
      " E3.3.Por.cyli.1.20150127.T: 17409.0\r\n",
      " E10.17.Gon.favu.2.20150630.T: 17422.0\r\n",
      " E7.7.Pav.vari.1.20150622.M: 17476.0\r\n",
      " E3.7.Pav.vari.1.20150124.M: 17487.0\r\n",
      " E1.2.Tur.reni.1.20140814.T: 17488.0\r\n",
      " E7.6.Mon.aequ.1.20150620.T: 17498.0\r\n",
      " E7.19.Lob.hemp.1.20150619.S: 17500.0\r\n",
      " E3.6.Acr.hyac.2.20150121.M: 17505.0\r\n",
      " E1.11.Psa.cont.1.20140730.T: 17506.0\r\n",
      " E10.Poc.dami.1.20150628.M: 17559.0\r\n",
      " E1.5.Gal.astr.1.20140814.T: 17574.0\r\n",
      " E3.6.Acr.hyac.4.20150105.T: 17590.0\r\n",
      " E10.Acr.pali.2.20150627.M: 17591.0\r\n",
      " E1.6.Acr.sp.1.20140905.M: 17681.0\r\n",
      " E7.6.Acr.hyac.3.20150622.M: 17687.0\r\n",
      " E1.3.Por.cyli.1.20140815.M: 17691.0\r\n",
      " E10.Acr.pali.3.20150627.T: 17708.0\r\n",
      " E10.Water.2.20150628: 17728.0\r\n",
      " E3.4.Pac.rugo.1.20150128.M: 17770.0\r\n",
      " E3.6.Acr.hyac.3.20150105.T: 17793.0\r\n",
      " E3.15.Dip.heli.2.20150105.S: 17815.0\r\n",
      " E10.Aca.hill.1.20150629.M: 17816.0\r\n",
      " E3.4.Pac.spec.1.20150129.M: 17829.0\r\n",
      " E1.6.Iso.pali.1.20140803.T: 17851.0\r\n",
      " E3.3.Por.cyli.1.20150127.M: 17871.0\r\n",
      " E10.Sed.1.20150628: 17898.0\r\n",
      " E7.3.Por.loba.2.20150619.T: 17986.0\r\n",
      " E7.17.Ech.lame.1.20150620.T: 18042.0\r\n",
      " E3.15.Dip.heli.3.20150105.S: 18042.0\r\n",
      " E10.Por.hero.2.20150628.M: 18085.0\r\n",
      " E7.17.Myc.elep.1.20150622.M: 18110.0\r\n",
      " E1.2.Tur.stel.1.20140805.S: 18125.0\r\n",
      " E3.15.Dip.heli.1.20150121.S: 18126.0\r\n",
      " E3.11.Fun.sp.2.20150125.M: 18141.0\r\n",
      " E3.15.Dip.heli.1.20150123.M: 18160.0\r\n",
      " E3.15.Dip.heli.3.20150106.S: 18171.0\r\n",
      " E3.4.Pac.rugo.1.20150127.M: 18194.0\r\n",
      " E1.7.Pav.veno.1.20140809.S: 18197.0\r\n",
      " E3.2.Tur.reni.2.20150124.M: 18213.0\r\n",
      " E1.17.Fav.sp.1.20140811.M: 18228.0\r\n",
      " E1.6.Iso.pali.1.20140815.S: 18233.0\r\n",
      " E7.17.Gon.edwa.1.20150623.T: 18255.0\r\n",
      " E1.15.Dip.heli.1.20140815.T: 18283.0\r\n",
      " E1.19.Sym.sp.1.20140815.M: 18286.0\r\n",
      " E3.15.Dip.heli.3.20150120.M: 18290.0\r\n",
      " E1.17.Hyd.exes.1.20140803.T: 18291.0\r\n",
      " E3.19.Sym.sp.1.20150127.M: 18350.0\r\n",
      " E7.17.Hyd.exes.1.20150622.T: 18385.0\r\n",
      " E1.19.Sym.sp.1.20140724.S: 18388.0\r\n",
      " E1.3.Por.cyli.1.20140801.M: 18430.0\r\n",
      " E1.6.Iso.pali.1.20140815.T: 18495.0\r\n",
      " E7.17.Gon.edwa.1.20150623.M: 18509.0\r\n",
      " E1.Out.Mil.sp.1.20140814.M: 18582.0\r\n",
      " E10.Alv.alli.3.20150628.T: 18588.0\r\n",
      " E7.17.Pla.sine.2.20150622.M: 18614.0\r\n",
      " E1.6.Iso.pali.1.20140803.S.2: 18622.0\r\n",
      " E3.15.Dip.heli.1.20150119.M: 18647.0\r\n",
      " E3.5.Gal.astr.1.20150124.T: 18651.0\r\n",
      " E1.15.Dip.heli.1.20140905.T: 18664.0\r\n",
      " E3.2.Tur.reni.1.20150124.M: 18704.0\r\n",
      " E3.7.Pav.vari.1.20150127.M: 18711.0\r\n",
      " E10.Mon.curt.1.20150629.T: 18712.0\r\n",
      " E3.6.Acr.hyac.1.20150123.T: 18730.0\r\n",
      " E7.19.Lob.hemp.2.20150619.M: 18732.0\r\n",
      " E1.5.Gal.astr.1.20140905.T: 18738.0\r\n",
      " E1.17.Ech.lame.1.20140905.S: 18768.0\r\n",
      " E1.4.Pac.spec.1.20140811.S: 18807.0\r\n",
      " E3.15.Dip.heli.3.20150121.M: 18820.0\r\n",
      " E7.10.Poc.dami.2.20150621.M: 18844.0\r\n",
      " E1.3.Por.cyli.1.20140801.T: 18858.0\r\n",
      " E3.6.Acr.hyac.3.20150105.M: 18872.0\r\n",
      " E7.17.Fav.abdi.1.20150623.S: 18892.0\r\n",
      " E7.6.Mon.aequ.3.20150620.T: 18900.0\r\n",
      " E3.14.Phy.lich.1.20150125.M: 19007.0\r\n",
      " E1.19.Lob.sp.1.20140809.S: 19114.0\r\n",
      " E3.6.Acr.hyac.2.20150123.T: 19142.0\r\n",
      " E3.6.Acr.hyac.2.20150106.S: 19219.0\r\n",
      " E3.6.Acr.hyac.1.20150106.T: 19265.0\r\n",
      " E7.19.Lob.hemp.1.20150621.S: 19320.0\r\n",
      " E7.17.Pla.sine.3.20150622.S: 19321.0\r\n",
      " E7.5.Gal.fasc.1.20150622.M: 19420.0\r\n",
      " E1.17.Fav.sp.1.20140811.T: 19436.0\r\n",
      " E1.5.Gal.astr.1.20140905.S: 19441.0\r\n",
      " E1.17.Fav.sp.2.20140805.S: 19459.0\r\n",
      " E3.6.Acr.hyac.2.20150123.S: 19493.0\r\n",
      " E7.17.Mer.ampl.3.20150621.T: 19502.0\r\n",
      " E3.15.Dip.heli.4.20150105.S: 19552.0\r\n",
      " E10.Mon.curt.1.20150629.M: 19590.0\r\n",
      " E10.10.Poc.dami.1.20150630.M: 19604.0\r\n",
      " E1.15.Dip.heli.1.20140809.S: 19656.0\r\n",
      " E7.17.Hyd.exes.2.20150622.M: 19656.0\r\n",
      " E3.15.Dip.heli.1.20150105.S: 19728.0\r\n",
      " E7.17.Hyd.exce.1.20150620.M: 19779.0\r\n",
      " E7.17.Hyd.exce.1.20150620.T: 19804.0\r\n",
      " E7.17.Ech.lame.2.20150620.M: 19849.0\r\n",
      " E7.17.Cyp.micr.3.20150621.M: 19867.0\r\n",
      " E1.3.Por.cyli.1.20140815.S: 19883.0\r\n",
      " E1.Out.Mil.sp.1.20140814.S: 19950.0\r\n",
      " E1.19.Lob.sp.1.20140809.M: 20084.0\r\n",
      " E3.15.Dip.heli.1.20150123.S: 20096.0\r\n",
      " E3.2.Tur.reni.1.20150127.M: 20125.0\r\n",
      " E1.6.Acr.sp.1.20140905.T: 20168.0\r\n",
      " E1.Out.Hel.coer.1.20140811.T: 20222.0\r\n",
      " E7.17.Fav.spA.2.20150623.S: 20268.0\r\n",
      " E10.17.Mon.curt.3.20150629.T: 20356.0\r\n",
      " E10.17.Gon.favu.1.20150630.S: 20382.0\r\n",
      " E3.6.Acr.hyac.1.20150105.S: 20410.0\r\n",
      " E3.19.Sym.sp.1.20150125.M: 20425.0\r\n",
      " E7.19.Lob.hemp.1.20150619.M: 20461.0\r\n",
      " E1.7.Pav.sp.1.20140905.S: 20517.0\r\n",
      " E7.3.Por.loba.2.20150619.M: 20540.0\r\n",
      " E1.6.Iso.pali.1.20140803.S: 20546.0\r\n",
      " E3.Out.Pal.caes.1.20150125.M: 20587.0\r\n",
      " E1.5.Gal.astr.1.20140814.M: 20640.0\r\n",
      " E7.5.Gal.astr.1.20150622.S: 20676.0\r\n",
      " E1.14.Phy.lich.1.20140811.T: 20731.0\r\n",
      " E1.17.Ech.lame.1.20140905.M: 20731.0\r\n",
      " E10.Cyp.micr.1.20150627.T: 20749.0\r\n",
      " E7.6.Mon.aequ.3.20150620.S: 20798.0\r\n",
      " E7.17.Ech.lame.3.20150620.M: 20835.0\r\n",
      " E1.11.Psa.cont.1.20140730.TS: 20837.0\r\n",
      " E1.Out.Pal.caes.1.20140815.T: 20853.0\r\n",
      " E7.5.Gal.fasc.1.20150621.M: 20962.0\r\n",
      " E3.15.Dip.heli.3.20150120.S: 21088.0\r\n",
      " E10.17.Gon.favu.2.20150630.M: 21147.0\r\n",
      " E10.Acr.cyth.3.20150627.S: 21149.0\r\n",
      " E1.6.Iso.pali.1.20140815.M: 21159.0\r\n",
      " E3.15.Dip.heli.4.20150105.M: 21188.0\r\n",
      " E7.17.Ech.lame.2.20150620.S: 21203.0\r\n",
      " E1.11.Psa.digi.1.20140807.M: 21342.0\r\n",
      " E3.6.Acr.hyac.1.20150123.M: 21371.0\r\n",
      " E3.7.Pav.vari.2.20150124.M: 21392.0\r\n",
      " E3.5.Gal.astr.2.20150127.M: 21434.0\r\n",
      " E3.15.Dip.heli.4.20150106.S: 21561.0\r\n",
      " E1.15.Dip.heli.1.20140905.S: 21633.0\r\n",
      " E7.17.Fav.spA.1.20150623.S: 21653.0\r\n",
      " E1.10.Sty.pist.1.20140815.T: 21705.0\r\n",
      " E1.4.Pac.rugo.1.20140814.T: 21754.0\r\n",
      " E10.Mon.curt.2.20150629.T: 21799.0\r\n",
      " E3.5.Gal.astr.2.20150124.M: 21831.0\r\n",
      " E7.17.Cyp.micr.1.20150621.S: 21929.0\r\n",
      " E1.19.Lob.sp.1.20140809.T: 21933.0\r\n",
      " E1.Out.Mil.sp.1.20140731.T: 21952.0\r\n",
      " E3.19.Sym.sp.2.20150125.S: 21981.0\r\n",
      " E1.6.Iso.pali.1.20140803.M.2: 21994.0\r\n",
      " E3.15.Dip.heli.1.20150106.S: 22035.0\r\n",
      " E3.15.Dip.heli.3.20150123.S: 22173.0\r\n",
      " E7.6.Mon.aequ.1.20150620.M: 22187.0\r\n",
      " E1.19.Sym.sp.1.20140815.S: 22229.0\r\n",
      " E7.17.Pla.sine.3.20150622.M: 22250.0\r\n",
      " E7.17.Hyd.exes.2.20150622.T: 22399.0\r\n",
      " E3.5.Gal.astr.2.20150124.S: 22464.0\r\n",
      " E7.5.Gal.astr.1.20150622.T: 22597.0\r\n",
      " E7.6.Mon.aequ.3.20150620.M: 22610.0\r\n",
      " E3.6.Acr.hyac.2.20150120.M: 22656.0\r\n",
      " E7.17.Fav.abdi.2.20150623.M: 22667.0\r\n",
      " E3.15.Dip.heli.1.20150105.M: 22674.0\r\n",
      " E7.10.Poc.dami.3.20150621.M: 22687.0\r\n",
      " E10.Acr.cyth.1.20150627.S: 22724.0\r\n",
      " E3.15.Dip.heli.2.20150123.S: 22874.0\r\n",
      " E7.10.Ser.cali.3.20150622.M: 22919.0\r\n",
      " E3.15.Dip.heli.3.20150105.M: 22931.0\r\n",
      " E1.15.Dip.heli.1.20140809.T: 22945.0\r\n",
      " E10.Mon.curt.2.20150629.M: 23048.0\r\n",
      " E10.Acr.pali.1.20150627.M: 23067.0\r\n",
      " E3.3.Por.cyli.2.20150123.M: 23076.0\r\n",
      " E10.Ser.cali.1.20150627.M: 23096.0\r\n",
      " E10.Por.hero.1.20150628.M: 23099.0\r\n",
      " E3.17.Ech.mamm.1.20150127.M: 23107.0\r\n",
      " E1.Out.Pal.caes.1.20140815.S: 23128.0\r\n",
      " E7.17.Cyp.micr.2.20150621.M: 23141.0\r\n",
      " E3.6.Acr.hyac.3.20150105.S: 23211.0\r\n",
      " E10.17.Fav.abdi.2.20150630.M: 23282.0\r\n",
      " E1.6.Iso.pali.1.20140803.T.2: 23341.0\r\n",
      " E3.15.Dip.heli.2.20150120.M: 23362.0\r\n",
      " E3.6.Acr.hyac.2.20150106.T: 23388.0\r\n",
      " E3.15.Dip.heli.2.20150106.S: 23400.0\r\n",
      " E7.17.Gon.edwa.2.20150623.M: 23443.0\r\n",
      " E3.6.Acr.hyac.1.20150120.M: 23484.0\r\n",
      " E7.17.Cyp.micr.1.20150621.M: 23530.0\r\n",
      " E10.Acr.cyth.3.20150627.M: 23539.0\r\n",
      " E1.11.Psa.digi.1.20140807.S: 23543.0\r\n",
      " E7.6.Acr.hyac.2.20150622.T: 23583.0\r\n",
      " E1.Out.Hel.coer.1.20140811.S: 23588.0\r\n",
      " E3.15.Dip.heli.3.20150123.M: 23655.0\r\n",
      " E1.5.Gal.astr.1.20140905.M: 23776.0\r\n",
      " E1.Out.Pal.caes.1.20140815.M: 23859.0\r\n",
      " E7.17.Hyd.exes.1.20150622.M: 24010.0\r\n",
      " E10.Fav.abdi.1.20150629.M: 24124.0\r\n",
      " E7.10.Ser.cali.2.20150622.M: 24222.0\r\n",
      " E10.Acr.cyth.2.20150627.S: 24267.0\r\n",
      " E7.5.Gal.fasc.1.20150621.T: 24385.0\r\n",
      " E7.3.Por.loba.4.20150619.M: 24448.0\r\n",
      " E10.Cyp.micr.1.20150627.M: 24528.0\r\n",
      " E3.6.Acr.hyac.1.20150105.M: 24644.0\r\n",
      " E10.17.Gon.favu.1.20150629.M: 24699.0\r\n",
      " E1.Out.Mil.sp.1.20140814.T: 24730.0\r\n",
      " E10.Acr.pali.3.20150627.M: 24768.0\r\n",
      " E1.14.Phy.lich.1.20140811.S: 24846.0\r\n",
      " E7.17.Fav.spB.1.20150623.M: 25081.0\r\n",
      " E3.15.Dip.heli.1.20150120.S: 25191.0\r\n",
      " E3.6.Acr.hyac.2.20150123.M: 25208.0\r\n",
      " E1.11.Psa.digi.1.20140807.T: 25567.0\r\n",
      " E3.6.Acr.hyac.3.20150123.M: 25664.0\r\n",
      " E10.Sed.1.20150627: 25668.0\r\n",
      " E7.17.Mer.ampl.3.20150621.M: 25690.0\r\n",
      " E7.3.Por.loba.1.20150619.M: 25879.0\r\n",
      " E3.6.Acr.hyac.1.20150106.M: 25937.0\r\n",
      " E3.Out.Pal.caes.2.20150125.M: 26230.0\r\n",
      " E10.Acr.cyth.2.20150627.T: 26327.0\r\n",
      " E10.17.Fav.abdi.1.20150630.M: 26462.0\r\n",
      " E1.4.Pac.rugo.1.20140814.S: 26511.0\r\n",
      " E7.17.Ech.lame.1.20150620.M: 26583.0\r\n",
      " E7.6.Mon.aequ.4.20150620.M: 26690.0\r\n",
      " E7.6.Mon.aequ.2.20150620.M: 26877.0\r\n",
      " E7.6.Acr.hyac.2.20150622.M: 26934.0\r\n",
      " E7.5.Gal.astr.1.20150622.M: 27075.0\r\n",
      " E7.17.Mer.ampl.1.20150621.M: 27128.0\r\n",
      " E10.Acr.cyth.2.20150627.M: 27180.0\r\n",
      " E3.15.Dip.heli.2.20150121.S: 27242.0\r\n",
      " E3.15.Dip.heli.3.20150121.S: 27387.0\r\n",
      " E1.3.Por.loba.1.20140905.S: 27482.0\r\n",
      " E7.6.Acr.hyac.1.20150622.M: 27595.0\r\n",
      " E1.11.Fun.sp.1.20140811.T: 27603.0\r\n",
      " E3.Out.Mil.sp.1.20150125.T: 27894.0\r\n",
      " E7.10.Ser.cali.1.20150622.M: 28052.0\r\n",
      " E10.Aca.hill.2.20150629.M: 28263.0\r\n",
      " E3.15.Dip.heli.1.20150121.M: 28304.0\r\n",
      " E10.Alv.alli.1.20150628.M: 28318.0\r\n",
      " E3.19.Sym.sp.2.20150125.M: 28347.0\r\n",
      " E3.6.Acr.hyac.2.20150105.M: 28794.0\r\n",
      " E3.Out.Mil.sp.2.20150125.T: 28870.0\r\n",
      " E3.17.Ech.mamm.1.20150123.M: 28896.0\r\n",
      " E7.19.Lob.hemp.1.20150621.M: 29541.0\r\n",
      " E3.11.Fun.sp.1.20150125.M: 30168.0\r\n",
      " E10.Alv.alli.2.20150628.M: 31070.0\r\n",
      " E3.Out.Mil.sp.2.20150125.S: 31091.0\r\n",
      " E1.11.Fun.sp.1.20140811.M: 32780.0\r\n",
      " E3.Out.Mil.sp.1.20150125.S: 33141.0\r\n",
      " E1.11.Fun.sp.1.20140811.S: 35581.0\r\n",
      " E10.Acr.cyth.1.20150627.M: 38523.0\r\n"
     ]
    }
   ],
   "source": [
    "#Summarize BIOM\n",
    "!biom summarize-table -i $otu_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
