{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 16S_procedure_3b_bdiv_australia_analysis: Here we will test the hypothesis that compartments have distinct microbial communities via beta-diversity assessment\n",
    "\n",
    "# NOTE: This analysis uses the updated Greengenes constrained tree."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis\n"
     ]
    }
   ],
   "source": [
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Import libraries\n",
    "from os.path import join,abspath\n",
    "from os import listdir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Set file paths\n",
    "input_dir = abspath(\"input/\")\n",
    "otu_table = abspath(\"input/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom\")\n",
    "tree_fp = abspath(\"input/gg_constrained_rep_set_fastttree.tre\")\n",
    "mapping = abspath(\"input/gcmp16S_map_r22.txt\")\n",
    "output_dir = abspath(\"output/\")\n",
    "\n",
    "#Preference file indicating that Curtis-Brays distance should also be calculated by beta_diversity\n",
    "bc_prefs_fp = abspath(\"input/bdiv_bc_prefs.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Num samples: 692\r\n",
      "Num observations: 129305\r\n",
      "Total count: 9441738\r\n",
      "Table density (fraction of non-zero values): 0.006\r\n",
      "\r\n",
      "Counts/sample summary:\r\n",
      " Min: 2.0\r\n",
      " Max: 38523.0\r\n",
      " Median: 14010.000\r\n",
      " Mean: 13644.130\r\n",
      " Std. dev.: 7564.457\r\n",
      " Sample Metadata Categories: None provided\r\n",
      " Observation Metadata Categories: taxonomy\r\n",
      "\r\n",
      "Counts/sample detail:\r\n",
      " E7.17.Pla.sine.1.20150622.M: 2.0\r\n",
      " E3.17.Ech.mamm.1.20150127.S: 3.0\r\n",
      " E10.17.Fav.pali.2.20150630.T: 5.0\r\n",
      " E7.17.Fav.spB.1.20150623.T: 7.0\r\n",
      " E3.17.Ech.horr.1.20150123.M: 8.0\r\n",
      " E3.15.Dip.heli.2.20150120.T: 9.0\r\n",
      " E10.Cyp.sera.1.20150628.M: 17.0\r\n",
      " E10.17.Fav.pali.1.20150630.T: 17.0\r\n",
      " E3.Out.Mil.sp.2.20150125.M: 24.0\r\n",
      " E10.17.Fav.pali.1.20150630.M: 36.0\r\n",
      " E10.17.Fav.pali.2.20150630.S: 41.0\r\n",
      " E10.Cyp.sera.2.20150628.T: 42.0\r\n",
      " E1.6.Acr.sp.1.20140726.T: 42.0\r\n",
      " E3.5.Gal.astr.2.20150127.T: 53.0\r\n",
      " E10.Ser.cali.2.20150627.T: 66.0\r\n",
      " E10.Alv.alli.1.20150628.S: 96.0\r\n",
      " E10.Ser.cali.1.20150627.T: 102.0\r\n",
      " E10.17.Fav.pali.2.20150630.M: 127.0\r\n",
      " E10.17.Fav.abdi.2.20150630.T: 165.0\r\n",
      " E7.17.Gon.edwa.2.20150623.S: 197.0\r\n",
      " E10.17.Fav.pali.3.20150630.T: 200.0\r\n",
      " E10.17.Fav.pali.3.20150630.S: 252.0\r\n",
      " E10.17.Fav.pali.3.20150630.M: 276.0\r\n",
      " E3.6.Acr.hyac.3.20150106.S: 470.0\r\n",
      " E10.Ser.cali.3.20150627.T: 522.0\r\n",
      " E10.10.Sty.pist.2.20150630.S: 566.0\r\n",
      " E3.6.Acr.hyac.1.20150120.T: 576.0\r\n",
      " E3.2.Tur.reni.2.20150124.S: 627.0\r\n",
      " E7.7.Pav.vari.1.20150622.T: 690.0\r\n",
      " E3.2.Tur.reni.2.20150124.T: 788.0\r\n",
      " E3.6.Acr.hyac.4.20150106.M: 941.0\r\n",
      " E10.Ser.cali.3.20150627.S: 1004.0\r\n",
      " E3.10.Sty.pist.2.20150124.S: 1005.0\r\n",
      " E1.6.Acr.sp.1.20140726.M: 1009.0\r\n",
      " E7.10.Poc.dami.2.20150621.T: 1033.0\r\n",
      " E7.10.Poc.dami.1.20150621.T: 1114.0\r\n",
      " E3.2.Tur.reni.1.20150124.S: 1138.0\r\n",
      " E1.10.Poc.dami.1.20140728.S: 1176.0\r\n",
      " E10.10.Sty.pist.3.20150630.T: 1256.0\r\n",
      " E10.10.Sty.pist.3.20150630.S: 1270.0\r\n",
      " E3.10.Sty.pist.1.20150124.S: 1358.0\r\n",
      " E10.Ser.cali.1.20150627.S: 1390.0\r\n",
      " E3.11.Fun.sp.2.20150125.S: 1415.0\r\n",
      " Negative.Control.2: 1442.0\r\n",
      " E7.17.Mer.ampl.3.20150621.S: 1468.0\r\n",
      " E3.10.Sty.pist.1.20150127.S: 1490.0\r\n",
      " E10.Ser.cali.2.20150627.S: 1547.0\r\n",
      " E3.15.Dip.heli.2.20150123.T: 1558.0\r\n",
      " E7.17.Cyp.micr.1.20150621.T: 1563.0\r\n",
      " E7.17.Mer.ampl.1.20150621.S: 1595.0\r\n",
      " E7.17.Mer.ampl.2.20150621.S: 1640.0\r\n",
      " E3.2.Tur.reni.1.20150124.T: 1673.0\r\n",
      " E3.17.Ech.mamm.1.20150129.T: 1678.0\r\n",
      " E3.10.Sty.pist.1.20150124.T: 1700.0\r\n",
      " E3.6.Acr.hyac.2.20150121.T: 1763.0\r\n",
      " Negative.Control.1: 1771.0\r\n",
      " E1.17.Ech.mamm.1.20140815.S: 1993.0\r\n",
      " E1.14.Phy.lich.1.20140801.M: 2027.0\r\n",
      " E3.10.Sty.pist.1.20150127.T: 2059.0\r\n",
      " E10.10.Sty.pist.1.20150630.T: 2109.0\r\n",
      " Negative.Control.3: 2120.0\r\n",
      " E1.2.Tur.reni.1.20140814.S: 2194.0\r\n",
      " E3.10.Sty.pist.2.20150124.T: 2232.0\r\n",
      " E7.10.Ser.cali.3.20150622.T: 2268.0\r\n",
      " E7.7.Pav.vari.1.20150622.S: 2309.0\r\n",
      " E3.15.Dip.heli.1.20150119.T: 2325.0\r\n",
      " E3.4.Pac.spec.1.20150128.S: 2332.0\r\n",
      " E3.14.Phy.lich.2.20150125.T: 2353.0\r\n",
      " E10.Por.hero.1.20150628.S: 2355.0\r\n",
      " Negative.Control.4: 2493.0\r\n",
      " E7.10.Poc.dami.3.20150621.T: 2544.0\r\n",
      " E1.14.Phy.lich.1.20140801.T: 2568.0\r\n",
      " E10.Cyp.micr.1.20150627.S: 2638.0\r\n",
      " E3.14.Phy.lich.1.20150125.T: 2679.0\r\n",
      " E3.15.Dip.heli.3.20150106.M: 2719.0\r\n",
      " E7.10.Ser.cali.2.20150622.S: 3296.0\r\n",
      " E1.17.Ech.mamm.1.20140728.T: 3523.0\r\n",
      " E10.Aca.hill.2.20150629.T: 3617.0\r\n",
      " E7.17.Fav.abdi.2.20150623.S: 3626.0\r\n",
      " E10.10.Sty.pist.1.20150630.S: 3678.0\r\n",
      " E7.10.Ser.cali.1.20150622.T: 3837.0\r\n",
      " E1.6.Mon.hisp.1.20140807.T: 3857.0\r\n",
      " E3.19.Sym.sp.2.20150125.T: 3902.0\r\n",
      " E1.4.Pac.spec.1.20140801.M: 3954.0\r\n",
      " E3.11.Fun.sp.1.20150127.S: 3963.0\r\n",
      " E10.10.Sty.pist.2.20150630.T: 3977.0\r\n",
      " E10.Aca.hill.2.20150629.S: 3979.0\r\n",
      " E7.3.Por.loba.2.20150619.S: 4098.0\r\n",
      " E3.15.Dip.heli.1.20150121.T: 4127.0\r\n",
      " E10.Poc.dami.1.20150628.S: 4135.0\r\n",
      " E1.10.Poc.dami.1.20140905.S: 4139.0\r\n",
      " E7.10.Poc.dami.1.20150621.S: 4148.0\r\n",
      " E7.10.Ser.cali.2.20150622.T: 4274.0\r\n",
      " E3.19.Sym.sp.1.20150127.T: 4368.0\r\n",
      " E3.6.Acr.hyac.1.20150105.T: 4432.0\r\n",
      " E10.Aca.hill.1.20150629.T: 4436.0\r\n",
      " E1.10.Poc.dami.1.20140905.T: 4444.0\r\n",
      " E10.Aca.hill.3.20150629.T: 4458.0\r\n",
      " E3.15.Dip.heli.1.20150120.T: 4509.0\r\n",
      " E1.10.Poc.dami.1.20140728.T: 4523.0\r\n",
      " E1.15.Dip.heli.1.20140726.M: 4568.0\r\n",
      " E10.17.Fav.abdi.1.20150630.S: 4588.0\r\n",
      " E3.17.Ech.horr.1.20150123.T: 4609.0\r\n",
      " E1.19.Sym.sp.1.20140724.T: 4623.0\r\n",
      " E3.15.Dip.heli.2.20150121.T: 4681.0\r\n",
      " E7.17.Gon.edwa.2.20150623.T: 4685.0\r\n",
      " E3.19.Sym.sp.1.20150125.T: 4712.0\r\n",
      " E7.6.Mon.aequ.2.20150620.S: 4727.0\r\n",
      " E3.15.Dip.heli.1.20150118.S: 4736.0\r\n",
      " E1.10.Sty.pist.1.20140815.S: 4810.0\r\n",
      " E3.11.Fun.sp.1.20150127.M: 4878.0\r\n",
      " E1.5.Gal.astr.1.20140807.T: 4889.0\r\n",
      " E10.Poc.dami.2.20150628.M: 4954.0\r\n",
      " E7.17.Fav.spC.1.20150623.T: 4958.0\r\n",
      " E1.7.Pav.vari.1.20140805.M: 4965.0\r\n",
      " E1.3.Por.loba.1.20140724.S: 5152.0\r\n",
      " E10.Ser.cali.2.20150627.M: 5153.0\r\n",
      " E3.11.Fun.sp.1.20150125.S: 5204.0\r\n",
      " E1.6.Acr.humi.1.20140731.T: 5281.0\r\n",
      " E1.4.Pac.spec.1.20140801.T: 5320.0\r\n",
      " E1.Out.Lob.sp.1.20140803.S: 5325.0\r\n",
      " E3.15.Dip.heli.1.20150123.T: 5652.0\r\n",
      " E3.6.Acr.hyac.3.20150120.M: 5696.0\r\n",
      " E7.3.Por.loba.3.20150619.S: 5724.0\r\n",
      " E7.10.Poc.dami.2.20150621.S: 5754.0\r\n",
      " E7.17.Fav.spC.1.20150623.M: 5793.0\r\n",
      " E1.6.Mon.hisp.1.20140807.M: 5835.0\r\n",
      " E1.3.Por.loba.1.20140724.T: 5878.0\r\n",
      " E3.7.Pav.vari.2.20150124.T: 5887.0\r\n",
      " E3.6.Acr.hyac.1.20150118.M: 6035.0\r\n",
      " E10.Cyp.sera.2.20150628.S: 6081.0\r\n",
      " E1.10.Sty.pist.1.20140807.S: 6086.0\r\n",
      " E7.17.Hyd.exes.1.20150622.S: 6132.0\r\n",
      " E1.17.Ech.mamm.1.20140728.S: 6196.0\r\n",
      " E1.19.Sym.sp.1.20140724.M: 6234.0\r\n",
      " E3.6.Acr.hyac.4.20150106.T: 6245.0\r\n",
      " E3.19.Sym.sp.1.20150127.S: 6346.0\r\n",
      " E7.3.Por.loba.1.20150619.S: 6360.0\r\n",
      " E3.14.Phy.lich.1.20150125.S: 6382.0\r\n",
      " E7.7.Pav.vari.1.20150623.S: 6384.0\r\n",
      " E1.3.Por.loba.1.20140724.M: 6450.0\r\n",
      " E7.10.Poc.dami.3.20150621.S: 6471.0\r\n",
      " E1.2.Tur.stel.1.20140805.M: 6489.0\r\n",
      " E3.15.Dip.heli.3.20150120.T: 6496.0\r\n",
      " E7.17.Ech.lame.1.20150620.S: 6515.0\r\n",
      " E3.17.Ech.mamm.1.20150127.T: 6562.0\r\n",
      " E3.7.Pav.vari.1.20150124.T: 6573.0\r\n",
      " E3.6.Acr.hyac.1.20150118.T: 6694.0\r\n",
      " E3.15.Dip.heli.1.20150119.S: 6721.0\r\n",
      " E7.7.Pav.vari.2.20150622.T: 6791.0\r\n",
      " E10.Poc.dami.1.20150628.T: 6834.0\r\n",
      " E7.17.Fav.spA.2.20150623.T: 6850.0\r\n",
      " E3.15.Dip.heli.1.20150105.T: 6878.0\r\n",
      " E10.Mon.curt.2.20150629.S: 6910.0\r\n",
      " E10.Poc.dami.2.20150628.T: 6942.0\r\n",
      " E3.15.Dip.heli.3.20150121.T: 7078.0\r\n",
      " E3.2.Tur.reni.1.20150127.T: 7111.0\r\n",
      " E3.15.Dip.heli.1.20150106.M: 7172.0\r\n",
      " E3.6.Acr.hyac.4.20150106.S: 7236.0\r\n",
      " E1.7.Pav.veno.1.20140811.M: 7261.0\r\n",
      " E1.4.Pac.rugo.1.20140728.M: 7284.0\r\n",
      " E10.17.Fav.abdi.2.20150630.S: 7421.0\r\n",
      " E3.6.Acr.hyac.3.20150106.T: 7437.0\r\n",
      " E1.4.Pac.spec.1.20140801.S: 7482.0\r\n",
      " E10.Aca.hill.3.20150629.S: 7517.0\r\n",
      " E3.5.Gal.astr.2.20150127.S: 7623.0\r\n",
      " E10.Por.hero.2.20150628.S: 7626.0\r\n",
      " E7.5.Gal.fasc.1.20150622.S: 7667.0\r\n",
      " E1.Out.Mil.sp.1.20140731.S: 7697.0\r\n",
      " E1.17.Ech.mamm.1.20140815.T: 7698.0\r\n",
      " E10.Fav.abdi.1.20150629.S: 7702.0\r\n",
      " E1.17.Cau.furc.1.20140801.S: 7706.0\r\n",
      " E7.6.Acr.hyac.3.20150622.S: 7720.0\r\n",
      " E1.Out.Mil.sp.1.20140731.M: 7725.0\r\n",
      " E1.7.Pav.vari.1.20140805.T: 7751.0\r\n",
      " E3.15.Dip.heli.1.20150118.T: 7804.0\r\n",
      " E3.7.Pav.vari.2.20150124.S: 7866.0\r\n",
      " E7.10.Ser.cali.1.20150622.S: 7932.0\r\n",
      " E3.14.Phy.lich.1.20150129.S: 7970.0\r\n",
      " E1.19.Sym.sp.1.20140814.M: 7998.0\r\n",
      " E1.6.Acr.sp.1.20140726.S: 8071.0\r\n",
      " E3.4.Pac.rugo.1.20150128.S: 8078.0\r\n",
      " E3.7.Pav.vari.1.20150127.T: 8078.0\r\n",
      " E1.4.Pac.rugo.1.20140728.S: 8081.0\r\n",
      " E1.19.Lob.sp.1.20140730.S: 8089.0\r\n",
      " E3.3.Por.cyli.2.20150123.T: 8118.0\r\n",
      " E3.6.Acr.hyac.2.20150120.S: 8179.0\r\n",
      " E1.Out.Lob.sp.1.20140803.T: 8310.0\r\n",
      " E10.10.Poc.dami.1.20150630.S: 8322.0\r\n",
      " E1.10.Sty.pist.1.20140807.T: 8333.0\r\n",
      " E3.6.Acr.hyac.2.20150120.T: 8446.0\r\n",
      " E10.Sty.pist.1.20150628.S: 8462.0\r\n",
      " E3.15.Dip.heli.2.20150105.M: 8662.0\r\n",
      " E3.10.Sty.pist.2.20150124.M: 8700.0\r\n",
      " E3.5.Gal.astr.1.20150124.M: 8700.0\r\n",
      " E3.15.Dip.heli.3.20150123.T: 8711.0\r\n",
      " E10.Aca.hill.1.20150629.S: 8733.0\r\n",
      " E7.17.Fav.abdi.1.20150623.T: 8825.0\r\n",
      " E3.4.Pac.rugo.1.20150127.T: 8844.0\r\n",
      " E1.11.Lep.tran.1.20140731.M: 8850.0\r\n",
      " E1.15.Dip.heli.1.20140809.M: 8871.0\r\n",
      " E7.17.Fav.spC.1.20150623.S: 8946.0\r\n",
      " E1.6.Acr.humi.1.20140731.S: 8959.0\r\n",
      " E3.15.Dip.heli.2.20150106.T: 9032.0\r\n",
      " E3.17.Ech.mamm.1.20150129.S: 9038.0\r\n",
      " E1.10.Poc.dami.1.20140728.M: 9100.0\r\n",
      " E1.7.Pav.vari.1.20140815.M: 9121.0\r\n",
      " E1.15.Dip.heli.1.20140726.T: 9140.0\r\n",
      " E3.6.Acr.hyac.3.20150121.T: 9149.0\r\n",
      " E1.7.Pav.veno.1.20140809.M: 9250.0\r\n",
      " E3.17.Ech.mamm.1.20150123.T: 9287.0\r\n",
      " E1.15.Dip.heli.1.20140815.M: 9297.0\r\n",
      " E3.4.Pac.spec.1.20150128.T: 9325.0\r\n",
      " E10.Acr.pali.3.20150627.S: 9469.0\r\n",
      " E7.10.Ser.cali.3.20150622.S: 9472.0\r\n",
      " E3.6.Acr.hyac.4.20150105.M: 9494.0\r\n",
      " E3.14.Phy.lich.1.20150129.T: 9510.0\r\n",
      " E7.17.Fav.abdi.2.20150623.T: 9523.0\r\n",
      " E7.17.Hyd.exes.2.20150622.S: 9524.0\r\n",
      " E1.10.Sty.pist.1.20140815.M: 9623.0\r\n",
      " E3.2.Tur.reni.1.20150127.S: 9661.0\r\n",
      " E10.17.Fav.pali.1.20150630.S: 9706.0\r\n",
      " E3.17.Ech.horr.1.20150123.S: 9716.0\r\n",
      " E3.10.Sty.pist.1.20150124.M: 9732.0\r\n",
      " E7.17.Pla.sine.1.20150622.S: 9760.0\r\n",
      " E3.Out.Mil.sp.1.20150125.M: 9796.0\r\n",
      " E7.17.Ech.lame.3.20150620.S: 9799.0\r\n",
      " E3.15.Dip.heli.3.20150106.T: 9846.0\r\n",
      " E3.6.Acr.hyac.1.20150120.S: 9941.0\r\n",
      " E3.3.Por.cyli.1.20150127.S: 9966.0\r\n",
      " E1.2.Tur.reni.1.20140730.M: 9999.0\r\n",
      " E7.3.Por.loba.3.20150619.T: 10007.0\r\n",
      " E1.5.Gal.astr.1.20140807.M: 10133.0\r\n",
      " E7.17.Fav.spA.1.20150623.T: 10201.0\r\n",
      " E10.17.Gon.favu.1.20150629.S: 10212.0\r\n",
      " E10.10.Sty.pist.3.20150630.M: 10212.0\r\n",
      " E3.11.Fun.sp.2.20150125.T: 10291.0\r\n",
      " E1.17.Ech.mamm.1.20140815.M: 10295.0\r\n",
      " E3.15.Dip.heli.2.20150106.M: 10296.0\r\n",
      " E7.6.Acr.hyac.1.20150622.T: 10332.0\r\n",
      " E1.17.Fav.sp.1.20140805.M: 10355.0\r\n",
      " E1.11.Psa.cont.1.20140730.M: 10387.0\r\n",
      " E3.6.Acr.hyac.1.20150121.S: 10419.0\r\n",
      " E7.6.Mon.aequ.2.20150620.T: 10431.0\r\n",
      " E3.14.Phy.lich.2.20150125.S: 10481.0\r\n",
      " E3.6.Acr.hyac.3.20150120.S: 10502.0\r\n",
      " E10.Por.hero.3.20150628.S: 10512.0\r\n",
      " E10.Acr.pali.2.20150627.S: 10513.0\r\n",
      " E3.4.Pac.rugo.1.20150127.S: 10549.0\r\n",
      " E1.17.Ech.mamm.1.20140728.M: 10591.0\r\n",
      " E3.15.Dip.heli.2.20150119.T: 10595.0\r\n",
      " E3.6.Acr.hyac.3.20150121.M: 10706.0\r\n",
      " E1.7.Pav.sp.1.20140905.T: 10731.0\r\n",
      " E3.14.Phy.lich.1.20150129.M: 10734.0\r\n",
      " E3.7.Pav.vari.1.20150124.S: 10839.0\r\n",
      " E3.7.Pav.vari.1.20150127.S: 10953.0\r\n",
      " E3.4.Pac.spec.1.20150129.T: 10967.0\r\n",
      " E1.Out.Hel.coer.1.20140811.M: 10974.0\r\n",
      " E10.Ser.cali.3.20150627.M: 11005.0\r\n",
      " E1.5.Gal.astr.1.20140728.M: 11074.0\r\n",
      " E10.Sty.pist.1.20150628.M: 11224.0\r\n",
      " E1.2.Tur.stel.1.20140805.T: 11275.0\r\n",
      " E3.15.Dip.heli.4.20150106.T: 11328.0\r\n",
      " E1.11.Lep.tran.1.20140731.T: 11360.0\r\n",
      " E3.15.Dip.heli.1.20150118.M: 11439.0\r\n",
      " E1.3.Por.loba.1.20140726.T: 11460.0\r\n",
      " E1.6.Acr.humi.1.20140731.M: 11485.0\r\n",
      " E1.15.Dip.heli.1.20140726.S: 11540.0\r\n",
      " E7.Sed.1.20150621: 11605.0\r\n",
      " E7.17.Mer.ampl.1.20150621.T: 11605.0\r\n",
      " E1.4.Pac.rugo.1.20140728.T: 11640.0\r\n",
      " E1.17.Hyd.exes.1.20140803.M: 11669.0\r\n",
      " E1.2.Tur.reni.1.20140814.M: 11675.0\r\n",
      " E1.7.Pav.veno.1.20140809.T: 11682.0\r\n",
      " E3.6.Acr.hyac.1.20150106.S: 11700.0\r\n",
      " E3.Out.Pal.caes.1.20150125.S: 11757.0\r\n",
      " E1.17.Fav.sp.2.20140805.M: 11768.0\r\n",
      " E1.5.Gal.astr.1.20140807.S: 11801.0\r\n",
      " E10.Poc.dami.2.20150628.S: 11816.0\r\n",
      " E10.Por.hero.3.20150628.T: 11850.0\r\n",
      " E1.10.Poc.dami.1.20140731.M: 11857.0\r\n",
      " E3.17.Ech.mamm.1.20150123.S: 11859.0\r\n",
      " E7.19.Lob.hemp.2.20150619.T: 11937.0\r\n",
      " E1.11.Lep.tran.1.20140731.S: 11958.0\r\n",
      " E7.Sed.1.20150623: 12037.0\r\n",
      " E1.Out.Pal.caes.1.20140809.M: 12073.0\r\n",
      " E7.6.Acr.hyac.3.20150622.T: 12131.0\r\n",
      " E1.15.Dip.heli.1.20140905.M: 12132.0\r\n",
      " E7.17.Pla.sine.1.20150622.T: 12177.0\r\n",
      " E7.17.Myc.elep.1.20150622.S: 12216.0\r\n",
      " E1.6.Acr.sp.1.20140905.S: 12219.0\r\n",
      " E10.Alv.alli.2.20150628.S: 12252.0\r\n",
      " E1.11.Fun.sp.1.20140730.T: 12254.0\r\n",
      " E1.11.Fun.sp.1.20140730.M: 12261.0\r\n",
      " E10.17.Gon.favu.1.20150629.T: 12292.0\r\n",
      " E1.10.Poc.dami.1.20140731.T: 12309.0\r\n",
      " E1.19.Lob.sp.1.20140730.T: 12327.0\r\n",
      " E10.Acr.cyth.1.20150627.T: 12366.0\r\n",
      " E1.5.Gal.astr.1.20140728.S: 12471.0\r\n",
      " E1.Out.Lob.sp.1.20140803.M: 12475.0\r\n",
      " E1.17.Cau.furc.1.20140801.T: 12518.0\r\n",
      " E10.Sed.1.20150630: 12638.0\r\n",
      " E1.3.Por.loba.1.20140726.M: 12685.0\r\n",
      " E1.7.Pav.vari.1.20140815.S: 12686.0\r\n",
      " E3.6.Acr.hyac.1.20150118.S: 12699.0\r\n",
      " E10.17.Mon.curt.3.20150629.S: 12727.0\r\n",
      " E7.17.Fav.spA.1.20150623.M: 12747.0\r\n",
      " E1.2.Tur.pelt.1.20140814.T: 12762.0\r\n",
      " E7.5.Gal.fasc.1.20150622.T: 12834.0\r\n",
      " E3.Out.Pal.caes.1.20150125.T: 12848.0\r\n",
      " E3.15.Dip.heli.1.20150120.M: 12866.0\r\n",
      " E3.15.Dip.heli.1.20150106.T: 12870.0\r\n",
      " E7.17.Gon.edwa.1.20150623.S: 13000.0\r\n",
      " E7.17.Fav.spA.2.20150623.M: 13033.0\r\n",
      " E1.19.Sym.sp.1.20140905.M: 13096.0\r\n",
      " E3.6.Acr.hyac.1.20150123.S: 13103.0\r\n",
      " E7.6.Mon.aequ.4.20150620.T: 13151.0\r\n",
      " E1.17.Fav.sp.2.20140805.T: 13158.0\r\n",
      " E7.17.Pla.sine.2.20150622.T: 13281.0\r\n",
      " E10.17.Fav.abdi.1.20150630.T: 13351.0\r\n",
      " E1.17.Cau.furc.1.20140801.DS: 13371.0\r\n",
      " E10.Alv.alli.1.20150628.T: 13379.0\r\n",
      " E10.10.Sty.pist.2.20150630.M: 13456.0\r\n",
      " E3.3.Por.cyli.1.20150123.S: 13460.0\r\n",
      " E7.7.Pav.vari.2.20150622.S: 13468.0\r\n",
      " E3.5.Gal.astr.1.20150124.S: 13520.0\r\n",
      " E1.2.Tur.pelt.1.20140814.M: 13525.0\r\n",
      " E1.3.Por.loba.1.20140905.T: 13564.0\r\n",
      " E3.3.Por.cyli.1.20150123.T: 13625.0\r\n",
      " E1.10.Poc.dami.1.20140731.S: 13668.0\r\n",
      " E3.10.Sty.pist.1.20150127.M: 13701.0\r\n",
      " E1.19.Sym.sp.1.20140814.S: 13719.0\r\n",
      " E1.19.Lob.sp.1.20140730.M: 13733.0\r\n",
      " E3.Out.Pal.caes.2.20150125.T: 13758.0\r\n",
      " E10.Alv.alli.2.20150628.T: 13766.0\r\n",
      " E1.Out.Pal.caes.1.20140809.T: 13770.0\r\n",
      " E1.2.Tur.reni.1.20140730.T: 13771.0\r\n",
      " E3.3.Por.cyli.2.20150123.S: 13787.0\r\n",
      " E3.6.Acr.hyac.3.20150121.S: 13804.0\r\n",
      " E1.19.Sym.sp.1.20140814.T: 13818.0\r\n",
      " E1.2.Tur.reni.1.20140730.S: 13874.0\r\n",
      " E10.Alv.alli.3.20150628.S: 13879.0\r\n",
      " E1.3.Por.loba.1.20140905.M: 13894.0\r\n",
      " E3.11.Fun.sp.1.20150127.T: 13942.0\r\n",
      " E1.15.Dip.heli.1.20140815.S: 13966.0\r\n",
      " E7.17.Pla.sine.3.20150622.T: 14004.0\r\n",
      " E1.4.Pac.spec.1.20140811.T: 14016.0\r\n",
      " E1.Out.Pal.caes.1.20140809.S: 14058.0\r\n",
      " E1.4.Pac.spec.1.20140811.M: 14104.0\r\n",
      " E1.19.Sym.sp.1.20140905.S: 14166.0\r\n",
      " E1.3.Por.cyli.1.20140815.T: 14202.0\r\n",
      " E1.14.Phy.lich.1.20140801.S: 14225.0\r\n",
      " E7.3.Por.loba.4.20150619.S: 14240.0\r\n",
      " E7.19.Lob.hemp.2.20150619.S: 14261.0\r\n",
      " E7.Sed.1.20150622: 14269.0\r\n",
      " E7.17.Cyp.micr.2.20150621.S: 14282.0\r\n",
      " E10.Alv.alli.3.20150628.M: 14382.0\r\n",
      " E3.14.Phy.lich.2.20150125.M: 14440.0\r\n",
      " E1.14.Phy.lich.1.20140811.M: 14481.0\r\n",
      " E1.10.Sty.pist.1.20140807.M: 14553.0\r\n",
      " E1.17.Fav.sp.1.20140805.T: 14566.0\r\n",
      " E3.6.Acr.hyac.3.20150123.S: 14573.0\r\n",
      " E3.4.Pac.rugo.1.20150128.T: 14621.0\r\n",
      " E7.6.Acr.hyac.2.20150622.S: 14626.0\r\n",
      " E3.15.Dip.heli.4.20150105.T: 14639.0\r\n",
      " E7.19.Lob.hemp.1.20150619.T: 14654.0\r\n",
      " E7.17.Cyp.micr.3.20150621.S: 14686.0\r\n",
      " E7.17.Ech.lame.3.20150620.T: 14704.0\r\n",
      " E7.7.Pav.vari.1.20150623.M: 14704.0\r\n",
      " E7.17.Fav.spB.1.20150623.S: 14721.0\r\n",
      " E3.19.Sym.sp.1.20150125.S: 14724.0\r\n",
      " E7.Sed.1.20150620: 14765.0\r\n",
      " E1.5.Gal.astr.1.20140728.T: 14799.0\r\n",
      " E3.6.Acr.hyac.1.20150121.T: 14804.0\r\n",
      " E1.3.Por.loba.1.20140726.S: 14856.0\r\n",
      " E1.7.Pav.vari.1.20140805.S: 14858.0\r\n",
      " E10.Por.hero.1.20150628.T: 14863.0\r\n",
      " E3.15.Dip.heli.2.20150119.M: 14877.0\r\n",
      " E7.3.Por.loba.1.20150619.T: 14958.0\r\n",
      " E3.6.Acr.hyac.3.20150120.T: 14973.0\r\n",
      " E7.3.Por.loba.4.20150619.T: 14995.0\r\n",
      " E7.Sed.1.20150619: 15134.0\r\n",
      " E10.Aca.hill.3.20150629.M: 15140.0\r\n",
      " E3.6.Acr.hyac.2.20150105.T: 15221.0\r\n",
      " E10.17.Gon.favu.2.20150630.S: 15284.0\r\n",
      " E10.17.Gon.favu.1.20150630.M: 15322.0\r\n",
      " E10.10.Sty.pist.1.20150630.M: 15352.0\r\n",
      " E1.4.Pac.rugo.1.20140814.M: 15371.0\r\n",
      " E7.7.Pav.vari.1.20150623.T: 15420.0\r\n",
      " E1.7.Pav.veno.1.20140811.T: 15431.0\r\n",
      " E7.17.Hyd.exce.1.20150620.S: 15457.0\r\n",
      " E3.6.Acr.hyac.1.20150121.M: 15535.0\r\n",
      " E3.Out.Pal.caes.2.20150125.S: 15618.0\r\n",
      " E3.4.Pac.spec.1.20150128.M: 15656.0\r\n",
      " E3.15.Dip.heli.2.20150120.S: 15660.0\r\n",
      " E7.10.Poc.dami.1.20150621.M: 15705.0\r\n",
      " E3.6.Acr.hyac.2.20150121.S: 15725.0\r\n",
      " E10.17.Mon.curt.3.20150629.M: 15728.0\r\n",
      " E10.Acr.pali.2.20150627.T: 15797.0\r\n",
      " E7.6.Mon.aequ.1.20150620.S: 15819.0\r\n",
      " E3.15.Dip.heli.2.20150105.T: 15879.0\r\n",
      " E1.6.Mon.hisp.1.20140807.S: 15883.0\r\n",
      " E1.19.Sym.sp.1.20140815.T: 15883.0\r\n",
      " E7.6.Mon.aequ.4.20150620.S: 15948.0\r\n",
      " E10.Por.hero.3.20150628.M: 15976.0\r\n",
      " E7.7.Pav.vari.2.20150622.M: 15983.0\r\n",
      " E3.6.Acr.hyac.3.20150123.T: 15998.0\r\n",
      " E10.Cyp.sera.2.20150628.M: 16001.0\r\n",
      " E7.17.Myc.elep.1.20150622.T: 16042.0\r\n",
      " E7.6.Acr.hyac.1.20150622.S: 16048.0\r\n",
      " E1.7.Pav.sp.1.20140905.M: 16110.0\r\n",
      " E10.Acr.cyth.3.20150627.T: 16112.0\r\n",
      " E7.5.Gal.fasc.1.20150621.S: 16132.0\r\n",
      " E3.11.Fun.sp.1.20150125.T: 16167.0\r\n",
      " E1.2.Tur.pelt.1.20140814.S: 16191.0\r\n",
      " E3.5.Gal.astr.2.20150124.T: 16244.0\r\n",
      " E1.7.Pav.vari.1.20140815.T: 16254.0\r\n",
      " E7.17.Mer.ampl.2.20150621.T: 16316.0\r\n",
      " E1.19.Sym.sp.1.20140905.T: 16446.0\r\n",
      " E1.6.Iso.pali.1.20140803.M: 16469.0\r\n",
      " E1.17.Hyd.exes.1.20140803.S: 16484.0\r\n",
      " E10.Por.hero.2.20150628.T: 16494.0\r\n",
      " E3.6.Acr.hyac.4.20150105.S: 16508.0\r\n",
      " E1.17.Fav.sp.1.20140805.S: 16542.0\r\n",
      " E3.6.Acr.hyac.2.20150106.M: 16605.0\r\n",
      " E1.3.Por.cyli.1.20140801.S: 16629.0\r\n",
      " E7.17.Ech.lame.2.20150620.T: 16650.0\r\n",
      " E10.Fav.abdi.1.20150629.T: 16677.0\r\n",
      " E3.15.Dip.heli.2.20150123.M: 16697.0\r\n",
      " E3.15.Dip.heli.2.20150119.S: 16707.0\r\n",
      " E3.3.Por.cyli.1.20150123.M: 16743.0\r\n",
      " E10.Sty.pist.1.20150628.T: 16760.0\r\n",
      " E10.10.Poc.dami.1.20150630.T: 16791.0\r\n",
      " E1.7.Pav.veno.1.20140811.S: 16884.0\r\n",
      " E3.17.Ech.mamm.1.20150129.M: 16913.0\r\n",
      " E1.11.Psa.cont.1.20140730.S: 16932.0\r\n",
      " E1.17.Ech.lame.1.20140905.T: 16973.0\r\n",
      " E7.17.Fav.abdi.1.20150623.M: 16982.0\r\n",
      " E1.5.Gal.astr.1.20140814.S: 16996.0\r\n",
      " E3.15.Dip.heli.2.20150121.M: 17017.0\r\n",
      " E7.19.Lob.hemp.1.20150621.T: 17030.0\r\n",
      " E10.17.Gon.favu.1.20150630.T: 17038.0\r\n",
      " E3.15.Dip.heli.3.20150105.T: 17105.0\r\n",
      " E10.Water.1.20150629: 17107.0\r\n",
      " E10.Mon.curt.1.20150629.S: 17126.0\r\n",
      " E3.4.Pac.spec.1.20150129.S: 17135.0\r\n",
      " E7.3.Por.loba.3.20150619.M: 17158.0\r\n",
      " E7.17.Pla.sine.2.20150622.S: 17182.0\r\n",
      " E1.17.Cau.furc.1.20140801.M: 17185.0\r\n",
      " E1.17.Fav.sp.1.20140811.S: 17261.0\r\n",
      " E3.6.Acr.hyac.2.20150105.S: 17279.0\r\n",
      " E3.3.Por.cyli.1.20150127.T: 17409.0\r\n",
      " E10.17.Gon.favu.2.20150630.T: 17422.0\r\n",
      " E7.7.Pav.vari.1.20150622.M: 17476.0\r\n",
      " E3.7.Pav.vari.1.20150124.M: 17487.0\r\n",
      " E1.2.Tur.reni.1.20140814.T: 17488.0\r\n",
      " E7.6.Mon.aequ.1.20150620.T: 17498.0\r\n",
      " E7.19.Lob.hemp.1.20150619.S: 17500.0\r\n",
      " E3.6.Acr.hyac.2.20150121.M: 17505.0\r\n",
      " E1.11.Psa.cont.1.20140730.T: 17506.0\r\n",
      " E10.Poc.dami.1.20150628.M: 17559.0\r\n",
      " E1.5.Gal.astr.1.20140814.T: 17574.0\r\n",
      " E3.6.Acr.hyac.4.20150105.T: 17590.0\r\n",
      " E10.Acr.pali.2.20150627.M: 17591.0\r\n",
      " E1.6.Acr.sp.1.20140905.M: 17681.0\r\n",
      " E7.6.Acr.hyac.3.20150622.M: 17687.0\r\n",
      " E1.3.Por.cyli.1.20140815.M: 17691.0\r\n",
      " E10.Acr.pali.3.20150627.T: 17708.0\r\n",
      " E10.Water.2.20150628: 17728.0\r\n",
      " E3.4.Pac.rugo.1.20150128.M: 17770.0\r\n",
      " E3.6.Acr.hyac.3.20150105.T: 17793.0\r\n",
      " E3.15.Dip.heli.2.20150105.S: 17815.0\r\n",
      " E10.Aca.hill.1.20150629.M: 17816.0\r\n",
      " E3.4.Pac.spec.1.20150129.M: 17829.0\r\n",
      " E1.6.Iso.pali.1.20140803.T: 17851.0\r\n",
      " E3.3.Por.cyli.1.20150127.M: 17871.0\r\n",
      " E10.Sed.1.20150628: 17898.0\r\n",
      " E7.3.Por.loba.2.20150619.T: 17986.0\r\n",
      " E7.17.Ech.lame.1.20150620.T: 18042.0\r\n",
      " E3.15.Dip.heli.3.20150105.S: 18042.0\r\n",
      " E10.Por.hero.2.20150628.M: 18085.0\r\n",
      " E7.17.Myc.elep.1.20150622.M: 18110.0\r\n",
      " E1.2.Tur.stel.1.20140805.S: 18125.0\r\n",
      " E3.15.Dip.heli.1.20150121.S: 18126.0\r\n",
      " E3.11.Fun.sp.2.20150125.M: 18141.0\r\n",
      " E3.15.Dip.heli.1.20150123.M: 18160.0\r\n",
      " E3.15.Dip.heli.3.20150106.S: 18171.0\r\n",
      " E3.4.Pac.rugo.1.20150127.M: 18194.0\r\n",
      " E1.7.Pav.veno.1.20140809.S: 18197.0\r\n",
      " E3.2.Tur.reni.2.20150124.M: 18213.0\r\n",
      " E1.17.Fav.sp.1.20140811.M: 18228.0\r\n",
      " E1.6.Iso.pali.1.20140815.S: 18233.0\r\n",
      " E7.17.Gon.edwa.1.20150623.T: 18255.0\r\n",
      " E1.15.Dip.heli.1.20140815.T: 18283.0\r\n",
      " E1.19.Sym.sp.1.20140815.M: 18286.0\r\n",
      " E3.15.Dip.heli.3.20150120.M: 18290.0\r\n",
      " E1.17.Hyd.exes.1.20140803.T: 18291.0\r\n",
      " E3.19.Sym.sp.1.20150127.M: 18350.0\r\n",
      " E7.17.Hyd.exes.1.20150622.T: 18385.0\r\n",
      " E1.19.Sym.sp.1.20140724.S: 18388.0\r\n",
      " E1.3.Por.cyli.1.20140801.M: 18430.0\r\n",
      " E1.6.Iso.pali.1.20140815.T: 18495.0\r\n",
      " E7.17.Gon.edwa.1.20150623.M: 18509.0\r\n",
      " E1.Out.Mil.sp.1.20140814.M: 18582.0\r\n",
      " E10.Alv.alli.3.20150628.T: 18588.0\r\n",
      " E7.17.Pla.sine.2.20150622.M: 18614.0\r\n",
      " E1.6.Iso.pali.1.20140803.S.2: 18622.0\r\n",
      " E3.15.Dip.heli.1.20150119.M: 18647.0\r\n",
      " E3.5.Gal.astr.1.20150124.T: 18651.0\r\n",
      " E1.15.Dip.heli.1.20140905.T: 18664.0\r\n",
      " E3.2.Tur.reni.1.20150124.M: 18704.0\r\n",
      " E3.7.Pav.vari.1.20150127.M: 18711.0\r\n",
      " E10.Mon.curt.1.20150629.T: 18712.0\r\n",
      " E3.6.Acr.hyac.1.20150123.T: 18730.0\r\n",
      " E7.19.Lob.hemp.2.20150619.M: 18732.0\r\n",
      " E1.5.Gal.astr.1.20140905.T: 18738.0\r\n",
      " E1.17.Ech.lame.1.20140905.S: 18768.0\r\n",
      " E1.4.Pac.spec.1.20140811.S: 18807.0\r\n",
      " E3.15.Dip.heli.3.20150121.M: 18820.0\r\n",
      " E7.10.Poc.dami.2.20150621.M: 18844.0\r\n",
      " E1.3.Por.cyli.1.20140801.T: 18858.0\r\n",
      " E3.6.Acr.hyac.3.20150105.M: 18872.0\r\n",
      " E7.17.Fav.abdi.1.20150623.S: 18892.0\r\n",
      " E7.6.Mon.aequ.3.20150620.T: 18900.0\r\n",
      " E3.14.Phy.lich.1.20150125.M: 19007.0\r\n",
      " E1.19.Lob.sp.1.20140809.S: 19114.0\r\n",
      " E3.6.Acr.hyac.2.20150123.T: 19142.0\r\n",
      " E3.6.Acr.hyac.2.20150106.S: 19219.0\r\n",
      " E3.6.Acr.hyac.1.20150106.T: 19265.0\r\n",
      " E7.19.Lob.hemp.1.20150621.S: 19320.0\r\n",
      " E7.17.Pla.sine.3.20150622.S: 19321.0\r\n",
      " E7.5.Gal.fasc.1.20150622.M: 19420.0\r\n",
      " E1.17.Fav.sp.1.20140811.T: 19436.0\r\n",
      " E1.5.Gal.astr.1.20140905.S: 19441.0\r\n",
      " E1.17.Fav.sp.2.20140805.S: 19459.0\r\n",
      " E3.6.Acr.hyac.2.20150123.S: 19493.0\r\n",
      " E7.17.Mer.ampl.3.20150621.T: 19502.0\r\n",
      " E3.15.Dip.heli.4.20150105.S: 19552.0\r\n",
      " E10.Mon.curt.1.20150629.M: 19590.0\r\n",
      " E10.10.Poc.dami.1.20150630.M: 19604.0\r\n",
      " E1.15.Dip.heli.1.20140809.S: 19656.0\r\n",
      " E7.17.Hyd.exes.2.20150622.M: 19656.0\r\n",
      " E3.15.Dip.heli.1.20150105.S: 19728.0\r\n",
      " E7.17.Hyd.exce.1.20150620.M: 19779.0\r\n",
      " E7.17.Hyd.exce.1.20150620.T: 19804.0\r\n",
      " E7.17.Ech.lame.2.20150620.M: 19849.0\r\n",
      " E7.17.Cyp.micr.3.20150621.M: 19867.0\r\n",
      " E1.3.Por.cyli.1.20140815.S: 19883.0\r\n",
      " E1.Out.Mil.sp.1.20140814.S: 19950.0\r\n",
      " E1.19.Lob.sp.1.20140809.M: 20084.0\r\n",
      " E3.15.Dip.heli.1.20150123.S: 20096.0\r\n",
      " E3.2.Tur.reni.1.20150127.M: 20125.0\r\n",
      " E1.6.Acr.sp.1.20140905.T: 20168.0\r\n",
      " E1.Out.Hel.coer.1.20140811.T: 20222.0\r\n",
      " E7.17.Fav.spA.2.20150623.S: 20268.0\r\n",
      " E10.17.Mon.curt.3.20150629.T: 20356.0\r\n",
      " E10.17.Gon.favu.1.20150630.S: 20382.0\r\n",
      " E3.6.Acr.hyac.1.20150105.S: 20410.0\r\n",
      " E3.19.Sym.sp.1.20150125.M: 20425.0\r\n",
      " E7.19.Lob.hemp.1.20150619.M: 20461.0\r\n",
      " E1.7.Pav.sp.1.20140905.S: 20517.0\r\n",
      " E7.3.Por.loba.2.20150619.M: 20540.0\r\n",
      " E1.6.Iso.pali.1.20140803.S: 20546.0\r\n",
      " E3.Out.Pal.caes.1.20150125.M: 20587.0\r\n",
      " E1.5.Gal.astr.1.20140814.M: 20640.0\r\n",
      " E7.5.Gal.astr.1.20150622.S: 20676.0\r\n",
      " E1.14.Phy.lich.1.20140811.T: 20731.0\r\n",
      " E1.17.Ech.lame.1.20140905.M: 20731.0\r\n",
      " E10.Cyp.micr.1.20150627.T: 20749.0\r\n",
      " E7.6.Mon.aequ.3.20150620.S: 20798.0\r\n",
      " E7.17.Ech.lame.3.20150620.M: 20835.0\r\n",
      " E1.11.Psa.cont.1.20140730.TS: 20837.0\r\n",
      " E1.Out.Pal.caes.1.20140815.T: 20853.0\r\n",
      " E7.5.Gal.fasc.1.20150621.M: 20962.0\r\n",
      " E3.15.Dip.heli.3.20150120.S: 21088.0\r\n",
      " E10.17.Gon.favu.2.20150630.M: 21147.0\r\n",
      " E10.Acr.cyth.3.20150627.S: 21149.0\r\n",
      " E1.6.Iso.pali.1.20140815.M: 21159.0\r\n",
      " E3.15.Dip.heli.4.20150105.M: 21188.0\r\n",
      " E7.17.Ech.lame.2.20150620.S: 21203.0\r\n",
      " E1.11.Psa.digi.1.20140807.M: 21342.0\r\n",
      " E3.6.Acr.hyac.1.20150123.M: 21371.0\r\n",
      " E3.7.Pav.vari.2.20150124.M: 21392.0\r\n",
      " E3.5.Gal.astr.2.20150127.M: 21434.0\r\n",
      " E3.15.Dip.heli.4.20150106.S: 21561.0\r\n",
      " E1.15.Dip.heli.1.20140905.S: 21633.0\r\n",
      " E7.17.Fav.spA.1.20150623.S: 21653.0\r\n",
      " E1.10.Sty.pist.1.20140815.T: 21705.0\r\n",
      " E1.4.Pac.rugo.1.20140814.T: 21754.0\r\n",
      " E10.Mon.curt.2.20150629.T: 21799.0\r\n",
      " E3.5.Gal.astr.2.20150124.M: 21831.0\r\n",
      " E7.17.Cyp.micr.1.20150621.S: 21929.0\r\n",
      " E1.19.Lob.sp.1.20140809.T: 21933.0\r\n",
      " E1.Out.Mil.sp.1.20140731.T: 21952.0\r\n",
      " E3.19.Sym.sp.2.20150125.S: 21981.0\r\n",
      " E1.6.Iso.pali.1.20140803.M.2: 21994.0\r\n",
      " E3.15.Dip.heli.1.20150106.S: 22035.0\r\n",
      " E3.15.Dip.heli.3.20150123.S: 22173.0\r\n",
      " E7.6.Mon.aequ.1.20150620.M: 22187.0\r\n",
      " E1.19.Sym.sp.1.20140815.S: 22229.0\r\n",
      " E7.17.Pla.sine.3.20150622.M: 22250.0\r\n",
      " E7.17.Hyd.exes.2.20150622.T: 22399.0\r\n",
      " E3.5.Gal.astr.2.20150124.S: 22464.0\r\n",
      " E7.5.Gal.astr.1.20150622.T: 22597.0\r\n",
      " E7.6.Mon.aequ.3.20150620.M: 22610.0\r\n",
      " E3.6.Acr.hyac.2.20150120.M: 22656.0\r\n",
      " E7.17.Fav.abdi.2.20150623.M: 22667.0\r\n",
      " E3.15.Dip.heli.1.20150105.M: 22674.0\r\n",
      " E7.10.Poc.dami.3.20150621.M: 22687.0\r\n",
      " E10.Acr.cyth.1.20150627.S: 22724.0\r\n",
      " E3.15.Dip.heli.2.20150123.S: 22874.0\r\n",
      " E7.10.Ser.cali.3.20150622.M: 22919.0\r\n",
      " E3.15.Dip.heli.3.20150105.M: 22931.0\r\n",
      " E1.15.Dip.heli.1.20140809.T: 22945.0\r\n",
      " E10.Mon.curt.2.20150629.M: 23048.0\r\n",
      " E10.Acr.pali.1.20150627.M: 23067.0\r\n",
      " E3.3.Por.cyli.2.20150123.M: 23076.0\r\n",
      " E10.Ser.cali.1.20150627.M: 23096.0\r\n",
      " E10.Por.hero.1.20150628.M: 23099.0\r\n",
      " E3.17.Ech.mamm.1.20150127.M: 23107.0\r\n",
      " E1.Out.Pal.caes.1.20140815.S: 23128.0\r\n",
      " E7.17.Cyp.micr.2.20150621.M: 23141.0\r\n",
      " E3.6.Acr.hyac.3.20150105.S: 23211.0\r\n",
      " E10.17.Fav.abdi.2.20150630.M: 23282.0\r\n",
      " E1.6.Iso.pali.1.20140803.T.2: 23341.0\r\n",
      " E3.15.Dip.heli.2.20150120.M: 23362.0\r\n",
      " E3.6.Acr.hyac.2.20150106.T: 23388.0\r\n",
      " E3.15.Dip.heli.2.20150106.S: 23400.0\r\n",
      " E7.17.Gon.edwa.2.20150623.M: 23443.0\r\n",
      " E3.6.Acr.hyac.1.20150120.M: 23484.0\r\n",
      " E7.17.Cyp.micr.1.20150621.M: 23530.0\r\n",
      " E10.Acr.cyth.3.20150627.M: 23539.0\r\n",
      " E1.11.Psa.digi.1.20140807.S: 23543.0\r\n",
      " E7.6.Acr.hyac.2.20150622.T: 23583.0\r\n",
      " E1.Out.Hel.coer.1.20140811.S: 23588.0\r\n",
      " E3.15.Dip.heli.3.20150123.M: 23655.0\r\n",
      " E1.5.Gal.astr.1.20140905.M: 23776.0\r\n",
      " E1.Out.Pal.caes.1.20140815.M: 23859.0\r\n",
      " E7.17.Hyd.exes.1.20150622.M: 24010.0\r\n",
      " E10.Fav.abdi.1.20150629.M: 24124.0\r\n",
      " E7.10.Ser.cali.2.20150622.M: 24222.0\r\n",
      " E10.Acr.cyth.2.20150627.S: 24267.0\r\n",
      " E7.5.Gal.fasc.1.20150621.T: 24385.0\r\n",
      " E7.3.Por.loba.4.20150619.M: 24448.0\r\n",
      " E10.Cyp.micr.1.20150627.M: 24528.0\r\n",
      " E3.6.Acr.hyac.1.20150105.M: 24644.0\r\n",
      " E10.17.Gon.favu.1.20150629.M: 24699.0\r\n",
      " E1.Out.Mil.sp.1.20140814.T: 24730.0\r\n",
      " E10.Acr.pali.3.20150627.M: 24768.0\r\n",
      " E1.14.Phy.lich.1.20140811.S: 24846.0\r\n",
      " E7.17.Fav.spB.1.20150623.M: 25081.0\r\n",
      " E3.15.Dip.heli.1.20150120.S: 25191.0\r\n",
      " E3.6.Acr.hyac.2.20150123.M: 25208.0\r\n",
      " E1.11.Psa.digi.1.20140807.T: 25567.0\r\n",
      " E3.6.Acr.hyac.3.20150123.M: 25664.0\r\n",
      " E10.Sed.1.20150627: 25668.0\r\n",
      " E7.17.Mer.ampl.3.20150621.M: 25690.0\r\n",
      " E7.3.Por.loba.1.20150619.M: 25879.0\r\n",
      " E3.6.Acr.hyac.1.20150106.M: 25937.0\r\n",
      " E3.Out.Pal.caes.2.20150125.M: 26230.0\r\n",
      " E10.Acr.cyth.2.20150627.T: 26327.0\r\n",
      " E10.17.Fav.abdi.1.20150630.M: 26462.0\r\n",
      " E1.4.Pac.rugo.1.20140814.S: 26511.0\r\n",
      " E7.17.Ech.lame.1.20150620.M: 26583.0\r\n",
      " E7.6.Mon.aequ.4.20150620.M: 26690.0\r\n",
      " E7.6.Mon.aequ.2.20150620.M: 26877.0\r\n",
      " E7.6.Acr.hyac.2.20150622.M: 26934.0\r\n",
      " E7.5.Gal.astr.1.20150622.M: 27075.0\r\n",
      " E7.17.Mer.ampl.1.20150621.M: 27128.0\r\n",
      " E10.Acr.cyth.2.20150627.M: 27180.0\r\n",
      " E3.15.Dip.heli.2.20150121.S: 27242.0\r\n",
      " E3.15.Dip.heli.3.20150121.S: 27387.0\r\n",
      " E1.3.Por.loba.1.20140905.S: 27482.0\r\n",
      " E7.6.Acr.hyac.1.20150622.M: 27595.0\r\n",
      " E1.11.Fun.sp.1.20140811.T: 27603.0\r\n",
      " E3.Out.Mil.sp.1.20150125.T: 27894.0\r\n",
      " E7.10.Ser.cali.1.20150622.M: 28052.0\r\n",
      " E10.Aca.hill.2.20150629.M: 28263.0\r\n",
      " E3.15.Dip.heli.1.20150121.M: 28304.0\r\n",
      " E10.Alv.alli.1.20150628.M: 28318.0\r\n",
      " E3.19.Sym.sp.2.20150125.M: 28347.0\r\n",
      " E3.6.Acr.hyac.2.20150105.M: 28794.0\r\n",
      " E3.Out.Mil.sp.2.20150125.T: 28870.0\r\n",
      " E3.17.Ech.mamm.1.20150123.M: 28896.0\r\n",
      " E7.19.Lob.hemp.1.20150621.M: 29541.0\r\n",
      " E3.11.Fun.sp.1.20150125.M: 30168.0\r\n",
      " E10.Alv.alli.2.20150628.M: 31070.0\r\n",
      " E3.Out.Mil.sp.2.20150125.S: 31091.0\r\n",
      " E1.11.Fun.sp.1.20140811.M: 32780.0\r\n",
      " E3.Out.Mil.sp.1.20150125.S: 33141.0\r\n",
      " E1.11.Fun.sp.1.20140811.S: 35581.0\r\n",
      " E10.Acr.cyth.1.20150627.M: 38523.0\r\n"
     ]
    }
   ],
   "source": [
    "# Summarize BIOM\n",
    "!biom summarize-table -i $otu_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Subsampling to 1000 loses 31 samples, but this seems reasonable.\n",
    "depth=1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.203636422742 and the largest is 17.4947292856.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -1.26986829308 and the largest is 26.0525845582.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on all samples\n",
    "bdiv_dir_all = join(output_dir,'bdiv_all')\n",
    "!beta_diversity_through_plots.py -i $otu_table -o $bdiv_dir_all -m $mapping -p $bc_prefs_fp --force -t $tree_fp -e $depth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Create OTU table with only Scleractinian corals\n",
    "#Export Sceleractinian coral compartment specific biom file\n",
    "subset_output = join(output_dir,'compartment_tables')\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_scleractinian.biom -m $mapping -s 'taxonomy_string_to_order:Cnidaria_Anthozoa_Scleractinia'\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_scleractinian.biom -o $subset_output/otu_table_subset_scleractinian_summary.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.218644592699 and the largest is 16.6966060546.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -1.18331362097 and the largest is 24.4731203701.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on only Scleractinian samples\n",
    "subset_output = join(output_dir,'compartment_tables')\n",
    "!beta_diversity_through_plots.py -i $subset_output/otu_table_subset_scleractinian.biom -o $subset_output/Scleractinian_only -m $mapping -p $bc_prefs_fp --force -t $tree_fp -e $depth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "curr_cat: BiologicalMatter\n",
      "curr_metric: bray_curtis\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   3    10.706  3.5687  7.8026 0.0369 0.000999\n",
      "Residuals                       611   279.454  0.4574         0.9631         \n",
      "Total                           614   290.160                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: weighted_unifrac\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   3    10.029  3.3431  13.498 0.06216 0.000999\n",
      "Residuals                       611   151.325  0.2477         0.93784         \n",
      "Total                           614   161.355                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: unweighted_unifrac\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   3     6.844 2.28138  6.0842 0.02901 0.000999\n",
      "Residuals                       611   229.105 0.37497         0.97099         \n",
      "Total                           614   235.949                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n"
     ]
    }
   ],
   "source": [
    "# Test for significant differences in community composition among compartments\n",
    "scleractinian_subset_output = join(output_dir,'compartment_tables','Scleractinian_only')\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'BiologicalMatter'\n",
    "for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "    curr_dm = join(scleractinian_subset_output,\"%s_dm.txt\" %metric)   \n",
    "    for curr_cat in categories.split(\",\"):\n",
    "        print \"curr_cat: %s\" %curr_cat\n",
    "        print \"curr_metric: %s\" %metric\n",
    "        curr_compare_cat_output_dir = join(scleractinian_subset_output,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "        !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "        the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "        !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries: compartments\n",
    "\n",
    "scleractinian_subset_output = join(output_dir,'compartment_tables','Scleractinian_only')\n",
    "n_permutations = 1000\n",
    "categories = 'BiologicalMatter'\n",
    "for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "    curr_dm = join(scleractinian_subset_output,\"%s_dm.txt\" %metric)\n",
    "    curr_boxplot_output_dir = join(scleractinian_subset_output,\"distance_boxplots_all_within_%s\" %metric)\n",
    "    !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "    curr_boxplot_output_dir = join(scleractinian_subset_output,\"distance_boxplots_%s\" %metric)\n",
    "    !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries: compartments\n",
    "\n",
    "scleractinian_subset_output = join(output_dir,'compartment_tables','Scleractinian_only')\n",
    "n_permutations = 1000\n",
    "categories = 'BiologicalMatter'\n",
    "for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "    curr_dm = join(scleractinian_subset_output,\"%s_dm.txt\" %metric)\n",
    "    curr_boxplot_output_dir = join(scleractinian_subset_output,\"distance_boxplots_all_comparisons_%s\" %metric)\n",
    "    !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20  -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create OTU table with only Scleractinian corals (NOTE: NOW WE EXCLUDE Whole_Coral sample)\n",
    "#Export Sceleractinian coral compartment specific biom file (NOTE: NOW WE EXCLUDE Whole_Coral sample)\n",
    "subset_output = join(output_dir,'compartment_tables')\n",
    "!filter_samples_from_otu_table.py -i $subset_output/otu_table_subset_scleractinian.biom -o $subset_output/otu_table_subset_scleractinian_no_whole_coral.biom -m $mapping -s 'BiologicalMatter:*,!Whole_Coral'\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_scleractinian_no_whole_coral.biom -o $subset_output/otu_table_subset_scleractinian_no_whole_coral_summary.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.19072619308 and the largest is 16.6467128078.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -1.1786781633 and the largest is 24.4562972479.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on rarified (1000) subset of Scleractinian corals (NOTE: NOW WE EXCLUDE Whole_Coral sample)\n",
    "tissue_otu_table_dir = join(output_dir,\"compartment_tables\")\n",
    "depth = 1000\n",
    "\n",
    "!beta_diversity_through_plots.py -i $subset_output/otu_table_subset_scleractinian_no_whole_coral.biom -o $subset_output/Scleractinian_only_no_whole_coral -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "curr_cat: BiologicalMatter\n",
      "curr_metric: bray_curtis\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2    10.159  5.0794  11.104 0.03507 0.000999\n",
      "Residuals                       611   279.506  0.4575         0.96493         \n",
      "Total                           613   289.665                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: weighted_unifrac\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     9.723  4.8617  19.656 0.06045 0.000999\n",
      "Residuals                       611   151.127  0.2473         0.93955         \n",
      "Total                           613   160.851                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: unweighted_unifrac\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     6.675  3.3374  8.8889 0.02827 0.000999\n",
      "Residuals                       611   229.405  0.3755         0.97173         \n",
      "Total                           613   236.080                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n"
     ]
    }
   ],
   "source": [
    "# Test for significant differences in community composition among compartments\n",
    "scleractinian_subset_output = join(output_dir,'compartment_tables','Scleractinian_only_no_whole_coral')\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'BiologicalMatter'\n",
    "for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "    curr_dm = join(scleractinian_subset_output,\"%s_dm.txt\" %metric)   \n",
    "    for curr_cat in categories.split(\",\"):\n",
    "        print \"curr_cat: %s\" %curr_cat\n",
    "        print \"curr_metric: %s\" %metric\n",
    "        curr_compare_cat_output_dir = join(scleractinian_subset_output,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "        !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "        the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "        !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries: compartments\n",
    "\n",
    "scleractinian_subset_output = join(output_dir,'compartment_tables','Scleractinian_only_no_whole_coral')\n",
    "n_permutations = 1000\n",
    "categories = 'BiologicalMatter'\n",
    "for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "    curr_dm = join(scleractinian_subset_output,\"%s_dm.txt\" %metric)\n",
    "    curr_boxplot_output_dir = join(scleractinian_subset_output,\"distance_boxplots_all_within_%s\" %metric)\n",
    "    !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "    curr_boxplot_output_dir = join(scleractinian_subset_output,\"distance_boxplots_%s\" %metric)\n",
    "    !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Export Sceleractinian coral compartment specific biom files (i.e. coral t, m and s)\n",
    "subset_output = join(output_dir,'compartment_tables')\n",
    "!filter_samples_from_otu_table.py -i $subset_output/otu_table_subset_scleractinian.biom -o $subset_output/otu_table_subset_mucus.biom -m $mapping -s 'sample_type:Coral_Mucus'\n",
    "!filter_samples_from_otu_table.py -i $subset_output/otu_table_subset_scleractinian.biom -o $subset_output/otu_table_subset_skeleton.biom -m $mapping -s 'sample_type:Coral_Skeleton'\n",
    "!filter_samples_from_otu_table.py -i $subset_output/otu_table_subset_scleractinian.biom -o $subset_output/otu_table_subset_tissue.biom -m $mapping -s 'sample_type:Coral_Tissue'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Summarize BIOM to confirm filter functioned correctly\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_mucus.biom -o $subset_output/otu_table_subset_mucus_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_tissue.biom -o $subset_output/otu_table_subset_tissue_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_skeleton.biom -o $subset_output/otu_table_subset_skeleton_summary.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Summaries look good."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.100459226024 and the largest is 9.60549079436.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.23079800206 and the largest is 9.67522628678.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.201274352463 and the largest is 16.8005403628.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -1.20362918709 and the largest is 24.4263721713.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.113652116153 and the largest is 5.82405629261.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.191948176139 and the largest is 11.0031160197.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0858951706045 and the largest is 7.89900539945.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.471949705422 and the largest is 9.81978502459.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on rarified (1000) subset of each sample type\n",
    "tissue_otu_table_dir = join(output_dir,\"compartment_tables\")\n",
    "depth = 1000\n",
    "\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith(\".biom\"):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    !beta_diversity_through_plots.py -i $curr_otu_table -o $curr_bdiv_dir -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'log_20161031170530.txt', 'otu_table_subset_mucus_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     8.813 0.73442   1.777 0.09903 0.000999\n",
      "Residuals                       194    80.180 0.41330         0.90097         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    17.292 0.86461  2.2429 0.19431 0.000999\n",
      "Residuals                       186    71.701 0.38549         0.80569         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  52    30.629 0.58901  1.5541 0.34417 0.000999\n",
      "Residuals                       154    58.365 0.37899         0.65583         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  30    19.592 0.65306  1.6562 0.22015 0.000999\n",
      "Residuals                       176    69.401 0.39433         0.77985         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   5     3.869 0.77378  1.8271 0.04347 0.000999\n",
      "Residuals                       201    85.124 0.42350         0.95653         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: visibility\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13    12.395 0.95344  2.4023 0.13928 0.000999\n",
      "Residuals                       193    76.598 0.39688         0.86072         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12    11.644 0.97037  2.4338 0.13085 0.000999\n",
      "Residuals                       194    77.349 0.39871         0.86915         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     1.334 0.66692  1.5521 0.01499 0.004995\n",
      "Residuals                       204    87.659 0.42970         0.98501         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     3.648 0.91192  2.1584 0.04099 0.000999\n",
      "Residuals                       202    85.346 0.42250         0.95901         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     2.441 2.44058  5.7805 0.02742 0.000999\n",
      "Residuals                       205    86.553 0.42221         0.97258         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     7.014  3.5069  8.7266 0.07881 0.000999\n",
      "Residuals                       204    81.979  0.4019         0.92119         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     0.940 0.94021   2.189 0.01056 0.000999\n",
      "Residuals                       205    88.053 0.42953         0.98944         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     5.161 0.43007   2.197 0.11964 0.000999\n",
      "Residuals                       194    37.975 0.19575         0.88036         \n",
      "Total                           206    43.136                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20     9.026 0.45132  2.4611 0.20926 0.000999\n",
      "Residuals                       186    34.110 0.18338         0.79074         \n",
      "Total                           206    43.136                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  52    15.845 0.30471  1.7194 0.36733 0.000999\n",
      "Residuals                       154    27.291 0.17721         0.63267         \n",
      "Total                           206    43.136                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  30    10.699 0.35665  1.9351 0.24804 0.000999\n",
      "Residuals                       176    32.437 0.18430         0.75196         \n",
      "Total                           206    43.136                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   5     1.898 0.37951  1.8498 0.04399 0.000999\n",
      "Residuals                       201    41.238 0.20517         0.95601         \n",
      "Total                           206    43.136                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: visibility\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13     6.704 0.51566  2.7317 0.15541 0.000999\n",
      "Residuals                       193    36.432 0.18877         0.84459         \n",
      "Total                           206    43.136                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     6.319 0.52657  2.7747 0.14649 0.000999\n",
      "Residuals                       194    36.817 0.18978         0.85351         \n",
      "Total                           206    43.136                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   2     0.732 0.36600  1.7608 0.01697 0.01998 *\n",
      "Residuals                       204    42.404 0.20786         0.98303          \n",
      "Total                           206    43.136                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     2.463 0.61579  3.0583 0.0571 0.000999\n",
      "Residuals                       202    40.673 0.20135         0.9429         \n",
      "Total                           206    43.136                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     1.710 1.71010  8.4626 0.03964 0.000999\n",
      "Residuals                       205    41.426 0.20208         0.96036         \n",
      "Total                           206    43.136                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     3.992 1.99588  10.402 0.09254 0.000999\n",
      "Residuals                       204    39.144 0.19188         0.90746         \n",
      "Total                           206    43.136                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     0.547 0.54731  2.6345 0.01269 0.007992\n",
      "Residuals                       205    42.589 0.20775         0.98731         \n",
      "Total                           206    43.136                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     5.475 0.45621  1.3041 0.07465 0.000999\n",
      "Residuals                       194    67.865 0.34982         0.92535         \n",
      "Total                           206    73.340                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    10.367 0.51837  1.5311 0.14136 0.000999\n",
      "Residuals                       186    62.972 0.33856         0.85864         \n",
      "Total                           206    73.340                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  52    20.947 0.40282   1.184 0.28561 0.000999\n",
      "Residuals                       154    52.393 0.34021         0.71439         \n",
      "Total                           206    73.340                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  30    12.897 0.42990  1.2518 0.17585 0.000999\n",
      "Residuals                       176    60.443 0.34342         0.82415         \n",
      "Total                           206    73.340                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   5     2.842 0.56831  1.6203 0.03874 0.000999\n",
      "Residuals                       201    70.498 0.35074         0.96126         \n",
      "Total                           206    73.340                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: visibility\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13      7.42 0.57075   1.671 0.10117 0.000999\n",
      "Residuals                       193     65.92 0.34155         0.89883         \n",
      "Total                           206     73.34                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     7.659 0.63823  1.8851 0.10443 0.000999\n",
      "Residuals                       194    65.681 0.33856         0.89557         \n",
      "Total                           206    73.340                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   2     0.906 0.45292  1.2756 0.01235 0.02398 *\n",
      "Residuals                       204    72.434 0.35507         0.98765          \n",
      "Total                           206    73.340                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     3.106 0.77656  2.2335 0.04235 0.000999\n",
      "Residuals                       202    70.233 0.34769         0.95765         \n",
      "Total                           206    73.340                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     1.871 1.87080  5.3662 0.02551 0.000999\n",
      "Residuals                       205    71.469 0.34863         0.97449         \n",
      "Total                           206    73.340                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     3.688 1.84391  5.4006 0.05028 0.000999\n",
      "Residuals                       204    69.652 0.34143         0.94972         \n",
      "Total                           206    73.340                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model    R2   Pr(>F)   \n",
      "qiime.data$map[[opts$category]]   1     0.586 0.58638  1.6523 0.008 0.007992 **\n",
      "Residuals                       205    72.753 0.35489         0.992            \n",
      "Total                           206    73.340                 1.000            \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'log_20161031170717.txt', 'otu_table_subset_scleractinian_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12    19.792 1.64934  3.6721 0.0682 0.000999\n",
      "Residuals                       602   270.393 0.44916         0.9318         \n",
      "Total                           614   290.185                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20     22.01 1.10050  2.4376 0.07585 0.000999\n",
      "Residuals                       594    268.18 0.45147         0.92415         \n",
      "Total                           614    290.19                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model    R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  54    55.717 1.03179  2.4643 0.192 0.000999\n",
      "Residuals                       560   234.469 0.41869         0.808         \n",
      "Total                           614   290.185                 1.000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  31    39.607 1.27764  2.9726 0.13649 0.000999\n",
      "Residuals                       583   250.578 0.42981         0.86351         \n",
      "Total                           614   290.185                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   5     5.284 1.05690  2.2592 0.01821 0.000999\n",
      "Residuals                       609   284.901 0.46782         0.98179         \n",
      "Total                           614   290.185                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: visibility\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13    15.125 1.16343  2.5421 0.05212 0.000999\n",
      "Residuals                       601   275.061 0.45767         0.94788         \n",
      "Total                           614   290.185                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12    13.923 1.16029  2.5284 0.04798 0.000999\n",
      "Residuals                       602   276.262 0.45891         0.95202         \n",
      "Total                           614   290.185                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     1.455 0.72743  1.5419 0.00501 0.000999\n",
      "Residuals                       612   288.730 0.47178         0.99499         \n",
      "Total                           614   290.185                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     3.924 0.98110  2.0907 0.01352 0.000999\n",
      "Residuals                       610   286.261 0.46928         0.98648         \n",
      "Total                           614   290.185                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     2.216 2.21632  4.7179 0.00764 0.000999\n",
      "Residuals                       613   287.969 0.46977         0.99236         \n",
      "Total                           614   290.185                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     6.387  3.1937  6.8871 0.02201 0.000999\n",
      "Residuals                       612   283.798  0.4637         0.97799         \n",
      "Total                           614   290.185                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     1.008 1.00808  2.1369 0.00347 0.000999\n",
      "Residuals                       613   289.177 0.47174         0.99653         \n",
      "Total                           614   290.185                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12    17.188 1.43234  5.9777 0.10647 0.000999\n",
      "Residuals                       602   144.247 0.23961         0.89353         \n",
      "Total                           614   161.435                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    14.701 0.73507  2.9757 0.09107 0.000999\n",
      "Residuals                       594   146.734 0.24703         0.90893         \n",
      "Total                           614   161.435                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  54    35.697 0.66106  2.9442 0.22113 0.000999\n",
      "Residuals                       560   125.737 0.22453         0.77887         \n",
      "Total                           614   161.435                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  31    28.158 0.90833  3.9733 0.17442 0.000999\n",
      "Residuals                       583   133.277 0.22861         0.82558         \n",
      "Total                           614   161.435                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   5     2.479 0.49588  1.8998 0.01536 0.000999\n",
      "Residuals                       609   158.956 0.26101         0.98464         \n",
      "Total                           614   161.435                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: visibility\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13    10.384 0.79879  3.1782 0.06432 0.000999\n",
      "Residuals                       601   151.051 0.25133         0.93568         \n",
      "Total                           614   161.435                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     8.973 0.74771  2.9524 0.05558 0.000999\n",
      "Residuals                       602   152.462 0.25326         0.94442         \n",
      "Total                           614   161.435                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   2      0.76 0.37992  1.4471 0.00471 0.04895 *\n",
      "Residuals                       612    160.68 0.26254         0.99529          \n",
      "Total                           614    161.44                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4      3.29 0.82238  3.1721 0.02038 0.000999\n",
      "Residuals                       610    158.15 0.25925         0.97962         \n",
      "Total                           614    161.44                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     2.204 2.20357  8.4832 0.01365 0.000999\n",
      "Residuals                       613   159.231 0.25976         0.98635         \n",
      "Total                           614   161.435                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2      4.61 2.30517  8.9958 0.02856 0.000999\n",
      "Residuals                       612    156.82 0.25625         0.97144         \n",
      "Total                           614    161.44                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   1     0.489 0.48901  1.8625 0.00303 0.02198 *\n",
      "Residuals                       613   160.946 0.26255         0.99697          \n",
      "Total                           614   161.435                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12    13.364 1.11367  3.0027 0.05647 0.000999\n",
      "Residuals                       602   223.272 0.37088         0.94353         \n",
      "Total                           614   236.636                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    13.373 0.66866   1.779 0.05651 0.000999\n",
      "Residuals                       594   223.262 0.37586         0.94349         \n",
      "Total                           614   236.636                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  54    34.644 0.64156  1.7787 0.1464 0.000999\n",
      "Residuals                       560   201.991 0.36070         0.8536         \n",
      "Total                           614   236.636                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  31    24.441 0.78843  2.1662 0.10329 0.000999\n",
      "Residuals                       583   212.194 0.36397         0.89671         \n",
      "Total                           614   236.636                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   5     3.409 0.68182  1.7804 0.01441 0.000999\n",
      "Residuals                       609   233.226 0.38297         0.98559         \n",
      "Total                           614   236.636                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: visibility\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13     9.222 0.70937  1.8747 0.03897 0.000999\n",
      "Residuals                       601   227.414 0.37839         0.96103         \n",
      "Total                           614   236.636                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12    10.058 0.83820  2.2271 0.04251 0.000999\n",
      "Residuals                       602   226.577 0.37637         0.95749         \n",
      "Total                           614   236.636                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   2      0.90 0.44997  1.1682 0.0038 0.06893 .\n",
      "Residuals                       612    235.74 0.38519         0.9962          \n",
      "Total                           614    236.64                 1.0000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     3.817 0.95423  2.5001 0.01613 0.000999\n",
      "Residuals                       610   232.819 0.38167         0.98387         \n",
      "Total                           614   236.636                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     2.433 2.43260  6.3671 0.01028 0.000999\n",
      "Residuals                       613   234.203 0.38206         0.98972         \n",
      "Total                           614   236.636                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     4.248 2.12405  5.5938 0.01795 0.000999\n",
      "Residuals                       612   232.387 0.37972         0.98205         \n",
      "Total                           614   236.636                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   1     0.501 0.50070  1.2998 0.00212 0.05395 .\n",
      "Residuals                       613   236.135 0.38521         0.99788          \n",
      "Total                           614   236.636                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'log_20161031173523.txt', 'otu_table_subset_skeleton_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model    R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12    11.575 0.96459   2.195 0.119 0.000999\n",
      "Residuals                       195    85.693 0.43945         0.881         \n",
      "Total                           207    97.268                 1.000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    13.075 0.65373   1.452 0.13442 0.000999\n",
      "Residuals                       187    84.193 0.45023         0.86558         \n",
      "Total                           207    97.268                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  54    35.437 0.65624  1.6239 0.36433 0.000999\n",
      "Residuals                       153    61.831 0.40412         0.63567         \n",
      "Total                           207    97.268                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  31    24.343 0.78525  1.8952 0.25027 0.000999\n",
      "Residuals                       176    72.925 0.41435         0.74973         \n",
      "Total                           207    97.268                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   5     2.919 0.58384    1.25 0.03001 0.001998\n",
      "Residuals                       202    94.349 0.46707         0.96999         \n",
      "Total                           207    97.268                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: visibility\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13     8.667 0.66670  1.4598 0.08911 0.000999\n",
      "Residuals                       194    88.601 0.45671         0.91089         \n",
      "Total                           207    97.268                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     7.276 0.60634  1.3138 0.0748 0.000999\n",
      "Residuals                       195    89.992 0.46150         0.9252         \n",
      "Total                           207    97.268                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   2     1.050 0.52491  1.1183 0.01079 0.09191 .\n",
      "Residuals                       205    96.218 0.46936         0.98921          \n",
      "Total                           207    97.268                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   4     2.090 0.52255  1.1145 0.02149 0.04795 *\n",
      "Residuals                       203    95.178 0.46886         0.97851          \n",
      "Total                           207    97.268                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     0.804 0.80388  1.7167 0.00826 0.000999\n",
      "Residuals                       206    96.464 0.46827         0.99174         \n",
      "Total                           207    97.268                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     2.213 1.10654  2.3864 0.02275 0.000999\n",
      "Residuals                       205    95.055 0.46368         0.97725         \n",
      "Total                           207    97.268                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   1     0.628 0.62807  1.3388 0.00646 0.01499 *\n",
      "Residuals                       206    96.640 0.46913         0.99354          \n",
      "Total                           207    97.268                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12    10.751 0.89594  4.0691 0.20026 0.000999\n",
      "Residuals                       195    42.935 0.22018         0.79974         \n",
      "Total                           207    53.687                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20     8.748 0.43742  1.8202 0.16295 0.000999\n",
      "Residuals                       187    44.938 0.24031         0.83705         \n",
      "Total                           207    53.687                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  54    23.725 0.43935  2.2435 0.44191 0.000999\n",
      "Residuals                       153    29.962 0.19583         0.55809         \n",
      "Total                           207    53.687                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  31    18.153 0.58558  2.9004 0.33813 0.000999\n",
      "Residuals                       176    35.534 0.20190         0.66187         \n",
      "Total                           207    53.687                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]   5     1.217 0.24332 0.93675 0.02266 0.6084\n",
      "Residuals                       202    52.470 0.25975         0.97734       \n",
      "Total                           207    53.687                 1.00000       \n",
      "curr_cat: visibility\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13     6.415 0.49348  2.0252 0.11949 0.000999\n",
      "Residuals                       194    47.271 0.24367         0.88051         \n",
      "Total                           207    53.687                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     5.826 0.48549   1.978 0.10852 0.000999\n",
      "Residuals                       195    47.861 0.24544         0.89148         \n",
      "Total                           207    53.687                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model    R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     0.483 0.24154 0.93067 0.009 0.5305\n",
      "Residuals                       205    53.204 0.25953         0.991       \n",
      "Total                           207    53.687                 1.000       \n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     1.704 0.42611   1.664 0.03175 0.005994\n",
      "Residuals                       203    51.982 0.25607         0.96825         \n",
      "Total                           207    53.687                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     1.084 1.08392  4.2448 0.02019 0.000999\n",
      "Residuals                       206    52.603 0.25535         0.97981         \n",
      "Total                           207    53.687                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     2.450 1.22481  4.9005 0.04563 0.000999\n",
      "Residuals                       205    51.237 0.24994         0.95437         \n",
      "Total                           207    53.687                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     0.321 0.32097   1.239 0.00598 0.1888\n",
      "Residuals                       206    53.366 0.25906         0.99402       \n",
      "Total                           207    53.687                 1.00000       \n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12    10.409 0.86741  2.4235 0.12978 0.000999\n",
      "Residuals                       195    69.794 0.35792         0.87022         \n",
      "Total                           207    80.203                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20     9.803 0.49015  1.3019 0.12223 0.000999\n",
      "Residuals                       187    70.400 0.37647         0.87777         \n",
      "Total                           207    80.203                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  54    27.894 0.51656  1.5109 0.34779 0.000999\n",
      "Residuals                       153    52.309 0.34189         0.65221         \n",
      "Total                           207    80.203                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  31    19.321 0.62325  1.8017 0.2409 0.000999\n",
      "Residuals                       176    60.882 0.34592         0.7591         \n",
      "Total                           207    80.203                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   5     2.123 0.42462  1.0985 0.02647 0.07692 .\n",
      "Residuals                       202    78.080 0.38653         0.97353          \n",
      "Total                           207    80.203                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: visibility\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13     6.527 0.50205   1.322 0.08138 0.000999\n",
      "Residuals                       194    73.676 0.37977         0.91862         \n",
      "Total                           207    80.203                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     6.302 0.52515  1.3857 0.07857 0.000999\n",
      "Residuals                       195    73.901 0.37898         0.92143         \n",
      "Total                           207    80.203                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     0.832 0.41601  1.0745 0.01037 0.1828\n",
      "Residuals                       205    79.371 0.38718         0.98963       \n",
      "Total                           207    80.203                 1.00000       \n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     1.964 0.49104  1.2741 0.02449 0.006993\n",
      "Residuals                       203    78.239 0.38541         0.97551         \n",
      "Total                           207    80.203                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     0.883 0.88349  2.2945 0.01102 0.000999\n",
      "Residuals                       206    79.319 0.38505         0.98898         \n",
      "Total                           207    80.203                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     1.876 0.93802   2.455 0.02339 0.000999\n",
      "Residuals                       205    78.327 0.38208         0.97661         \n",
      "Total                           207    80.203                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   1     0.476 0.47618  1.2304 0.00594 0.07193 .\n",
      "Residuals                       206    79.727 0.38702         0.99406          \n",
      "Total                           207    80.203                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'log_20161031173800.txt', 'otu_table_subset_tissue_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12    14.732 1.22765  2.9101 0.15807 0.000999\n",
      "Residuals                       186    78.465 0.42186         0.84193         \n",
      "Total                           198    93.197                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20    14.883 0.74417  1.6914 0.1597 0.000999\n",
      "Residuals                       178    78.314 0.43996         0.8403         \n",
      "Total                           198    93.197                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  51    39.228 0.76917  2.0951 0.42091 0.000999\n",
      "Residuals                       147    53.969 0.36714         0.57909         \n",
      "Total                           198    93.197                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  30    28.173 0.93909  2.4263 0.30229 0.000999\n",
      "Residuals                       168    65.025 0.38705         0.69771         \n",
      "Total                           198    93.197                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   5     2.751 0.55027  1.1742 0.02952 0.02498 *\n",
      "Residuals                       193    90.446 0.46863         0.97048          \n",
      "Total                           198    93.197                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: visibility\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13     9.076 0.69812  1.5353 0.09738 0.000999\n",
      "Residuals                       185    84.122 0.45471         0.90262         \n",
      "Total                           198    93.197                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     7.949 0.66243  1.4453 0.08529 0.000999\n",
      "Residuals                       186    85.248 0.45832         0.91471         \n",
      "Total                           198    93.197                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     0.963 0.48165  1.0235 0.01034 0.3626\n",
      "Residuals                       196    92.234 0.47058         0.98966       \n",
      "Total                           198    93.197                 1.00000       \n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   4     2.125 0.53135  1.1319 0.02281 0.07093 .\n",
      "Residuals                       194    91.072 0.46944         0.97719          \n",
      "Total                           198    93.197                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     0.814 0.81376  1.7353 0.00873 0.001998\n",
      "Residuals                       197    92.383 0.46895         0.99127         \n",
      "Total                           198    93.197                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     2.868 1.43407  3.1117 0.03077 0.000999\n",
      "Residuals                       196    90.329 0.46086         0.96923         \n",
      "Total                           198    93.197                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   1     0.635 0.63542  1.3524 0.00682 0.03796 *\n",
      "Residuals                       197    92.562 0.46986         0.99318          \n",
      "Total                           198    93.197                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12    11.909 0.99245  4.3444 0.21892 0.000999\n",
      "Residuals                       186    42.490 0.22844         0.78108         \n",
      "Total                           198    54.400                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20     9.927 0.49634  1.9866 0.18248 0.000999\n",
      "Residuals                       178    44.473 0.24985         0.81752         \n",
      "Total                           198    54.400                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  51    26.258 0.51486  2.6894 0.48268 0.000999\n",
      "Residuals                       147    28.142 0.19144         0.51732         \n",
      "Total                           198    54.400                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  30    20.644 0.68814  3.4248 0.37949 0.000999\n",
      "Residuals                       168    33.756 0.20093         0.62051         \n",
      "Total                           198    54.400                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]   5     1.538 0.30765  1.1233 0.02828 0.2128\n",
      "Residuals                       193    52.862 0.27389         0.97172       \n",
      "Total                           198    54.400                 1.00000       \n",
      "curr_cat: visibility\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13     5.485 0.42192  1.5958 0.10083 0.000999\n",
      "Residuals                       185    48.915 0.26440         0.89917         \n",
      "Total                           198    54.400                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  12     3.983 0.33194  1.2246 0.07322 0.05694 .\n",
      "Residuals                       186    50.417 0.27106         0.92678          \n",
      "Total                           198    54.400                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     0.616 0.30801  1.1225 0.01132 0.2887\n",
      "Residuals                       196    53.784 0.27441         0.98868       \n",
      "Total                           198    54.400                 1.00000       \n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   4     1.412 0.35307  1.2927 0.02596 0.07692 .\n",
      "Residuals                       194    52.988 0.27313         0.97404          \n",
      "Total                           198    54.400                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     0.576 0.57583  2.1076 0.01059 0.007992\n",
      "Residuals                       197    53.824 0.27322         0.98941         \n",
      "Total                           198    54.400                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     1.902 0.95095  3.5504 0.03496 0.000999\n",
      "Residuals                       196    52.498 0.26785         0.96504         \n",
      "Total                           198    54.400                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]   1      0.42 0.42013  1.5333 0.00772 0.08292 .\n",
      "Residuals                       197     53.98 0.27401         0.99228          \n",
      "Total                           198     54.40                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     9.288 0.77397  2.1689 0.12275 0.000999\n",
      "Residuals                       186    66.375 0.35685         0.87725         \n",
      "Total                           198    75.663                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  20     9.686 0.48429  1.3066 0.12801 0.000999\n",
      "Residuals                       178    65.977 0.37066         0.87199         \n",
      "Total                           198    75.663                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  51    26.265 0.51500  1.5326 0.34713 0.000999\n",
      "Residuals                       147    49.398 0.33604         0.65287         \n",
      "Total                           198    75.663                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  30    18.207 0.60689  1.7745 0.24063 0.000999\n",
      "Residuals                       168    57.456 0.34200         0.75937         \n",
      "Total                           198    75.663                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: photosynthetically_active_radiation\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]   5     2.069 0.41371   1.085 0.02734 0.1149\n",
      "Residuals                       193    73.594 0.38132         0.97266       \n",
      "Total                           198    75.663                 1.00000       \n",
      "curr_cat: visibility\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  13     6.214 0.47803  1.2734 0.08213 0.000999\n",
      "Residuals                       185    69.448 0.37540         0.91787         \n",
      "Total                           198    75.663                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: turf_contact_percent\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  12     5.923 0.49355  1.3163 0.07828 0.000999\n",
      "Residuals                       186    69.740 0.37495         0.92172         \n",
      "Total                           198    75.663                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: n_macroalgal_contacts\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     0.783 0.39162  1.0251 0.01035 0.3217\n",
      "Residuals                       196    74.879 0.38204         0.98965       \n",
      "Total                           198    75.663                 1.00000       \n",
      "curr_cat: cyanobacteria_percent\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     1.986 0.49652  1.3074 0.02625 0.003996\n",
      "Residuals                       194    73.677 0.37978         0.97375         \n",
      "Total                           198    75.663                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: cyanobacteria_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     0.854 0.85357  2.2478 0.01128 0.001998\n",
      "Residuals                       197    74.809 0.37974         0.98872         \n",
      "Total                           198    75.663                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_turf_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     1.921 0.96066  2.5534 0.02539 0.000999\n",
      "Residuals                       196    73.741 0.37623         0.97461         \n",
      "Total                           198    75.663                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: binary_macroalgal_contact\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     0.437 0.43656  1.1433 0.00577 0.1439\n",
      "Residuals                       197    75.226 0.38186         0.99423       \n",
      "Total                           198    75.663                 1.00000       \n"
     ]
    },
    {
     "ename": "OSError",
     "evalue": "[Errno 20] Not a directory: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/compartment_tables/otu_table_subset_mucus.biom'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOSError\u001b[0m                                   Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-17-e78008f84888>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0;31m#    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m     \u001b[0mcurr_bdiv_dir\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtissue_otu_table_dir\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'%s'\u001b[0m \u001b[0;34m%\u001b[0m\u001b[0mc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 11\u001b[0;31m     \u001b[0;32mprint\u001b[0m \u001b[0mlistdir\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcurr_bdiv_dir\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     12\u001b[0m     \u001b[0mcurr_otu_table\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtissue_otu_table_dir\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     13\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mmetric\u001b[0m \u001b[0;32min\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'bray_curtis'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'weighted_unifrac'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'unweighted_unifrac'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mOSError\u001b[0m: [Errno 20] Not a directory: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/compartment_tables/otu_table_subset_mucus.biom'"
     ]
    }
   ],
   "source": [
    "#Check for differences among mapping categories within a given compartment\n",
    "tissue_otu_table_dir = join(output_dir,\"compartment_tables\")\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'host_clade_sensu_fukami,reef_name,field_host_name,field_host_genus_id,photosynthetically_active_radiation,visibility,turf_contact_percent,n_macroalgal_contacts,cyanobacteria_percent,cyanobacteria_contact,binary_turf_contact,binary_macroalgal_contact'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "#    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "    print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)   \n",
    "        for curr_cat in categories.split(\",\"):\n",
    "            print \"curr_cat: %s\" %curr_cat\n",
    "            print \"curr_metric: %s\" %metric\n",
    "            print \"curr_category: %s\" %c\n",
    "            curr_compare_cat_output_dir = join(curr_bdiv_dir,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "            !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "            the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "            !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'log_20161031170530.txt', 'otu_table_subset_mucus_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'log_20161031170717.txt', 'otu_table_subset_scleractinian_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'log_20161031173523.txt', 'otu_table_subset_skeleton_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'log_20161031173800.txt', 'otu_table_subset_tissue_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n"
     ]
    },
    {
     "ename": "OSError",
     "evalue": "[Errno 20] Not a directory: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/compartment_tables/otu_table_subset_mucus.biom'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOSError\u001b[0m                                   Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-18-610c8a8f4274>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;31m#    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0mcurr_bdiv_dir\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtissue_otu_table_dir\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'%s'\u001b[0m \u001b[0;34m%\u001b[0m\u001b[0mc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m     \u001b[0;32mprint\u001b[0m \u001b[0mlistdir\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcurr_bdiv_dir\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m     \u001b[0mcurr_otu_table\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtissue_otu_table_dir\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mmetric\u001b[0m \u001b[0;32min\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'bray_curtis'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'weighted_unifrac'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'unweighted_unifrac'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mOSError\u001b[0m: [Errno 20] Not a directory: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/compartment_tables/otu_table_subset_mucus.biom'"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries: pre\n",
    "tissue_otu_table_dir = join(output_dir,\"compartment_tables\")\n",
    "n_permutations = 1000\n",
    "categories = 'host_clade_sensu_fukami,reef_name,field_host_name,field_host_genus_id,photosynthetically_active_radiation,visibility,turf_contact_percent,n_macroalgal_contacts,cyanobacteria_percent,cyanobacteria_contact,binary_turf_contact,binary_macroalgal_contact'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "#    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "    print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Test for differences among functional groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'distance_boxplots_all_within_bray_curtis', 'distance_boxplots_all_within_unweighted_unifrac', 'distance_boxplots_all_within_weighted_unifrac', 'distance_boxplots_bray_curtis', 'distance_boxplots_unweighted_unifrac', 'distance_boxplots_weighted_unifrac', 'log_20161031170530.txt', 'otu_table_subset_mucus_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     2.508 0.62704  1.4646 0.02818 0.000999\n",
      "Residuals                       202    86.485 0.42814         0.97182         \n",
      "Total                           206    88.993                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     1.409 0.35234  1.7057 0.03267 0.006993\n",
      "Residuals                       202    41.727 0.20657         0.96733         \n",
      "Total                           206    43.136                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     1.786 0.44653  1.2606 0.02435 0.005994\n",
      "Residuals                       202    71.553 0.35422         0.97565         \n",
      "Total                           206    73.340                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'distance_boxplots_all_within_bray_curtis', 'distance_boxplots_all_within_unweighted_unifrac', 'distance_boxplots_all_within_weighted_unifrac', 'distance_boxplots_bray_curtis', 'distance_boxplots_unweighted_unifrac', 'distance_boxplots_weighted_unifrac', 'log_20161031170717.txt', 'otu_table_subset_scleractinian_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     7.446 1.86152  4.0162 0.02566 0.000999\n",
      "Residuals                       610   282.739 0.46351         0.97434         \n",
      "Total                           614   290.185                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     7.756 1.93902  7.6966 0.04804 0.000999\n",
      "Residuals                       610   153.679 0.25193         0.95196         \n",
      "Total                           614   161.435                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_scleractinian.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     7.218 1.80456  4.7981 0.0305 0.000999\n",
      "Residuals                       610   229.417 0.37609         0.9695         \n",
      "Total                           614   236.636                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'distance_boxplots_all_within_bray_curtis', 'distance_boxplots_all_within_unweighted_unifrac', 'distance_boxplots_all_within_weighted_unifrac', 'distance_boxplots_bray_curtis', 'distance_boxplots_unweighted_unifrac', 'distance_boxplots_weighted_unifrac', 'log_20161031173523.txt', 'otu_table_subset_skeleton_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     5.259 1.31477  2.9008 0.05407 0.000999\n",
      "Residuals                       203    92.009 0.45325         0.94593         \n",
      "Total                           207    97.268                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     6.159 1.53973  6.5765 0.11472 0.000999\n",
      "Residuals                       203    47.528 0.23413         0.88528         \n",
      "Total                           207    53.687                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     5.688 1.42212  3.8743 0.07093 0.000999\n",
      "Residuals                       203    74.515 0.36707         0.92907         \n",
      "Total                           207    80.203                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'distance_boxplots_all_within_bray_curtis', 'distance_boxplots_all_within_unweighted_unifrac', 'distance_boxplots_all_within_weighted_unifrac', 'distance_boxplots_bray_curtis', 'distance_boxplots_unweighted_unifrac', 'distance_boxplots_weighted_unifrac', 'log_20161031173800.txt', 'otu_table_subset_tissue_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     5.740 1.43505  3.1833 0.06159 0.000999\n",
      "Residuals                       194    87.457 0.45081         0.93841         \n",
      "Total                           198    93.197                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     5.957  1.4893  5.9641 0.10951 0.000999\n",
      "Residuals                       194    48.443  0.2497         0.89049         \n",
      "Total                           198    54.400                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: functional_group_sensu_darling\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   4     5.046 1.26140  3.4653 0.06669 0.000999\n",
      "Residuals                       194    70.617 0.36401         0.93331         \n",
      "Total                           198    75.663                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n"
     ]
    },
    {
     "ename": "OSError",
     "evalue": "[Errno 20] Not a directory: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/compartment_tables/otu_table_subset_mucus.biom'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOSError\u001b[0m                                   Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-5-ad9480c91b22>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0;31m#    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m     \u001b[0mcurr_bdiv_dir\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtissue_otu_table_dir\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'%s'\u001b[0m \u001b[0;34m%\u001b[0m\u001b[0mc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 11\u001b[0;31m     \u001b[0;32mprint\u001b[0m \u001b[0mlistdir\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcurr_bdiv_dir\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     12\u001b[0m     \u001b[0mcurr_otu_table\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtissue_otu_table_dir\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     13\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mmetric\u001b[0m \u001b[0;32min\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'bray_curtis'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'weighted_unifrac'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'unweighted_unifrac'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mOSError\u001b[0m: [Errno 20] Not a directory: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/compartment_tables/otu_table_subset_mucus.biom'"
     ]
    }
   ],
   "source": [
    "#Check for differences among mapping categories within a given compartment\n",
    "tissue_otu_table_dir = join(output_dir,\"compartment_tables\")\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'functional_group_sensu_darling'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "#    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "    print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)   \n",
    "        for curr_cat in categories.split(\",\"):\n",
    "            print \"curr_cat: %s\" %curr_cat\n",
    "            print \"curr_metric: %s\" %metric\n",
    "            print \"curr_category: %s\" %c\n",
    "            curr_compare_cat_output_dir = join(curr_bdiv_dir,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "            !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "            the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "            !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_functional_group_sensu_darling_bray_curtis', 'compare_categories_functional_group_sensu_darling_unweighted_unifrac', 'compare_categories_functional_group_sensu_darling_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'distance_boxplots_all_within_bray_curtis', 'distance_boxplots_all_within_unweighted_unifrac', 'distance_boxplots_all_within_weighted_unifrac', 'distance_boxplots_bray_curtis', 'distance_boxplots_unweighted_unifrac', 'distance_boxplots_weighted_unifrac', 'log_20161031170530.txt', 'otu_table_subset_mucus_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_functional_group_sensu_darling_bray_curtis', 'compare_categories_functional_group_sensu_darling_unweighted_unifrac', 'compare_categories_functional_group_sensu_darling_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'distance_boxplots_all_within_bray_curtis', 'distance_boxplots_all_within_unweighted_unifrac', 'distance_boxplots_all_within_weighted_unifrac', 'distance_boxplots_bray_curtis', 'distance_boxplots_unweighted_unifrac', 'distance_boxplots_weighted_unifrac', 'log_20161031170717.txt', 'otu_table_subset_scleractinian_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_functional_group_sensu_darling_bray_curtis', 'compare_categories_functional_group_sensu_darling_unweighted_unifrac', 'compare_categories_functional_group_sensu_darling_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'distance_boxplots_all_within_bray_curtis', 'distance_boxplots_all_within_unweighted_unifrac', 'distance_boxplots_all_within_weighted_unifrac', 'distance_boxplots_bray_curtis', 'distance_boxplots_unweighted_unifrac', 'distance_boxplots_weighted_unifrac', 'log_20161031173523.txt', 'otu_table_subset_skeleton_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "['.DS_Store', 'bray_curtis_dm.txt', 'bray_curtis_emperor_pcoa_plot', 'bray_curtis_pc.txt', 'compare_categories_binary_macroalgal_contact_bray_curtis', 'compare_categories_binary_macroalgal_contact_unweighted_unifrac', 'compare_categories_binary_macroalgal_contact_weighted_unifrac', 'compare_categories_binary_turf_contact_bray_curtis', 'compare_categories_binary_turf_contact_unweighted_unifrac', 'compare_categories_binary_turf_contact_weighted_unifrac', 'compare_categories_cyanobacteria_contact_bray_curtis', 'compare_categories_cyanobacteria_contact_unweighted_unifrac', 'compare_categories_cyanobacteria_contact_weighted_unifrac', 'compare_categories_cyanobacteria_percent_bray_curtis', 'compare_categories_cyanobacteria_percent_unweighted_unifrac', 'compare_categories_cyanobacteria_percent_weighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_bray_curtis', 'compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_field_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_field_host_genus_id_bray_curtis', 'compare_categories_field_host_genus_id_unweighted_unifrac', 'compare_categories_field_host_genus_id_weighted_unifrac', 'compare_categories_field_host_name_bray_curtis', 'compare_categories_field_host_name_unweighted_unifrac', 'compare_categories_field_host_name_weighted_unifrac', 'compare_categories_functional_group_sensu_darling_bray_curtis', 'compare_categories_functional_group_sensu_darling_unweighted_unifrac', 'compare_categories_functional_group_sensu_darling_weighted_unifrac', 'compare_categories_host_clade_sensu_fukami_bray_curtis', 'compare_categories_host_clade_sensu_fukami_unweighted_unifrac', 'compare_categories_host_clade_sensu_fukami_weighted_unifrac', 'compare_categories_n_macroalgal_contacts_bray_curtis', 'compare_categories_n_macroalgal_contacts_unweighted_unifrac', 'compare_categories_n_macroalgal_contacts_weighted_unifrac', 'compare_categories_photosynthetically_active_radiation_bray_curtis', 'compare_categories_photosynthetically_active_radiation_unweighted_unifrac', 'compare_categories_photosynthetically_active_radiation_weighted_unifrac', 'compare_categories_reef_name.x_bray_curtis', 'compare_categories_reef_name.x_unweighted_unifrac', 'compare_categories_reef_name.x_weighted_unifrac', 'compare_categories_reef_name_bray_curtis', 'compare_categories_reef_name_unweighted_unifrac', 'compare_categories_reef_name_weighted_unifrac', 'compare_categories_turf_contact_percent_bray_curtis', 'compare_categories_turf_contact_percent_unweighted_unifrac', 'compare_categories_turf_contact_percent_weighted_unifrac', 'compare_categories_visibility_bray_curtis', 'compare_categories_visibility_unweighted_unifrac', 'compare_categories_visibility_weighted_unifrac', 'distance_boxplots_all_within_bray_curtis', 'distance_boxplots_all_within_unweighted_unifrac', 'distance_boxplots_all_within_weighted_unifrac', 'distance_boxplots_bray_curtis', 'distance_boxplots_unweighted_unifrac', 'distance_boxplots_weighted_unifrac', 'log_20161031173800.txt', 'otu_table_subset_tissue_even1000.biom', 'unweighted_unifrac_dm.txt', 'unweighted_unifrac_emperor_pcoa_plot', 'unweighted_unifrac_pc.txt', 'weighted_unifrac_dm.txt', 'weighted_unifrac_emperor_pcoa_plot', 'weighted_unifrac_pc.txt']\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n"
     ]
    },
    {
     "ename": "OSError",
     "evalue": "[Errno 20] Not a directory: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/compartment_tables/otu_table_subset_mucus.biom'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOSError\u001b[0m                                   Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-6-51990cc96656>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;31m#    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0mcurr_bdiv_dir\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtissue_otu_table_dir\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'%s'\u001b[0m \u001b[0;34m%\u001b[0m\u001b[0mc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m     \u001b[0;32mprint\u001b[0m \u001b[0mlistdir\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcurr_bdiv_dir\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m     \u001b[0mcurr_otu_table\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtissue_otu_table_dir\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mc\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mmetric\u001b[0m \u001b[0;32min\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'bray_curtis'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'weighted_unifrac'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'unweighted_unifrac'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mOSError\u001b[0m: [Errno 20] Not a directory: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/compartment_tables/otu_table_subset_mucus.biom'"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries:\n",
    "tissue_otu_table_dir = join(output_dir,\"compartment_tables\")\n",
    "n_permutations = 1000\n",
    "categories = 'functional_group_sensu_darling'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "#    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "    print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Test for differences in all samples (i.e. not partioned by compartment). This is useful for identifying differences among compartments.\n",
    "categories = 'BiologicalMatter,tissue_compartment,host_clade_sensu_fukami,reef_name,field_host_name,field_host_genus_id,photosynthetically_active_radiation,visibility,turf_contact_percent,n_macroalgal_contacts,cyanobacteria_percent,cyanobacteria_contact,binary_turf_contact,binary_macroalgal_contact'\n",
    "for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "    curr_boxplot_output_dir = join(bdiv_dir_all,\"distance_boxplots_%s\" %metric)\n",
    "    curr_dm = join(bdiv_dir_all,\"%s_dm.txt\" %metric)\n",
    "    !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Run beta diversity on all samples in dataset rarified to 1000 reads\n",
    "tissue_otu_table_dir = join(output_dir,\"bdiv_all\")\n",
    "depth = 1000\n",
    "\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith(\".biom\"):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(output_dir,'bdiv_%s' %c)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "\n",
    "!beta_diversity_through_plots.py -i $curr_otu_table -o $curr_bdiv_dir -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Convert unrarified BIOM to CSV for manipulation in PRIMER\n",
    "!biom convert -i $input_dir/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom -o $input_dir/otu_table_mc2_wtax_no_pynast_failures_no_organelles_TXT.txt --to-tsv --header-key taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Convert rarified (1000) BIOM to CSV for manipulation in PRIMER\n",
    "!biom convert -i $output_dir/bdiv_otu_table_mc2_wtax_no_pynast_failures_no_organelles_even1000.biom/otu_table_mc2_wtax_no_pynast_failures_no_organelles_even1000.biom -o $input_dir/otu_table_mc2_wtax_no_pynast_failures_no_organelles_even1000_TXT.txt --to-tsv --header-key taxonomy"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Test for significant differences in community composition among coral compartments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Test for significant differences among coral compartments\n",
    "!compare_categories.py --method adonis -i output/bdiv_all/unweighted_unifrac_dm.txt -m input/gcmp16S_map_r20.txt -c BiologicalMatter -o output/significance_testing_compartments/compare_categories_compartments_unweighted_unifrac_adonis.txt -n 999"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/qiime/stats.py:1247: RuntimeWarning: divide by zero encountered in double_scalars\r\n",
      "  q = 1. + (a + 1.) / (6. * n)\r\n"
     ]
    }
   ],
   "source": [
    "#Explore which OTUs differ significantly among compartments\n",
    "!group_significance.py -i input/otu_table_mc2_wtax_no_pynast_failures_no_organelles_coral_tissue_mucus_skeleton_only.biom -m input/gcmp16S_map_r20.txt -c BiologicalMatter -s g_test -o output/significance_testing_compartments/group_significance_OTU_BiologicalMatter_g_test.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Does host genus drives microbiome structure in brooders to a greater extent than spawners? We will also look at reef and season."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Set file paths\n",
    "input_dir = abspath(\"input/\")\n",
    "otu_table = abspath(\"input/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom\")\n",
    "tree_fp = abspath(\"input/rep_set.tre\")\n",
    "mapping = abspath(\"input/gcmp16S_map.txt\")\n",
    "output_dir = abspath(\"output/\")\n",
    "\n",
    "#Set file paths\n",
    "input_dir = abspath(\"input/\")\n",
    "otu_table = abspath(\"input/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom\")\n",
    "tree_fp = abspath(\"input/rep_set.tre\")\n",
    "mapping = abspath(\"input/gcmp16S_map_r20.txt\")\n",
    "output_dir = abspath(\"output/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Split OTU table into brooder- and spawner-specific biom files\n",
    "subset_output = join(output_dir,'brooder_v_spawner')\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_brooder.biom -m $mapping -s 'Mode_of_larval_development:brooder'\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_spawner.biom -m $mapping -s 'Mode_of_larval_development:spawner'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Export Sceleractinian coral compartment specific biom files (i.e. coral t, m and s) for brooders\n",
    "subset_output = join(output_dir,'brooder_v_spawner')\n",
    "brooder_otu_table = join(subset_output,'otu_table_subset_brooder.biom')\n",
    "spawner_otu_table = join(subset_output,'otu_table_subset_spawner.biom')\n",
    "#Export Sceleractinian coral compartment specific biom files (i.e. coral t, m and s) for brooders\n",
    "!filter_samples_from_otu_table.py -i $brooder_otu_table -o $subset_output/otu_table_subset_brooder_mucus.biom -m $mapping -s 'BiologicalMatter:Coral Mucus'\n",
    "!filter_samples_from_otu_table.py -i $brooder_otu_table -o $subset_output/otu_table_subset_brooder_skeleton.biom -m $mapping -s 'BiologicalMatter:Coral Skeleton'\n",
    "!filter_samples_from_otu_table.py -i $brooder_otu_table -o $subset_output/otu_table_subset_brooder_tissue.biom -m $mapping -s 'BiologicalMatter:Coral Tissue'\n",
    "#Export Sceleractinian coral compartment specific biom files (i.e. coral t, m and s) for spawners\n",
    "!filter_samples_from_otu_table.py -i $spawner_otu_table -o $subset_output/otu_table_subset_spawner_mucus.biom -m $mapping -s 'BiologicalMatter:Coral Mucus'\n",
    "!filter_samples_from_otu_table.py -i $spawner_otu_table -o $subset_output/otu_table_subset_spawner_skeleton.biom -m $mapping -s 'BiologicalMatter:Coral Skeleton'\n",
    "!filter_samples_from_otu_table.py -i $spawner_otu_table -o $subset_output/otu_table_subset_spawner_tissue.biom -m $mapping -s 'BiologicalMatter:Coral Tissue'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#Summarize BIOM to confirm filter functioned correctly\n",
    "subset_output = join(output_dir,'brooder_v_spawner')\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_brooder_mucus.biom -o $subset_output/otu_table_subset_brooder_mucus_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_brooder_skeleton.biom -o $subset_output/otu_table_subset_brooder_skeleton_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_brooder_tissue.biom -o $subset_output/otu_table_subset_brooder_tissue_summary.txt\n",
    "\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_spawner_mucus.biom -o $subset_output/otu_table_subset_spawner_mucus_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_spawner_skeleton.biom -o $subset_output/otu_table_subset_spawner_skeleton_summary.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_spawner_tissue.biom -o $subset_output/otu_table_subset_spawner_tissue_summary.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Looks good!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0412282144692 and the largest is 3.13193392707.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0579053952104 and the largest is 1.99439774237.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00477819038945 and the largest is 1.51279918088.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0125366593789 and the largest is 1.62546157568.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00536576056968 and the largest is 0.545835871795.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.129330573672 and the largest is 10.7675230558.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.510264543752 and the largest is 14.6681864957.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0352210408835 and the largest is 6.01557564504.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.119477033685 and the largest is 6.80736188985.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0853420629271 and the largest is 3.52993370452.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.179626778374 and the largest is 7.92810281431.\n",
      "  RuntimeWarning\n",
      "/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0258044802679 and the largest is 4.10679319982.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.132583047246 and the largest is 5.24913276903.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on rarified (1000) subset of each sample type\n",
    "tissue_otu_table_dir = join(output_dir,\"brooder_v_spawner\")\n",
    "depth = 1000\n",
    "\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith(\".biom\"):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    print(curr_otu_table)\n",
    "    !beta_diversity_through_plots.py -i $curr_otu_table -o $curr_bdiv_dir -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    2.2918 2.29178  5.6962 0.09873 0.000999\n",
      "Residuals                       52   20.9213 0.40233         0.90127         \n",
      "Total                           53   23.2131                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    4.0682 1.01704   2.603 0.17525 0.000999\n",
      "Residuals                       49   19.1449 0.39071         0.82475         \n",
      "Total                           53   23.2131                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    3.3865 1.69323  4.3555 0.14589 0.000999\n",
      "Residuals                       51   19.8266 0.38876         0.85411         \n",
      "Total                           53   23.2131                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    3.3865 1.69323  4.3555 0.14589 0.000999\n",
      "Residuals                       51   19.8266 0.38876         0.85411         \n",
      "Total                           53   23.2131                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)   \n",
      "qiime.data$map[[opts$category]]  1    0.5117 0.51174  2.3023 0.0424 0.006993 **\n",
      "Residuals                       52   11.5581 0.22227         0.9576            \n",
      "Total                           53   12.0699                 1.0000            \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    1.5385 0.38463  1.7896 0.12747 0.000999\n",
      "Residuals                       49   10.5313 0.21493         0.87253         \n",
      "Total                           53   12.0699                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.1709 0.58543  2.7394 0.09701 0.000999\n",
      "Residuals                       51   10.8990 0.21371         0.90299         \n",
      "Total                           53   12.0699                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.1709 0.58543  2.7394 0.09701 0.000999\n",
      "Residuals                       51   10.8990 0.21371         0.90299         \n",
      "Total                           53   12.0699                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1     0.704 0.70397  2.0673 0.03823 0.000999\n",
      "Residuals                       52    17.708 0.34053         0.96177         \n",
      "Total                           53    18.412                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    1.9585 0.48963  1.4582 0.10637 0.000999\n",
      "Residuals                       49   16.4532 0.33578         0.89363         \n",
      "Total                           53   18.4117                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.2463 0.62317  1.8515 0.06769 0.000999\n",
      "Residuals                       51   17.1654 0.33658         0.93231         \n",
      "Total                           53   18.4117                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.2463 0.62317  1.8515 0.06769 0.000999\n",
      "Residuals                       51   17.1654 0.33658         0.93231         \n",
      "Total                           53   18.4117                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    1.1380 1.13797  2.9871 0.14233 0.000999\n",
      "Residuals                       18    6.8574 0.38097         0.85767         \n",
      "Total                           19    7.9953                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    2.5637 0.64092  1.7699 0.32064 0.000999\n",
      "Residuals                       15    5.4317 0.36211         0.67936         \n",
      "Total                           19    7.9953                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)   \n",
      "qiime.data$map[[opts$category]]  2    1.6047 0.80235  2.1344 0.2007 0.001998 **\n",
      "Residuals                       17    6.3906 0.37592         0.7993            \n",
      "Total                           19    7.9953                 1.0000            \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.6047 0.80235  2.1344 0.2007 0.000999\n",
      "Residuals                       17    6.3906 0.37592         0.7993         \n",
      "Total                           19    7.9953                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  1    0.3819 0.38188  1.8853 0.09481 0.04396 *\n",
      "Residuals                       18    3.6461 0.20256         0.90519          \n",
      "Total                           19    4.0279                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    1.4320 0.35801  2.0687 0.35553 0.000999\n",
      "Residuals                       15    2.5959 0.17306         0.64447         \n",
      "Total                           19    4.0279                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  2    0.5875 0.29374  1.4514 0.14585 0.06993 .\n",
      "Residuals                       17    3.4405 0.20238         0.85415          \n",
      "Total                           19    4.0279                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  2    0.5875 0.29374  1.4514 0.14585 0.07493 .\n",
      "Residuals                       17    3.4405 0.20238         0.85415          \n",
      "Total                           19    4.0279                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  1    0.4701 0.47013   1.454 0.07474 0.02198 *\n",
      "Residuals                       18    5.8202 0.32334         0.92526          \n",
      "Total                           19    6.2903                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    1.6706 0.41764  1.3561 0.26558 0.004995\n",
      "Residuals                       15    4.6197 0.30798         0.73442         \n",
      "Total                           19    6.2903                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2 Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  2    0.7668 0.38341    1.18 0.1219 0.0989 .\n",
      "Residuals                       17    5.5235 0.32491         0.8781         \n",
      "Total                           19    6.2903                 1.0000         \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    0.7668 0.38341    1.18 0.1219 0.1009\n",
      "Residuals                       17    5.5235 0.32491         0.8781       \n",
      "Total                           19    6.2903                 1.0000       \n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    0.8547 0.85475  2.2035 0.12105 0.002997\n",
      "Residuals                       16    6.2064 0.38790         0.87895         \n",
      "Total                           17    7.0612                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    2.2888  0.5722  1.5587 0.32414 0.001998\n",
      "Residuals                       13    4.7723  0.3671         0.67586         \n",
      "Total                           17    7.0612                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.8080 0.90399  2.5813 0.25605 0.000999\n",
      "Residuals                       15    5.2532 0.35021         0.74395         \n",
      "Total                           17    7.0612                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.8080 0.90399  2.5813 0.25605 0.000999\n",
      "Residuals                       15    5.2532 0.35021         0.74395         \n",
      "Total                           17    7.0612                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    0.2310 0.23101  0.9102 0.05383 0.5005\n",
      "Residuals                       16    4.0609 0.25381         0.94617       \n",
      "Total                           17    4.2919                 1.00000       \n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    1.6263 0.40656  1.9828 0.37891 0.002997\n",
      "Residuals                       13    2.6656 0.20505         0.62109         \n",
      "Total                           17    4.2919                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  2    1.0542 0.52708  2.4419 0.24561 0.01199 *\n",
      "Residuals                       15    3.2377 0.21585         0.75439          \n",
      "Total                           17    4.2919                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.0542 0.52708  2.4419 0.24561 0.008991\n",
      "Residuals                       15    3.2377 0.21585         0.75439         \n",
      "Total                           17    4.2919                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  1    0.4287 0.42867  1.3006 0.07518 0.02198 *\n",
      "Residuals                       16    5.2736 0.32960         0.92482          \n",
      "Total                           17    5.7022                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  4    1.4845 0.37113  1.1439 0.26034 0.03497 *\n",
      "Residuals                       13    4.2177 0.32444         0.73966          \n",
      "Total                           17    5.7022                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    0.8395 0.41974  1.2948 0.14722 0.001998\n",
      "Residuals                       15    4.8628 0.32418         0.85278         \n",
      "Total                           17    5.7022                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    0.8395 0.41974  1.2948 0.14722 0.002997\n",
      "Residuals                       15    4.8628 0.32418         0.85278         \n",
      "Total                           17    5.7022                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    1.0545 1.05448  3.0624 0.17948 0.002997\n",
      "Residuals                       14    4.8206 0.34433         0.82052         \n",
      "Total                           15    5.8751                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4    2.8506 0.71264  2.5918 0.48519 0.001998\n",
      "Residuals                       11    3.0246 0.27496         0.51481         \n",
      "Total                           15    5.8751                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.9162 0.95810  3.1461 0.32615 0.000999\n",
      "Residuals                       13    3.9589 0.30453         0.67385         \n",
      "Total                           15    5.8751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    1.9162 0.95810  3.1461 0.32615 0.000999\n",
      "Residuals                       13    3.9589 0.30453         0.67385         \n",
      "Total                           15    5.8751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  1   0.23771 0.23771  1.8194 0.11501 0.05794 .\n",
      "Residuals                       14   1.82913 0.13065         0.88499          \n",
      "Total                           15   2.06685                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]]  4   0.70004 0.17501  1.4085 0.3387 0.1329\n",
      "Residuals                       11   1.36680 0.12426         0.6613       \n",
      "Total                           15   2.06685                 1.0000       \n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2   0.55841 0.27921  2.4063 0.27018 0.004995\n",
      "Residuals                       13   1.50843 0.11603         0.72982         \n",
      "Total                           15   2.06685                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2   0.55841 0.27921  2.4063 0.27018 0.003996\n",
      "Residuals                       13   1.50843 0.11603         0.72982         \n",
      "Total                           15   2.06685                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    0.5304 0.53039  1.5603 0.10027 0.000999\n",
      "Residuals                       14    4.7590 0.33993         0.89973         \n",
      "Total                           15    5.2894                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]]  4    1.5642 0.39105  1.1547 0.29572 0.04595 *\n",
      "Residuals                       11    3.7252 0.33866         0.70428          \n",
      "Total                           15    5.2894                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    0.9258 0.46290  1.3791 0.17503 0.000999\n",
      "Residuals                       13    4.3636 0.33566         0.82497         \n",
      "Total                           15    5.2894                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_brooder_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  2    0.9258 0.46290  1.3791 0.17503 0.002997\n",
      "Residuals                       13    4.3636 0.33566         0.82497         \n",
      "Total                           15    5.2894                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     12.58 1.57249  3.5141 0.07115 0.000999\n",
      "Residuals                       367    164.23 0.44748         0.92885         \n",
      "Total                           375    176.81                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    18.255 0.96081  2.1574 0.10325 0.000999\n",
      "Residuals                       356   158.551 0.44537         0.89675         \n",
      "Total                           375   176.806                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26    27.644  1.0632  2.4877 0.15635 0.000999\n",
      "Residuals                       349   149.162  0.4274         0.84365         \n",
      "Total                           375   176.806                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18    22.482 1.24899  2.8893 0.12715 0.000999\n",
      "Residuals                       357   154.324 0.43228         0.87285         \n",
      "Total                           375   176.806                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8    10.977 1.37216  4.9533 0.09745 0.000999\n",
      "Residuals                       367   101.665 0.27702         0.90255         \n",
      "Total                           375   112.643                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    11.644 0.61283  2.1601 0.10337 0.000999\n",
      "Residuals                       356   100.999 0.28370         0.89663         \n",
      "Total                           375   112.643                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26    21.257 0.81758  3.1223 0.18871 0.000999\n",
      "Residuals                       349    91.385 0.26185         0.81129         \n",
      "Total                           375   112.643                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18    18.266 1.01475  3.8385 0.16215 0.000999\n",
      "Residuals                       357    94.377 0.26436         0.83785         \n",
      "Total                           375   112.643                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     9.104 1.13796  3.1229 0.06374 0.000999\n",
      "Residuals                       367   133.730 0.36439         0.93626         \n",
      "Total                           375   142.833                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    10.729 0.56469  1.5218 0.07512 0.000999\n",
      "Residuals                       356   132.104 0.37108         0.92488         \n",
      "Total                           375   142.833                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26    17.758 0.68300  1.9058 0.12433 0.000999\n",
      "Residuals                       349   125.076 0.35838         0.87567         \n",
      "Total                           375   142.833                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18     14.53 0.80720   2.246 0.10172 0.000999\n",
      "Residuals                       357    128.30 0.35939         0.89828         \n",
      "Total                           375    142.83                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     5.885 0.73561  1.8393 0.11257 0.000999\n",
      "Residuals                       116    46.394 0.39995         0.88743         \n",
      "Total                           124    52.279                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    13.840 0.72841  1.9897 0.26473 0.000999\n",
      "Residuals                       105    38.439 0.36608         0.73527         \n",
      "Total                           124    52.279                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24    13.965 0.58189  1.5188 0.26713 0.000999\n",
      "Residuals                       100    38.313 0.38313         0.73287         \n",
      "Total                           124    52.279                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17    10.813 0.63603  1.6412 0.20682 0.000999\n",
      "Residuals                       107    41.466 0.38753         0.79318         \n",
      "Total                           124    52.279                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8    3.9254 0.49067  2.1188 0.12749 0.000999\n",
      "Residuals                       116   26.8636 0.23158         0.87251         \n",
      "Total                           124   30.7890                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    8.7082 0.45832  2.1795 0.28283 0.000999\n",
      "Residuals                       105   22.0808 0.21029         0.71717         \n",
      "Total                           124   30.7890                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24    8.8741 0.36975  1.6872 0.28822 0.000999\n",
      "Residuals                       100   21.9149 0.21915         0.71178         \n",
      "Total                           124   30.7890                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17    7.0808 0.41652  1.8798 0.22998 0.000999\n",
      "Residuals                       107   23.7082 0.22157         0.77002         \n",
      "Total                           124   30.7890                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     3.398 0.42471  1.2519 0.07948 0.000999\n",
      "Residuals                       116    39.353 0.33925         0.92052         \n",
      "Total                           124    42.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     8.409 0.44256  1.3531 0.19669 0.000999\n",
      "Residuals                       105    34.342 0.32707         0.80331         \n",
      "Total                           124    42.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24     9.401 0.39172  1.1746 0.21991 0.000999\n",
      "Residuals                       100    33.350 0.33350         0.78009         \n",
      "Total                           124    42.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_mucus.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17     7.111 0.41830  1.2559 0.16634 0.000999\n",
      "Residuals                       107    35.640 0.33308         0.83366         \n",
      "Total                           124    42.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8      7.73 0.96630  2.2032 0.12807 0.000999\n",
      "Residuals                       120     52.63 0.43858         0.87193         \n",
      "Total                           128     60.36                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    11.834 0.62283   1.399 0.19605 0.000999\n",
      "Residuals                       109    48.526 0.44520         0.80395         \n",
      "Total                           128    60.360                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26     18.46 0.71001  1.7284 0.30584 0.000999\n",
      "Residuals                       102     41.90 0.41078         0.69416         \n",
      "Total                           128     60.36                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18    14.609 0.81163  1.9514 0.24204 0.000999\n",
      "Residuals                       110    45.751 0.41592         0.75796         \n",
      "Total                           128    60.360                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     8.532 1.06646  4.0227 0.21147 0.000999\n",
      "Residuals                       120    31.814 0.26511         0.78853         \n",
      "Total                           128    40.345                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     8.357 0.43982  1.4987 0.20713 0.000999\n",
      "Residuals                       109    31.989 0.29347         0.79287         \n",
      "Total                           128    40.345                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26    16.520 0.63538  2.7201 0.40946 0.000999\n",
      "Residuals                       102    23.825 0.23358         0.59054         \n",
      "Total                           128    40.345                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18    14.046 0.78034  3.2639 0.34815 0.000999\n",
      "Residuals                       110    26.299 0.23908         0.65185         \n",
      "Total                           128    40.345                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model    R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     7.355 0.91933  2.6056 0.148 0.000999\n",
      "Residuals                       120    42.338 0.35282         0.852         \n",
      "Total                           128    49.693                 1.000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     8.857 0.46613  1.2442 0.17822 0.000999\n",
      "Residuals                       109    40.837 0.37465         0.82178         \n",
      "Total                           128    49.693                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  26    14.646  0.5633  1.6394 0.29472 0.000999\n",
      "Residuals                       102    35.047  0.3436         0.70528         \n",
      "Total                           128    49.693                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_skeleton.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  18    11.688 0.64934  1.8794 0.2352 0.000999\n",
      "Residuals                       110    38.005 0.34550         0.7648         \n",
      "Total                           128    49.693                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     9.638 1.20478  2.8782 0.16927 0.000999\n",
      "Residuals                       113    47.300 0.41859         0.83073         \n",
      "Total                           121    56.938                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    12.690 0.66790  1.5396 0.22288 0.000999\n",
      "Residuals                       102    44.248 0.43381         0.77712         \n",
      "Total                           121    56.938                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24    20.264 0.84433  2.2332 0.35589 0.000999\n",
      "Residuals                        97    36.675 0.37809         0.64411         \n",
      "Total                           121    56.938                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17    16.616 0.97743   2.521 0.29183 0.000999\n",
      "Residuals                       104    40.322 0.38771         0.70817         \n",
      "Total                           121    56.938                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     7.292 0.91154  3.8984 0.2163 0.000999\n",
      "Residuals                       113    26.422 0.23382         0.7837         \n",
      "Total                           121    33.715                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     7.827 0.41194  1.6231 0.23215 0.000999\n",
      "Residuals                       102    25.888 0.25380         0.76785         \n",
      "Total                           121    33.715                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24    14.273 0.59472  2.9672 0.42335 0.000999\n",
      "Residuals                        97    19.441 0.20043         0.57665         \n",
      "Total                           121    33.715                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17    12.391 0.72888  3.5549 0.36753 0.000999\n",
      "Residuals                       104    21.324 0.20503         0.63247         \n",
      "Total                           121    33.715                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   8     6.526 0.81572   2.363 0.14332 0.000999\n",
      "Residuals                       113    39.008 0.34520         0.85668         \n",
      "Total                           121    45.533                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     8.522 0.44850   1.236 0.18715 0.000999\n",
      "Residuals                       102    37.012 0.36286         0.81285         \n",
      "Total                           121    45.533                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  24    13.647 0.56861  1.7297 0.29971 0.000999\n",
      "Residuals                        97    31.887 0.32873         0.70029         \n",
      "Total                           121    45.533                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_spawner_tissue.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  17    10.948 0.64400  1.9365 0.24044 0.000999\n",
      "Residuals                       104    34.585 0.33255         0.75956         \n",
      "Total                           121    45.533                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_mucus.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_skeleton.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_brooder_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_brooder_tissue.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_mucus.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_mucus.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_skeleton.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_skeleton.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_clade_sensu_fukami_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_reef_name.x_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_name_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_genus_id_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_clade_sensu_fukami_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_reef_name.x_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_name_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_genus_id_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_clade_sensu_fukami\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_clade_sensu_fukami_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: reef_name.x\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_reef_name.x_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_name\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_name_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: field_host_genus_id\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_spawner_tissue.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/brooder_v_spawner/otu_table_subset_spawner_tissue.biom/compare_categories_field_host_genus_id_unweighted_unifrac/adonis_results.txt: Not a directory\n"
     ]
    }
   ],
   "source": [
    "#Check for differences among mapping categories within a given compartment\n",
    "tissue_otu_table_dir = join(output_dir,\"brooder_v_spawner\")\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'field_host_clade_sensu_fukami,reef_name.x,field_host_name,field_host_genus_id'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    " #   print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)   \n",
    "        for curr_cat in categories.split(\",\"):\n",
    "            print \"curr_cat: %s\" %curr_cat\n",
    "            print \"curr_metric: %s\" %metric\n",
    "            print \"curr_category: %s\" %c\n",
    "            curr_compare_cat_output_dir = join(curr_bdiv_dir,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "            !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "            the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "            !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# We see significant correlations between alpha diversity and 1) latitude and 2) proportion of max colony size. These effects are most pronounced in stress-tolerant corals. Here, we will see if overall community structure changes 1) along latitudinal gradient and 2) as corals grow larger. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Set file paths\n",
    "input_dir = abspath(\"input/\")\n",
    "otu_table = abspath(\"input/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom\")\n",
    "tree_fp = abspath(\"input/rep_set.tre\")\n",
    "mapping = abspath(\"input/gcmp16S_map_r20.txt\")\n",
    "output_dir = abspath(\"output/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Split OTU table into different functional groups\n",
    "subset_output = join(output_dir,'functional_groups')\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_generalist.biom -m $mapping -s 'functional_group_sensu_darling:Generalist'\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_competitive.biom -m $mapping -s 'functional_group_sensu_darling:Competitive'\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_stress_tolerant.biom -m $mapping -s 'functional_group_sensu_darling:Stress-tolerant'\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_subset_weedy.biom -m $mapping -s 'functional_group_sensu_darling:Weedy'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#Summarize BIOM to confirm filter functioned correctly\n",
    "subset_output = join(output_dir,'functional_groups')\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_generalist.biom -o $subset_output/otu_table_subset_generalist.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_competitive.biom -o $subset_output/otu_table_subset_competitive.txt\n",
    "!biom summarize-table -i $subset_output/otu_t able_subset_stress_tolerant.biom -o $subset_output/otu_table_subset_stress_tolerant.txt\n",
    "!biom summarize-table -i $subset_output/otu_table_subset_weedy.biom -o $subset_output/otu_table_subset_weedy.txt\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Looks good!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00741759659045 and the largest is 3.42543681733.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0534624286624 and the largest is 3.60220386107.\n",
      "  RuntimeWarning\n",
      "/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00426119158986 and the largest is 2.52518339985.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0279330580674 and the largest is 3.08720378615.\n",
      "  RuntimeWarning\n",
      "/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0943930219986 and the largest is 7.68100544672.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.35954414476 and the largest is 14.1307473814.\n",
      "  RuntimeWarning\n",
      "/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0417159752951 and the largest is 5.5945087651.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0748657297364 and the largest is 2.68518614475.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on rarified (1000) subset of each sample type\n",
    "tissue_otu_table_dir = join(output_dir,'functional_groups')\n",
    "depth = 1000\n",
    "\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    print(curr_otu_table)\n",
    "    !beta_diversity_through_plots.py -i $curr_otu_table -o $curr_bdiv_dir -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]] 23    11.243 0.48881   1.113 0.32989 0.01499 *\n",
      "Residuals                       52    22.837 0.43917         0.67011          \n",
      "Total                           75    34.079                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12     8.065 0.67208  1.6276 0.23665 0.000999\n",
      "Residuals                       63    26.015 0.41293         0.76335         \n",
      "Total                           75    34.079                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12     8.065 0.67208  1.6276 0.23665 0.000999\n",
      "Residuals                       63    26.015 0.41293         0.76335         \n",
      "Total                           75    34.079                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]] 23    5.3402 0.23218   1.019 0.31068 0.4106\n",
      "Residuals                       52   11.8484 0.22785         0.68932       \n",
      "Total                           75   17.1886                 1.00000       \n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12    3.7498 0.31248  1.4649 0.21815 0.002997\n",
      "Residuals                       63   13.4389 0.21332         0.78185         \n",
      "Total                           75   17.1886                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12    3.7498 0.31248  1.4649 0.21815 0.000999\n",
      "Residuals                       63   13.4389 0.21332         0.78185         \n",
      "Total                           75   17.1886                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)\n",
      "qiime.data$map[[opts$category]] 23    8.3365 0.36246  0.9942 0.30543 0.5155\n",
      "Residuals                       52   18.9578 0.36457         0.69457       \n",
      "Total                           75   27.2943                 1.00000       \n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)   \n",
      "qiime.data$map[[opts$category]] 12    4.9295 0.41079  1.1572 0.1806 0.001998 **\n",
      "Residuals                       63   22.3649 0.35500         0.8194            \n",
      "Total                           75   27.2943                 1.0000            \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_competitive.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)   \n",
      "qiime.data$map[[opts$category]] 12    4.9295 0.41079  1.1572 0.1806 0.001998 **\n",
      "Residuals                       63   22.3649 0.35500         0.8194            \n",
      "Total                           75   27.2943                 1.0000            \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12    6.9065 0.57554  1.3728 0.32638 0.000999\n",
      "Residuals                       34   14.2541 0.41924         0.67362         \n",
      "Total                           46   21.1606                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    3.4111 0.68223  1.5759 0.1612 0.000999\n",
      "Residuals                       41   17.7494 0.43291         0.8388         \n",
      "Total                           46   21.1606                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    3.4111 0.68223  1.5759 0.1612 0.000999\n",
      "Residuals                       41   17.7494 0.43291         0.8388         \n",
      "Total                           46   21.1606                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12    5.2877 0.44064  1.7682 0.38427 0.000999\n",
      "Residuals                       34    8.4728 0.24920         0.61573         \n",
      "Total                           46   13.7605                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    2.2542 0.45085  1.6065 0.16382 0.005994\n",
      "Residuals                       41   11.5062 0.28064         0.83618         \n",
      "Total                           46   13.7605                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    2.2542 0.45085  1.6065 0.16382 0.003996\n",
      "Residuals                       41   11.5062 0.28064         0.83618         \n",
      "Total                           46   13.7605                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 12    4.7712 0.39760  1.1477 0.28829 0.003996\n",
      "Residuals                       34   11.7790 0.34644         0.71171         \n",
      "Total                           46   16.5501                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    2.2925 0.45849  1.3185 0.13852 0.000999\n",
      "Residuals                       41   14.2577 0.34775         0.86148         \n",
      "Total                           46   16.5501                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_generalist.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  5    2.2925 0.45849  1.3185 0.13852 0.000999\n",
      "Residuals                       41   14.2577 0.34775         0.86148         \n",
      "Total                           46   16.5501                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  62    34.758 0.56062   1.271 0.28572 0.000999\n",
      "Residuals                       197    86.893 0.44108         0.71428         \n",
      "Total                           259   121.651                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     16.63 0.87526  2.0002 0.1367 0.000999\n",
      "Residuals                       240    105.02 0.43759         0.8633         \n",
      "Total                           259    121.65                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model     R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19     16.63 0.87526  2.0002 0.1367 0.000999\n",
      "Residuals                       240    105.02 0.43759         0.8633         \n",
      "Total                           259    121.65                 1.0000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  62    21.846 0.35235  1.1784 0.27053 0.000999\n",
      "Residuals                       197    58.906 0.29901         0.72947         \n",
      "Total                           259    80.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    11.376 0.59874  2.0713 0.14088 0.000999\n",
      "Residuals                       240    69.375 0.28906         0.85912         \n",
      "Total                           259    80.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    11.376 0.59874  2.0713 0.14088 0.000999\n",
      "Residuals                       240    69.375 0.28906         0.85912         \n",
      "Total                           259    80.751                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  62    25.438 0.41028   1.118 0.26028 0.001998\n",
      "Residuals                       197    72.293 0.36697         0.73972         \n",
      "Total                           259    97.730                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    10.688 0.56251   1.551 0.10936 0.000999\n",
      "Residuals                       240    87.043 0.36268         0.89064         \n",
      "Total                           259    97.730                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_stress_tolerant.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  19    10.688 0.56251   1.551 0.10936 0.000999\n",
      "Residuals                       240    87.043 0.36268         0.89064         \n",
      "Total                           259    97.730                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 18    11.446 0.63588   1.814 0.40485 0.000999\n",
      "Residuals                       48    16.826 0.35054         0.59515         \n",
      "Total                           66    28.272                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7    6.6039 0.94342  2.5689 0.23359 0.000999\n",
      "Residuals                       59   21.6678 0.36725         0.76641         \n",
      "Total                           66   28.2717                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7    6.6039 0.94342  2.5689 0.23359 0.000999\n",
      "Residuals                       59   21.6678 0.36725         0.76641         \n",
      "Total                           66   28.2717                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2  Pr(>F)  \n",
      "qiime.data$map[[opts$category]] 18    4.8226 0.26792  1.3098 0.32938 0.01399 *\n",
      "Residuals                       48    9.8188 0.20456         0.67062          \n",
      "Total                           66   14.6414                 1.00000          \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7    2.7217 0.38882  1.9246 0.18589 0.001998\n",
      "Residuals                       59   11.9197 0.20203         0.81411         \n",
      "Total                           66   14.6414                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7    2.7217 0.38882  1.9246 0.18589 0.000999\n",
      "Residuals                       59   11.9197 0.20203         0.81411         \n",
      "Total                           66   14.6414                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]] 18    7.4096 0.41165  1.2081 0.31178 0.000999\n",
      "Residuals                       48   16.3558 0.34075         0.68822         \n",
      "Total                           66   23.7654                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7     3.713 0.53042  1.5607 0.15623 0.000999\n",
      "Residuals                       59    20.052 0.33987         0.84377         \n",
      "Total                           66    23.765                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_weedy.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  7     3.713 0.53042  1.5607 0.15623 0.000999\n",
      "Residuals                       59    20.052 0.33987         0.84377         \n",
      "Total                           66    23.765                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_prop_Colony_maximum_diameter_universal_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_latitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_longitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_prop_Colony_maximum_diameter_universal_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_latitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_longitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_prop_Colony_maximum_diameter_universal_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_latitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_competitive.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/compare_categories_longitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_prop_Colony_maximum_diameter_universal_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_latitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_longitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_prop_Colony_maximum_diameter_universal_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_latitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_longitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_prop_Colony_maximum_diameter_universal_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_latitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_generalist.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/compare_categories_longitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_prop_Colony_maximum_diameter_universal_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_latitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_longitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_prop_Colony_maximum_diameter_universal_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_latitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_longitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_prop_Colony_maximum_diameter_universal_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_latitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_stress_tolerant.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/compare_categories_longitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_prop_Colony_maximum_diameter_universal_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_latitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_longitude_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_prop_Colony_maximum_diameter_universal_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_latitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_longitude_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: prop_Colony_maximum_diameter_universal\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_prop_Colony_maximum_diameter_universal_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: latitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_latitude_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: longitude\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_weedy.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/compare_categories_longitude_unweighted_unifrac/adonis_results.txt: Not a directory\n"
     ]
    }
   ],
   "source": [
    "#Check for differences among mapping categories within a given functional group\n",
    "tissue_otu_table_dir = join(output_dir,\"functional_groups\")\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'prop_Colony_maximum_diameter_universal,latitude,longitude'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)   \n",
    "        for curr_cat in categories.split(\",\"):\n",
    "            print \"curr_cat: %s\" %curr_cat\n",
    "            print \"curr_metric: %s\" %metric\n",
    "            print \"curr_category: %s\" %c\n",
    "            curr_compare_cat_output_dir = join(curr_bdiv_dir,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "            !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "            the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "            !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries:\n",
    "tissue_otu_table_dir = join(output_dir,\"functional_groups\")\n",
    "n_permutations = 1000\n",
    "categories = 'prop_Colony_maximum_diameter_universal,latitude,longitude'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Use adonis to explore how much variability in coral microbiome is explained by compartment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Export Sceleractinian coral  specific biom file\n",
    "subset_output = join(output_dir,'coral_only')\n",
    "!filter_samples_from_otu_table.py -i $otu_table -o $subset_output/otu_table_coral_only.biom -m $mapping --sample_id_fp input/coral_only.txt --output_mapping_fp $subset_output/otu_table_coral_only_map.txt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Summarize BIOM to confirm filter functioned correctly\n",
    "subset_output = join(output_dir,'coral_only')\n",
    "!biom summarize-table -i $subset_output/otu_table_coral_only.biom -o $subset_output/otu_table_coral_only_summary.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Summaries look good."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.192225882455 and the largest is 17.2822267266.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -1.19476909769 and the largest is 22.2380806245.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on rarified (1000) subset of each sample type\n",
    "tissue_otu_table_dir = join(output_dir,\"coral_only\")\n",
    "depth = 1000\n",
    "\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith(\".biom\"):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'bdiv_%s' %c)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    !beta_diversity_through_plots.py -i $curr_otu_table -o $curr_bdiv_dir -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "curr_cat: BiologicalMatter\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_coral_only.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2    10.469  5.2344  11.443 0.03538 0.000999\n",
      "Residuals                       624   285.434  0.4574         0.96462         \n",
      "Total                           626   295.903                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_coral_only.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2    11.185  5.5925  20.442 0.06149 0.000999\n",
      "Residuals                       624   170.716  0.2736         0.93851         \n",
      "Total                           626   181.901                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_coral_only.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   2     7.248  3.6241  9.8508 0.03061 0.000999\n",
      "Residuals                       624   229.568  0.3679         0.96939         \n",
      "Total                           626   236.816                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_coral_only.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/compare_categories_BiologicalMatter_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_coral_only.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/compare_categories_BiologicalMatter_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: BiologicalMatter\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_coral_only.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/compare_categories_BiologicalMatter_unweighted_unifrac/adonis_results.txt: Not a directory\n"
     ]
    }
   ],
   "source": [
    "#Check for differences among mapping categories within a given functional group\n",
    "tissue_otu_table_dir = join(output_dir,\"coral_only\")\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'BiologicalMatter'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)   \n",
    "        for curr_cat in categories.split(\",\"):\n",
    "            print \"curr_cat: %s\" %curr_cat\n",
    "            print \"curr_metric: %s\" %metric\n",
    "            print \"curr_category: %s\" %c\n",
    "            curr_compare_cat_output_dir = join(curr_bdiv_dir,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "            !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "            the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "            !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Seasonal variability using only Lizard Island samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Subset Lizard Island samples from sceleractinian coral compartment specific biom files (i.e. coral t, m and s)\n",
    "subset_output = join(output_dir,'compartment_tables')\n",
    "lirs_output = join(output_dir,'lizard_island_tables')\n",
    "!filter_samples_from_otu_table.py -i $subset_output/otu_table_subset_mucus.biom -o $lirs_output/otu_table_subset_mucus_lirs.biom -m $mapping -s 'site_name:Great_Barrier_Reef_-_Lizard_Island_Sector'\n",
    "!filter_samples_from_otu_table.py -i $subset_output/otu_table_subset_skeleton.biom -o $lirs_output/otu_table_subset_skeleton_lirs.biom -m $mapping -s 'site_name:Great_Barrier_Reef_-_Lizard_Island_Sector'\n",
    "!filter_samples_from_otu_table.py -i $subset_output/otu_table_subset_tissue.biom -o $lirs_output/otu_table_subset_tissue_lirs.biom -m $mapping -s 'site_name:Great_Barrier_Reef_-_Lizard_Island_Sector'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0374272131787 and the largest is 5.62637812019.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0859660828597 and the largest is 4.58853552914.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0480181795419 and the largest is 3.32490705344.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.125021274305 and the largest is 6.2473739898.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2507: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0493280775174 and the largest is 3.7855402424.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.136379771221 and the largest is 4.68797513675.\n",
      "  RuntimeWarning\n"
     ]
    }
   ],
   "source": [
    "#Run beta diversity on rarified (1000) subset of each sample type from LIRS\n",
    "lirs_output = join(output_dir,'lizard_island_tables')\n",
    "depth = 1000\n",
    "\n",
    "for c in listdir(lirs_output):\n",
    "    if not c.endswith(\".biom\"):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(lirs_output,'bdiv_%s' %c)\n",
    "    curr_otu_table = join(lirs_output,c)\n",
    "    !beta_diversity_through_plots.py -i $curr_otu_table -o $curr_bdiv_dir -m $mapping -p $bc_prefs_fp -e $depth -f -t $tree_fp "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "curr_cat: expedition_number\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_mucus_lirs.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     3.538  3.5381  9.6083 0.08687 0.000999\n",
      "Residuals                       101    37.192  0.3682         0.91313         \n",
      "Total                           102    40.730                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: expedition_number\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus_lirs.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model    R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1    2.1328  2.1328  12.229 0.108 0.000999\n",
      "Residuals                       101   17.6149  0.1744         0.892         \n",
      "Total                           102   19.7477                 1.000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: expedition_number\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_mucus_lirs.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     2.072 2.07210  6.2837 0.05857 0.000999\n",
      "Residuals                       101    33.305 0.32975         0.94143         \n",
      "Total                           102    35.377                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: expedition_number\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_skeleton_lirs.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     0.963 0.96272  2.0743 0.02032 0.000999\n",
      "Residuals                       100    46.412 0.46412         0.97968         \n",
      "Total                           101    47.375                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: expedition_number\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton_lirs.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     1.035 1.03500  4.0374 0.03881 0.000999\n",
      "Residuals                       100    25.635 0.25635         0.96119         \n",
      "Total                           101    26.670                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: expedition_number\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_skeleton_lirs.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                 Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]   1     0.927 0.92694  2.4084 0.02352 0.001998\n",
      "Residuals                       100    38.488 0.38488         0.97648         \n",
      "Total                           101    39.415                 1.00000         \n",
      "                                  \n",
      "qiime.data$map[[opts$category]] **\n",
      "Residuals                         \n",
      "Total                             \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: expedition_number\n",
      "curr_metric: bray_curtis\n",
      "curr_category: bdiv_otu_table_subset_tissue_lirs.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1     1.255 1.25514  2.7441 0.02751 0.000999\n",
      "Residuals                       97    44.367 0.45739         0.97249         \n",
      "Total                           98    45.622                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: expedition_number\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue_lirs.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1    0.8452 0.84520  3.2937 0.03284 0.000999\n",
      "Residuals                       97   24.8910 0.25661         0.96716         \n",
      "Total                           98   25.7362                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: expedition_number\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: bdiv_otu_table_subset_tissue_lirs.biom\n",
      "\n",
      "Call:\n",
      "adonis(formula = as.dist(qiime.data$distmat) ~ qiime.data$map[[opts$category]],      permutations = opts$num_permutations) \n",
      "\n",
      "Permutation: free\n",
      "Number of permutations: 1000\n",
      "\n",
      "Terms added sequentially (first to last)\n",
      "\n",
      "                                Df SumsOfSqs MeanSqs F.Model      R2   Pr(>F)\n",
      "qiime.data$map[[opts$category]]  1     1.185 1.18494     3.2 0.03194 0.000999\n",
      "Residuals                       97    35.918 0.37029         0.96806         \n",
      "Total                           98    37.103                 1.00000         \n",
      "                                   \n",
      "qiime.data$map[[opts$category]] ***\n",
      "Residuals                          \n",
      "Total                              \n",
      "---\n",
      "Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1\n",
      "curr_cat: expedition_number\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_mucus_lirs.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_mucus_lirs.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_mucus_lirs.biom/compare_categories_expedition_number_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: expedition_number\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_mucus_lirs.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_mucus_lirs.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_mucus_lirs.biom/compare_categories_expedition_number_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: expedition_number\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_mucus_lirs.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_mucus_lirs.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_mucus_lirs.biom/compare_categories_expedition_number_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: expedition_number\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_skeleton_lirs.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_skeleton_lirs.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_skeleton_lirs.biom/compare_categories_expedition_number_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: expedition_number\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_skeleton_lirs.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_skeleton_lirs.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_skeleton_lirs.biom/compare_categories_expedition_number_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: expedition_number\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_skeleton_lirs.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_skeleton_lirs.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_skeleton_lirs.biom/compare_categories_expedition_number_unweighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: expedition_number\n",
      "curr_metric: bray_curtis\n",
      "curr_category: otu_table_subset_tissue_lirs.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_tissue_lirs.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_tissue_lirs.biom/compare_categories_expedition_number_bray_curtis/adonis_results.txt: Not a directory\n",
      "curr_cat: expedition_number\n",
      "curr_metric: weighted_unifrac\n",
      "curr_category: otu_table_subset_tissue_lirs.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_tissue_lirs.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_tissue_lirs.biom/compare_categories_expedition_number_weighted_unifrac/adonis_results.txt: Not a directory\n",
      "curr_cat: expedition_number\n",
      "curr_metric: unweighted_unifrac\n",
      "curr_category: otu_table_subset_tissue_lirs.biom\n",
      "Error in compare_categories.py: option -i: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_tissue_lirs.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "head: /Users/FJPollock/Dropbox/coral_microbiomes_postdoc/gcmp_osu_psu_shared_folder/coral_microbe_coevolution/Projects/australia_coevolution_paper/16S_analysis/3b_bdiv_australia_analysis/output/lizard_island_tables/otu_table_subset_tissue_lirs.biom/compare_categories_expedition_number_unweighted_unifrac/adonis_results.txt: Not a directory\n"
     ]
    }
   ],
   "source": [
    "#Check for differences among mapping categories within lizard island compartments\n",
    "lirs_output = join(output_dir,'lizard_island_tables')\n",
    "\n",
    "n_permutations = 1000\n",
    "categories = 'expedition_number'\n",
    "for c in listdir(lirs_output):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(lirs_output,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "    curr_otu_table = join(lirs_output,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)   \n",
    "        for curr_cat in categories.split(\",\"):\n",
    "            print \"curr_cat: %s\" %curr_cat\n",
    "            print \"curr_metric: %s\" %metric\n",
    "            print \"curr_category: %s\" %c\n",
    "            curr_compare_cat_output_dir = join(curr_bdiv_dir,\"compare_categories_%s_%s\" %(curr_cat,metric))\n",
    "            !compare_categories.py -i $curr_dm -o $curr_compare_cat_output_dir -m $mapping -c $curr_cat --method adonis -n $n_permutations\n",
    "            the_result = join(curr_compare_cat_output_dir,\"adonis_results.txt\")\n",
    "            !head -n 30 $the_result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Boxplot summaries:\n",
    "tissue_otu_table_dir = join(output_dir,\"coral_only\")\n",
    "n_permutations = 1000\n",
    "categories = 'BiologicalMatter'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Explore differences in traunches of latitude (latitude_traunch) and prop_Colony_maximum_diameter_universal (prop_Colony_maximum_diameter_universal_traunch)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Trauches added in \"input/gcmp16S_map_r20_lat_size_traunches.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/temp.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in all corals:\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"coral_only\")\n",
    "n_permutations = 1000\n",
    "categories = 'latitude_traunch,prop_Colony_maximum_diameter_universal_traunch'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in tissue compartment:\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"compartment_tables\")\n",
    "n_permutations = 1000\n",
    "categories = 'latitude_traunch,prop_Colony_maximum_diameter_universal_traunch'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in functional groups:\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"functional_groups\")\n",
    "n_permutations = 1000\n",
    "categories = 'latitude_traunch,prop_Colony_maximum_diameter_universal_traunch'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/coral_only/otu_table_coral_only.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in all corals for compartment differences:\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"coral_only\")\n",
    "n_permutations = 1000\n",
    "categories = 'BiologicalMatter'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/draw/_distributions.py:701: RuntimeWarning: Could not automatically resize plot to make room for axes labels and plot title. This can happen if the labels or title are extremely long and the plot size is too small. Your plot may have its labels and/or title cut-off. To fix this, try increasing the plot's size (in inches) and try again.\n",
      "  RuntimeWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_competitive.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_generalist.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_stress_tolerant.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Volumes/Macintosh_HDD/Dropbox/Coral_Microbiomes_Postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/functional_groups/otu_table_subset_weedy.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in brooders and spawners:\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"functional_groups\")\n",
    "n_permutations = 1000\n",
    "categories = 'latitude_traunch,prop_Colony_maximum_diameter_universal_traunch'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_all_within --suppress_all_between -n $n_permutations --sort median"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Explore differences in traunches of prop_Colony_maximum_diameter_universal (< 20% vs > 20% max size) w/in compartments..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Set file paths\n",
    "input_dir = abspath(\"input/\")\n",
    "otu_table = abspath(\"input/otu_table_mc2_wtax_no_pynast_failures_no_organelles.biom\")\n",
    "tree_fp = abspath(\"input/rep_set.tre\")\n",
    "mapping = abspath(\"input/gcmp16S_map_r20.txt\")\n",
    "output_dir = abspath(\"output/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_mucus.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_skeleton.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/weighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue.biom/unweighted_unifrac_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in make_distance_boxplots.py: option -d: file does not exist: '/Users/FJPollock/Dropbox/coral_microbiomes_postdoc/GCMP_OSU_PSU_Shared_Folder/Coral_microbe_coevolution/Projects/Australia_Coevolution_Paper/16S_analysis/3b_bdiv_australia_analysis/output/tissue_compartment_tables/otu_table_subset_tissue_1000.biom/bray_curtis_dm.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n"
     ]
    }
   ],
   "source": [
    "#Boxplot summaries w/in tissue compartment (with raw data save):\n",
    "mapping = abspath(\"input/gcmp16S_map_r20_lat_size_traunches.txt\")\n",
    "\n",
    "tissue_otu_table_dir = join(output_dir,\"compartment_tables\")\n",
    "n_permutations = 1000\n",
    "categories = 'latitude_traunch,prop_Colony_maximum_diameter_universal_traunch'\n",
    "for c in listdir(tissue_otu_table_dir):\n",
    "    if not c.endswith('.biom'):\n",
    "        continue\n",
    "    curr_bdiv_dir = join(tissue_otu_table_dir,'%s' %c)\n",
    "#    print listdir(curr_bdiv_dir)\n",
    "#    curr_otu_table = join(tissue_otu_table_dir,c)\n",
    "    for metric in ['bray_curtis','weighted_unifrac','unweighted_unifrac']:\n",
    "        curr_dm = join(curr_bdiv_dir,\"%s_dm.txt\" %metric)\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_all_within_%s\" %metric)\n",
    "#        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_individual_between --suppress_individual_within  -n $n_permutations --sort median\n",
    "        curr_boxplot_output_dir = join(curr_bdiv_dir,\"distance_boxplots_%s\" %metric)\n",
    "        !make_distance_boxplots.py -d $curr_dm -o $curr_boxplot_output_dir -m $mapping -f $categories --height 20 --suppress_all_within --suppress_all_between -n $n_permutations --sort median --save_raw_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# With continuous data (i.e. prop_Colony_maximum_diameter_universal,latitude,longitude) it might be better to bin before box plots"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Could explore any interesting differences deeper by looking within compartments of specific functional groups."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Num samples: 692\r\n",
      "Num observations: 129305\r\n",
      "Total count: 9441738\r\n",
      "Table density (fraction of non-zero values): 0.006\r\n",
      "\r\n",
      "Counts/sample summary:\r\n",
      " Min: 2.0\r\n",
      " Max: 38523.0\r\n",
      " Median: 14010.000\r\n",
      " Mean: 13644.130\r\n",
      " Std. dev.: 7564.457\r\n",
      " Sample Metadata Categories: None provided\r\n",
      " Observation Metadata Categories: taxonomy\r\n",
      "\r\n",
      "Counts/sample detail:\r\n",
      " E7.17.Pla.sine.1.20150622.M: 2.0\r\n",
      " E3.17.Ech.mamm.1.20150127.S: 3.0\r\n",
      " E10.17.Fav.pali.2.20150630.T: 5.0\r\n",
      " E7.17.Fav.spB.1.20150623.T: 7.0\r\n",
      " E3.17.Ech.horr.1.20150123.M: 8.0\r\n",
      " E3.15.Dip.heli.2.20150120.T: 9.0\r\n",
      " E10.Cyp.sera.1.20150628.M: 17.0\r\n",
      " E10.17.Fav.pali.1.20150630.T: 17.0\r\n",
      " E3.Out.Mil.sp.2.20150125.M: 24.0\r\n",
      " E10.17.Fav.pali.1.20150630.M: 36.0\r\n",
      " E10.17.Fav.pali.2.20150630.S: 41.0\r\n",
      " E10.Cyp.sera.2.20150628.T: 42.0\r\n",
      " E1.6.Acr.sp.1.20140726.T: 42.0\r\n",
      " E3.5.Gal.astr.2.20150127.T: 53.0\r\n",
      " E10.Ser.cali.2.20150627.T: 66.0\r\n",
      " E10.Alv.alli.1.20150628.S: 96.0\r\n",
      " E10.Ser.cali.1.20150627.T: 102.0\r\n",
      " E10.17.Fav.pali.2.20150630.M: 127.0\r\n",
      " E10.17.Fav.abdi.2.20150630.T: 165.0\r\n",
      " E7.17.Gon.edwa.2.20150623.S: 197.0\r\n",
      " E10.17.Fav.pali.3.20150630.T: 200.0\r\n",
      " E10.17.Fav.pali.3.20150630.S: 252.0\r\n",
      " E10.17.Fav.pali.3.20150630.M: 276.0\r\n",
      " E3.6.Acr.hyac.3.20150106.S: 470.0\r\n",
      " E10.Ser.cali.3.20150627.T: 522.0\r\n",
      " E10.10.Sty.pist.2.20150630.S: 566.0\r\n",
      " E3.6.Acr.hyac.1.20150120.T: 576.0\r\n",
      " E3.2.Tur.reni.2.20150124.S: 627.0\r\n",
      " E7.7.Pav.vari.1.20150622.T: 690.0\r\n",
      " E3.2.Tur.reni.2.20150124.T: 788.0\r\n",
      " E3.6.Acr.hyac.4.20150106.M: 941.0\r\n",
      " E10.Ser.cali.3.20150627.S: 1004.0\r\n",
      " E3.10.Sty.pist.2.20150124.S: 1005.0\r\n",
      " E1.6.Acr.sp.1.20140726.M: 1009.0\r\n",
      " E7.10.Poc.dami.2.20150621.T: 1033.0\r\n",
      " E7.10.Poc.dami.1.20150621.T: 1114.0\r\n",
      " E3.2.Tur.reni.1.20150124.S: 1138.0\r\n",
      " E1.10.Poc.dami.1.20140728.S: 1176.0\r\n",
      " E10.10.Sty.pist.3.20150630.T: 1256.0\r\n",
      " E10.10.Sty.pist.3.20150630.S: 1270.0\r\n",
      " E3.10.Sty.pist.1.20150124.S: 1358.0\r\n",
      " E10.Ser.cali.1.20150627.S: 1390.0\r\n",
      " E3.11.Fun.sp.2.20150125.S: 1415.0\r\n",
      " Negative.Control.2: 1442.0\r\n",
      " E7.17.Mer.ampl.3.20150621.S: 1468.0\r\n",
      " E3.10.Sty.pist.1.20150127.S: 1490.0\r\n",
      " E10.Ser.cali.2.20150627.S: 1547.0\r\n",
      " E3.15.Dip.heli.2.20150123.T: 1558.0\r\n",
      " E7.17.Cyp.micr.1.20150621.T: 1563.0\r\n",
      " E7.17.Mer.ampl.1.20150621.S: 1595.0\r\n",
      " E7.17.Mer.ampl.2.20150621.S: 1640.0\r\n",
      " E3.2.Tur.reni.1.20150124.T: 1673.0\r\n",
      " E3.17.Ech.mamm.1.20150129.T: 1678.0\r\n",
      " E3.10.Sty.pist.1.20150124.T: 1700.0\r\n",
      " E3.6.Acr.hyac.2.20150121.T: 1763.0\r\n",
      " Negative.Control.1: 1771.0\r\n",
      " E1.17.Ech.mamm.1.20140815.S: 1993.0\r\n",
      " E1.14.Phy.lich.1.20140801.M: 2027.0\r\n",
      " E3.10.Sty.pist.1.20150127.T: 2059.0\r\n",
      " E10.10.Sty.pist.1.20150630.T: 2109.0\r\n",
      " Negative.Control.3: 2120.0\r\n",
      " E1.2.Tur.reni.1.20140814.S: 2194.0\r\n",
      " E3.10.Sty.pist.2.20150124.T: 2232.0\r\n",
      " E7.10.Ser.cali.3.20150622.T: 2268.0\r\n",
      " E7.7.Pav.vari.1.20150622.S: 2309.0\r\n",
      " E3.15.Dip.heli.1.20150119.T: 2325.0\r\n",
      " E3.4.Pac.spec.1.20150128.S: 2332.0\r\n",
      " E3.14.Phy.lich.2.20150125.T: 2353.0\r\n",
      " E10.Por.hero.1.20150628.S: 2355.0\r\n",
      " Negative.Control.4: 2493.0\r\n",
      " E7.10.Poc.dami.3.20150621.T: 2544.0\r\n",
      " E1.14.Phy.lich.1.20140801.T: 2568.0\r\n",
      " E10.Cyp.micr.1.20150627.S: 2638.0\r\n",
      " E3.14.Phy.lich.1.20150125.T: 2679.0\r\n",
      " E3.15.Dip.heli.3.20150106.M: 2719.0\r\n",
      " E7.10.Ser.cali.2.20150622.S: 3296.0\r\n",
      " E1.17.Ech.mamm.1.20140728.T: 3523.0\r\n",
      " E10.Aca.hill.2.20150629.T: 3617.0\r\n",
      " E7.17.Fav.abdi.2.20150623.S: 3626.0\r\n",
      " E10.10.Sty.pist.1.20150630.S: 3678.0\r\n",
      " E7.10.Ser.cali.1.20150622.T: 3837.0\r\n",
      " E1.6.Mon.hisp.1.20140807.T: 3857.0\r\n",
      " E3.19.Sym.sp.2.20150125.T: 3902.0\r\n",
      " E1.4.Pac.spec.1.20140801.M: 3954.0\r\n",
      " E3.11.Fun.sp.1.20150127.S: 3963.0\r\n",
      " E10.10.Sty.pist.2.20150630.T: 3977.0\r\n",
      " E10.Aca.hill.2.20150629.S: 3979.0\r\n",
      " E7.3.Por.loba.2.20150619.S: 4098.0\r\n",
      " E3.15.Dip.heli.1.20150121.T: 4127.0\r\n",
      " E10.Poc.dami.1.20150628.S: 4135.0\r\n",
      " E1.10.Poc.dami.1.20140905.S: 4139.0\r\n",
      " E7.10.Poc.dami.1.20150621.S: 4148.0\r\n",
      " E7.10.Ser.cali.2.20150622.T: 4274.0\r\n",
      " E3.19.Sym.sp.1.20150127.T: 4368.0\r\n",
      " E3.6.Acr.hyac.1.20150105.T: 4432.0\r\n",
      " E10.Aca.hill.1.20150629.T: 4436.0\r\n",
      " E1.10.Poc.dami.1.20140905.T: 4444.0\r\n",
      " E10.Aca.hill.3.20150629.T: 4458.0\r\n",
      " E3.15.Dip.heli.1.20150120.T: 4509.0\r\n",
      " E1.10.Poc.dami.1.20140728.T: 4523.0\r\n",
      " E1.15.Dip.heli.1.20140726.M: 4568.0\r\n",
      " E10.17.Fav.abdi.1.20150630.S: 4588.0\r\n",
      " E3.17.Ech.horr.1.20150123.T: 4609.0\r\n",
      " E1.19.Sym.sp.1.20140724.T: 4623.0\r\n",
      " E3.15.Dip.heli.2.20150121.T: 4681.0\r\n",
      " E7.17.Gon.edwa.2.20150623.T: 4685.0\r\n",
      " E3.19.Sym.sp.1.20150125.T: 4712.0\r\n",
      " E7.6.Mon.aequ.2.20150620.S: 4727.0\r\n",
      " E3.15.Dip.heli.1.20150118.S: 4736.0\r\n",
      " E1.10.Sty.pist.1.20140815.S: 4810.0\r\n",
      " E3.11.Fun.sp.1.20150127.M: 4878.0\r\n",
      " E1.5.Gal.astr.1.20140807.T: 4889.0\r\n",
      " E10.Poc.dami.2.20150628.M: 4954.0\r\n",
      " E7.17.Fav.spC.1.20150623.T: 4958.0\r\n",
      " E1.7.Pav.vari.1.20140805.M: 4965.0\r\n",
      " E1.3.Por.loba.1.20140724.S: 5152.0\r\n",
      " E10.Ser.cali.2.20150627.M: 5153.0\r\n",
      " E3.11.Fun.sp.1.20150125.S: 5204.0\r\n",
      " E1.6.Acr.humi.1.20140731.T: 5281.0\r\n",
      " E1.4.Pac.spec.1.20140801.T: 5320.0\r\n",
      " E1.Out.Lob.sp.1.20140803.S: 5325.0\r\n",
      " E3.15.Dip.heli.1.20150123.T: 5652.0\r\n",
      " E3.6.Acr.hyac.3.20150120.M: 5696.0\r\n",
      " E7.3.Por.loba.3.20150619.S: 5724.0\r\n",
      " E7.10.Poc.dami.2.20150621.S: 5754.0\r\n",
      " E7.17.Fav.spC.1.20150623.M: 5793.0\r\n",
      " E1.6.Mon.hisp.1.20140807.M: 5835.0\r\n",
      " E1.3.Por.loba.1.20140724.T: 5878.0\r\n",
      " E3.7.Pav.vari.2.20150124.T: 5887.0\r\n",
      " E3.6.Acr.hyac.1.20150118.M: 6035.0\r\n",
      " E10.Cyp.sera.2.20150628.S: 6081.0\r\n",
      " E1.10.Sty.pist.1.20140807.S: 6086.0\r\n",
      " E7.17.Hyd.exes.1.20150622.S: 6132.0\r\n",
      " E1.17.Ech.mamm.1.20140728.S: 6196.0\r\n",
      " E1.19.Sym.sp.1.20140724.M: 6234.0\r\n",
      " E3.6.Acr.hyac.4.20150106.T: 6245.0\r\n",
      " E3.19.Sym.sp.1.20150127.S: 6346.0\r\n",
      " E7.3.Por.loba.1.20150619.S: 6360.0\r\n",
      " E3.14.Phy.lich.1.20150125.S: 6382.0\r\n",
      " E7.7.Pav.vari.1.20150623.S: 6384.0\r\n",
      " E1.3.Por.loba.1.20140724.M: 6450.0\r\n",
      " E7.10.Poc.dami.3.20150621.S: 6471.0\r\n",
      " E1.2.Tur.stel.1.20140805.M: 6489.0\r\n",
      " E3.15.Dip.heli.3.20150120.T: 6496.0\r\n",
      " E7.17.Ech.lame.1.20150620.S: 6515.0\r\n",
      " E3.17.Ech.mamm.1.20150127.T: 6562.0\r\n",
      " E3.7.Pav.vari.1.20150124.T: 6573.0\r\n",
      " E3.6.Acr.hyac.1.20150118.T: 6694.0\r\n",
      " E3.15.Dip.heli.1.20150119.S: 6721.0\r\n",
      " E7.7.Pav.vari.2.20150622.T: 6791.0\r\n",
      " E10.Poc.dami.1.20150628.T: 6834.0\r\n",
      " E7.17.Fav.spA.2.20150623.T: 6850.0\r\n",
      " E3.15.Dip.heli.1.20150105.T: 6878.0\r\n",
      " E10.Mon.curt.2.20150629.S: 6910.0\r\n",
      " E10.Poc.dami.2.20150628.T: 6942.0\r\n",
      " E3.15.Dip.heli.3.20150121.T: 7078.0\r\n",
      " E3.2.Tur.reni.1.20150127.T: 7111.0\r\n",
      " E3.15.Dip.heli.1.20150106.M: 7172.0\r\n",
      " E3.6.Acr.hyac.4.20150106.S: 7236.0\r\n",
      " E1.7.Pav.veno.1.20140811.M: 7261.0\r\n",
      " E1.4.Pac.rugo.1.20140728.M: 7284.0\r\n",
      " E10.17.Fav.abdi.2.20150630.S: 7421.0\r\n",
      " E3.6.Acr.hyac.3.20150106.T: 7437.0\r\n",
      " E1.4.Pac.spec.1.20140801.S: 7482.0\r\n",
      " E10.Aca.hill.3.20150629.S: 7517.0\r\n",
      " E3.5.Gal.astr.2.20150127.S: 7623.0\r\n",
      " E10.Por.hero.2.20150628.S: 7626.0\r\n",
      " E7.5.Gal.fasc.1.20150622.S: 7667.0\r\n",
      " E1.Out.Mil.sp.1.20140731.S: 7697.0\r\n",
      " E1.17.Ech.mamm.1.20140815.T: 7698.0\r\n",
      " E10.Fav.abdi.1.20150629.S: 7702.0\r\n",
      " E1.17.Cau.furc.1.20140801.S: 7706.0\r\n",
      " E7.6.Acr.hyac.3.20150622.S: 7720.0\r\n",
      " E1.Out.Mil.sp.1.20140731.M: 7725.0\r\n",
      " E1.7.Pav.vari.1.20140805.T: 7751.0\r\n",
      " E3.15.Dip.heli.1.20150118.T: 7804.0\r\n",
      " E3.7.Pav.vari.2.20150124.S: 7866.0\r\n",
      " E7.10.Ser.cali.1.20150622.S: 7932.0\r\n",
      " E3.14.Phy.lich.1.20150129.S: 7970.0\r\n",
      " E1.19.Sym.sp.1.20140814.M: 7998.0\r\n",
      " E1.6.Acr.sp.1.20140726.S: 8071.0\r\n",
      " E3.4.Pac.rugo.1.20150128.S: 8078.0\r\n",
      " E3.7.Pav.vari.1.20150127.T: 8078.0\r\n",
      " E1.4.Pac.rugo.1.20140728.S: 8081.0\r\n",
      " E1.19.Lob.sp.1.20140730.S: 8089.0\r\n",
      " E3.3.Por.cyli.2.20150123.T: 8118.0\r\n",
      " E3.6.Acr.hyac.2.20150120.S: 8179.0\r\n",
      " E1.Out.Lob.sp.1.20140803.T: 8310.0\r\n",
      " E10.10.Poc.dami.1.20150630.S: 8322.0\r\n",
      " E1.10.Sty.pist.1.20140807.T: 8333.0\r\n",
      " E3.6.Acr.hyac.2.20150120.T: 8446.0\r\n",
      " E10.Sty.pist.1.20150628.S: 8462.0\r\n",
      " E3.15.Dip.heli.2.20150105.M: 8662.0\r\n",
      " E3.10.Sty.pist.2.20150124.M: 8700.0\r\n",
      " E3.5.Gal.astr.1.20150124.M: 8700.0\r\n",
      " E3.15.Dip.heli.3.20150123.T: 8711.0\r\n",
      " E10.Aca.hill.1.20150629.S: 8733.0\r\n",
      " E7.17.Fav.abdi.1.20150623.T: 8825.0\r\n",
      " E3.4.Pac.rugo.1.20150127.T: 8844.0\r\n",
      " E1.11.Lep.tran.1.20140731.M: 8850.0\r\n",
      " E1.15.Dip.heli.1.20140809.M: 8871.0\r\n",
      " E7.17.Fav.spC.1.20150623.S: 8946.0\r\n",
      " E1.6.Acr.humi.1.20140731.S: 8959.0\r\n",
      " E3.15.Dip.heli.2.20150106.T: 9032.0\r\n",
      " E3.17.Ech.mamm.1.20150129.S: 9038.0\r\n",
      " E1.10.Poc.dami.1.20140728.M: 9100.0\r\n",
      " E1.7.Pav.vari.1.20140815.M: 9121.0\r\n",
      " E1.15.Dip.heli.1.20140726.T: 9140.0\r\n",
      " E3.6.Acr.hyac.3.20150121.T: 9149.0\r\n",
      " E1.7.Pav.veno.1.20140809.M: 9250.0\r\n",
      " E3.17.Ech.mamm.1.20150123.T: 9287.0\r\n",
      " E1.15.Dip.heli.1.20140815.M: 9297.0\r\n",
      " E3.4.Pac.spec.1.20150128.T: 9325.0\r\n",
      " E10.Acr.pali.3.20150627.S: 9469.0\r\n",
      " E7.10.Ser.cali.3.20150622.S: 9472.0\r\n",
      " E3.6.Acr.hyac.4.20150105.M: 9494.0\r\n",
      " E3.14.Phy.lich.1.20150129.T: 9510.0\r\n",
      " E7.17.Fav.abdi.2.20150623.T: 9523.0\r\n",
      " E7.17.Hyd.exes.2.20150622.S: 9524.0\r\n",
      " E1.10.Sty.pist.1.20140815.M: 9623.0\r\n",
      " E3.2.Tur.reni.1.20150127.S: 9661.0\r\n",
      " E10.17.Fav.pali.1.20150630.S: 9706.0\r\n",
      " E3.17.Ech.horr.1.20150123.S: 9716.0\r\n",
      " E3.10.Sty.pist.1.20150124.M: 9732.0\r\n",
      " E7.17.Pla.sine.1.20150622.S: 9760.0\r\n",
      " E3.Out.Mil.sp.1.20150125.M: 9796.0\r\n",
      " E7.17.Ech.lame.3.20150620.S: 9799.0\r\n",
      " E3.15.Dip.heli.3.20150106.T: 9846.0\r\n",
      " E3.6.Acr.hyac.1.20150120.S: 9941.0\r\n",
      " E3.3.Por.cyli.1.20150127.S: 9966.0\r\n",
      " E1.2.Tur.reni.1.20140730.M: 9999.0\r\n",
      " E7.3.Por.loba.3.20150619.T: 10007.0\r\n",
      " E1.5.Gal.astr.1.20140807.M: 10133.0\r\n",
      " E7.17.Fav.spA.1.20150623.T: 10201.0\r\n",
      " E10.17.Gon.favu.1.20150629.S: 10212.0\r\n",
      " E10.10.Sty.pist.3.20150630.M: 10212.0\r\n",
      " E3.11.Fun.sp.2.20150125.T: 10291.0\r\n",
      " E1.17.Ech.mamm.1.20140815.M: 10295.0\r\n",
      " E3.15.Dip.heli.2.20150106.M: 10296.0\r\n",
      " E7.6.Acr.hyac.1.20150622.T: 10332.0\r\n",
      " E1.17.Fav.sp.1.20140805.M: 10355.0\r\n",
      " E1.11.Psa.cont.1.20140730.M: 10387.0\r\n",
      " E3.6.Acr.hyac.1.20150121.S: 10419.0\r\n",
      " E7.6.Mon.aequ.2.20150620.T: 10431.0\r\n",
      " E3.14.Phy.lich.2.20150125.S: 10481.0\r\n",
      " E3.6.Acr.hyac.3.20150120.S: 10502.0\r\n",
      " E10.Por.hero.3.20150628.S: 10512.0\r\n",
      " E10.Acr.pali.2.20150627.S: 10513.0\r\n",
      " E3.4.Pac.rugo.1.20150127.S: 10549.0\r\n",
      " E1.17.Ech.mamm.1.20140728.M: 10591.0\r\n",
      " E3.15.Dip.heli.2.20150119.T: 10595.0\r\n",
      " E3.6.Acr.hyac.3.20150121.M: 10706.0\r\n",
      " E1.7.Pav.sp.1.20140905.T: 10731.0\r\n",
      " E3.14.Phy.lich.1.20150129.M: 10734.0\r\n",
      " E3.7.Pav.vari.1.20150124.S: 10839.0\r\n",
      " E3.7.Pav.vari.1.20150127.S: 10953.0\r\n",
      " E3.4.Pac.spec.1.20150129.T: 10967.0\r\n",
      " E1.Out.Hel.coer.1.20140811.M: 10974.0\r\n",
      " E10.Ser.cali.3.20150627.M: 11005.0\r\n",
      " E1.5.Gal.astr.1.20140728.M: 11074.0\r\n",
      " E10.Sty.pist.1.20150628.M: 11224.0\r\n",
      " E1.2.Tur.stel.1.20140805.T: 11275.0\r\n",
      " E3.15.Dip.heli.4.20150106.T: 11328.0\r\n",
      " E1.11.Lep.tran.1.20140731.T: 11360.0\r\n",
      " E3.15.Dip.heli.1.20150118.M: 11439.0\r\n",
      " E1.3.Por.loba.1.20140726.T: 11460.0\r\n",
      " E1.6.Acr.humi.1.20140731.M: 11485.0\r\n",
      " E1.15.Dip.heli.1.20140726.S: 11540.0\r\n",
      " E7.Sed.1.20150621: 11605.0\r\n",
      " E7.17.Mer.ampl.1.20150621.T: 11605.0\r\n",
      " E1.4.Pac.rugo.1.20140728.T: 11640.0\r\n",
      " E1.17.Hyd.exes.1.20140803.M: 11669.0\r\n",
      " E1.2.Tur.reni.1.20140814.M: 11675.0\r\n",
      " E1.7.Pav.veno.1.20140809.T: 11682.0\r\n",
      " E3.6.Acr.hyac.1.20150106.S: 11700.0\r\n",
      " E3.Out.Pal.caes.1.20150125.S: 11757.0\r\n",
      " E1.17.Fav.sp.2.20140805.M: 11768.0\r\n",
      " E1.5.Gal.astr.1.20140807.S: 11801.0\r\n",
      " E10.Poc.dami.2.20150628.S: 11816.0\r\n",
      " E10.Por.hero.3.20150628.T: 11850.0\r\n",
      " E1.10.Poc.dami.1.20140731.M: 11857.0\r\n",
      " E3.17.Ech.mamm.1.20150123.S: 11859.0\r\n",
      " E7.19.Lob.hemp.2.20150619.T: 11937.0\r\n",
      " E1.11.Lep.tran.1.20140731.S: 11958.0\r\n",
      " E7.Sed.1.20150623: 12037.0\r\n",
      " E1.Out.Pal.caes.1.20140809.M: 12073.0\r\n",
      " E7.6.Acr.hyac.3.20150622.T: 12131.0\r\n",
      " E1.15.Dip.heli.1.20140905.M: 12132.0\r\n",
      " E7.17.Pla.sine.1.20150622.T: 12177.0\r\n",
      " E7.17.Myc.elep.1.20150622.S: 12216.0\r\n",
      " E1.6.Acr.sp.1.20140905.S: 12219.0\r\n",
      " E10.Alv.alli.2.20150628.S: 12252.0\r\n",
      " E1.11.Fun.sp.1.20140730.T: 12254.0\r\n",
      " E1.11.Fun.sp.1.20140730.M: 12261.0\r\n",
      " E10.17.Gon.favu.1.20150629.T: 12292.0\r\n",
      " E1.10.Poc.dami.1.20140731.T: 12309.0\r\n",
      " E1.19.Lob.sp.1.20140730.T: 12327.0\r\n",
      " E10.Acr.cyth.1.20150627.T: 12366.0\r\n",
      " E1.5.Gal.astr.1.20140728.S: 12471.0\r\n",
      " E1.Out.Lob.sp.1.20140803.M: 12475.0\r\n",
      " E1.17.Cau.furc.1.20140801.T: 12518.0\r\n",
      " E10.Sed.1.20150630: 12638.0\r\n",
      " E1.3.Por.loba.1.20140726.M: 12685.0\r\n",
      " E1.7.Pav.vari.1.20140815.S: 12686.0\r\n",
      " E3.6.Acr.hyac.1.20150118.S: 12699.0\r\n",
      " E10.17.Mon.curt.3.20150629.S: 12727.0\r\n",
      " E7.17.Fav.spA.1.20150623.M: 12747.0\r\n",
      " E1.2.Tur.pelt.1.20140814.T: 12762.0\r\n",
      " E7.5.Gal.fasc.1.20150622.T: 12834.0\r\n",
      " E3.Out.Pal.caes.1.20150125.T: 12848.0\r\n",
      " E3.15.Dip.heli.1.20150120.M: 12866.0\r\n",
      " E3.15.Dip.heli.1.20150106.T: 12870.0\r\n",
      " E7.17.Gon.edwa.1.20150623.S: 13000.0\r\n",
      " E7.17.Fav.spA.2.20150623.M: 13033.0\r\n",
      " E1.19.Sym.sp.1.20140905.M: 13096.0\r\n",
      " E3.6.Acr.hyac.1.20150123.S: 13103.0\r\n",
      " E7.6.Mon.aequ.4.20150620.T: 13151.0\r\n",
      " E1.17.Fav.sp.2.20140805.T: 13158.0\r\n",
      " E7.17.Pla.sine.2.20150622.T: 13281.0\r\n",
      " E10.17.Fav.abdi.1.20150630.T: 13351.0\r\n",
      " E1.17.Cau.furc.1.20140801.DS: 13371.0\r\n",
      " E10.Alv.alli.1.20150628.T: 13379.0\r\n",
      " E10.10.Sty.pist.2.20150630.M: 13456.0\r\n",
      " E3.3.Por.cyli.1.20150123.S: 13460.0\r\n",
      " E7.7.Pav.vari.2.20150622.S: 13468.0\r\n",
      " E3.5.Gal.astr.1.20150124.S: 13520.0\r\n",
      " E1.2.Tur.pelt.1.20140814.M: 13525.0\r\n",
      " E1.3.Por.loba.1.20140905.T: 13564.0\r\n",
      " E3.3.Por.cyli.1.20150123.T: 13625.0\r\n",
      " E1.10.Poc.dami.1.20140731.S: 13668.0\r\n",
      " E3.10.Sty.pist.1.20150127.M: 13701.0\r\n",
      " E1.19.Sym.sp.1.20140814.S: 13719.0\r\n",
      " E1.19.Lob.sp.1.20140730.M: 13733.0\r\n",
      " E3.Out.Pal.caes.2.20150125.T: 13758.0\r\n",
      " E10.Alv.alli.2.20150628.T: 13766.0\r\n",
      " E1.Out.Pal.caes.1.20140809.T: 13770.0\r\n",
      " E1.2.Tur.reni.1.20140730.T: 13771.0\r\n",
      " E3.3.Por.cyli.2.20150123.S: 13787.0\r\n",
      " E3.6.Acr.hyac.3.20150121.S: 13804.0\r\n",
      " E1.19.Sym.sp.1.20140814.T: 13818.0\r\n",
      " E1.2.Tur.reni.1.20140730.S: 13874.0\r\n",
      " E10.Alv.alli.3.20150628.S: 13879.0\r\n",
      " E1.3.Por.loba.1.20140905.M: 13894.0\r\n",
      " E3.11.Fun.sp.1.20150127.T: 13942.0\r\n",
      " E1.15.Dip.heli.1.20140815.S: 13966.0\r\n",
      " E7.17.Pla.sine.3.20150622.T: 14004.0\r\n",
      " E1.4.Pac.spec.1.20140811.T: 14016.0\r\n",
      " E1.Out.Pal.caes.1.20140809.S: 14058.0\r\n",
      " E1.4.Pac.spec.1.20140811.M: 14104.0\r\n",
      " E1.19.Sym.sp.1.20140905.S: 14166.0\r\n",
      " E1.3.Por.cyli.1.20140815.T: 14202.0\r\n",
      " E1.14.Phy.lich.1.20140801.S: 14225.0\r\n",
      " E7.3.Por.loba.4.20150619.S: 14240.0\r\n",
      " E7.19.Lob.hemp.2.20150619.S: 14261.0\r\n",
      " E7.Sed.1.20150622: 14269.0\r\n",
      " E7.17.Cyp.micr.2.20150621.S: 14282.0\r\n",
      " E10.Alv.alli.3.20150628.M: 14382.0\r\n",
      " E3.14.Phy.lich.2.20150125.M: 14440.0\r\n",
      " E1.14.Phy.lich.1.20140811.M: 14481.0\r\n",
      " E1.10.Sty.pist.1.20140807.M: 14553.0\r\n",
      " E1.17.Fav.sp.1.20140805.T: 14566.0\r\n",
      " E3.6.Acr.hyac.3.20150123.S: 14573.0\r\n",
      " E3.4.Pac.rugo.1.20150128.T: 14621.0\r\n",
      " E7.6.Acr.hyac.2.20150622.S: 14626.0\r\n",
      " E3.15.Dip.heli.4.20150105.T: 14639.0\r\n",
      " E7.19.Lob.hemp.1.20150619.T: 14654.0\r\n",
      " E7.17.Cyp.micr.3.20150621.S: 14686.0\r\n",
      " E7.17.Ech.lame.3.20150620.T: 14704.0\r\n",
      " E7.7.Pav.vari.1.20150623.M: 14704.0\r\n",
      " E7.17.Fav.spB.1.20150623.S: 14721.0\r\n",
      " E3.19.Sym.sp.1.20150125.S: 14724.0\r\n",
      " E7.Sed.1.20150620: 14765.0\r\n",
      " E1.5.Gal.astr.1.20140728.T: 14799.0\r\n",
      " E3.6.Acr.hyac.1.20150121.T: 14804.0\r\n",
      " E1.3.Por.loba.1.20140726.S: 14856.0\r\n",
      " E1.7.Pav.vari.1.20140805.S: 14858.0\r\n",
      " E10.Por.hero.1.20150628.T: 14863.0\r\n",
      " E3.15.Dip.heli.2.20150119.M: 14877.0\r\n",
      " E7.3.Por.loba.1.20150619.T: 14958.0\r\n",
      " E3.6.Acr.hyac.3.20150120.T: 14973.0\r\n",
      " E7.3.Por.loba.4.20150619.T: 14995.0\r\n",
      " E7.Sed.1.20150619: 15134.0\r\n",
      " E10.Aca.hill.3.20150629.M: 15140.0\r\n",
      " E3.6.Acr.hyac.2.20150105.T: 15221.0\r\n",
      " E10.17.Gon.favu.2.20150630.S: 15284.0\r\n",
      " E10.17.Gon.favu.1.20150630.M: 15322.0\r\n",
      " E10.10.Sty.pist.1.20150630.M: 15352.0\r\n",
      " E1.4.Pac.rugo.1.20140814.M: 15371.0\r\n",
      " E7.7.Pav.vari.1.20150623.T: 15420.0\r\n",
      " E1.7.Pav.veno.1.20140811.T: 15431.0\r\n",
      " E7.17.Hyd.exce.1.20150620.S: 15457.0\r\n",
      " E3.6.Acr.hyac.1.20150121.M: 15535.0\r\n",
      " E3.Out.Pal.caes.2.20150125.S: 15618.0\r\n",
      " E3.4.Pac.spec.1.20150128.M: 15656.0\r\n",
      " E3.15.Dip.heli.2.20150120.S: 15660.0\r\n",
      " E7.10.Poc.dami.1.20150621.M: 15705.0\r\n",
      " E3.6.Acr.hyac.2.20150121.S: 15725.0\r\n",
      " E10.17.Mon.curt.3.20150629.M: 15728.0\r\n",
      " E10.Acr.pali.2.20150627.T: 15797.0\r\n",
      " E7.6.Mon.aequ.1.20150620.S: 15819.0\r\n",
      " E3.15.Dip.heli.2.20150105.T: 15879.0\r\n",
      " E1.6.Mon.hisp.1.20140807.S: 15883.0\r\n",
      " E1.19.Sym.sp.1.20140815.T: 15883.0\r\n",
      " E7.6.Mon.aequ.4.20150620.S: 15948.0\r\n",
      " E10.Por.hero.3.20150628.M: 15976.0\r\n",
      " E7.7.Pav.vari.2.20150622.M: 15983.0\r\n",
      " E3.6.Acr.hyac.3.20150123.T: 15998.0\r\n",
      " E10.Cyp.sera.2.20150628.M: 16001.0\r\n",
      " E7.17.Myc.elep.1.20150622.T: 16042.0\r\n",
      " E7.6.Acr.hyac.1.20150622.S: 16048.0\r\n",
      " E1.7.Pav.sp.1.20140905.M: 16110.0\r\n",
      " E10.Acr.cyth.3.20150627.T: 16112.0\r\n",
      " E7.5.Gal.fasc.1.20150621.S: 16132.0\r\n",
      " E3.11.Fun.sp.1.20150125.T: 16167.0\r\n",
      " E1.2.Tur.pelt.1.20140814.S: 16191.0\r\n",
      " E3.5.Gal.astr.2.20150124.T: 16244.0\r\n",
      " E1.7.Pav.vari.1.20140815.T: 16254.0\r\n",
      " E7.17.Mer.ampl.2.20150621.T: 16316.0\r\n",
      " E1.19.Sym.sp.1.20140905.T: 16446.0\r\n",
      " E1.6.Iso.pali.1.20140803.M: 16469.0\r\n",
      " E1.17.Hyd.exes.1.20140803.S: 16484.0\r\n",
      " E10.Por.hero.2.20150628.T: 16494.0\r\n",
      " E3.6.Acr.hyac.4.20150105.S: 16508.0\r\n",
      " E1.17.Fav.sp.1.20140805.S: 16542.0\r\n",
      " E3.6.Acr.hyac.2.20150106.M: 16605.0\r\n",
      " E1.3.Por.cyli.1.20140801.S: 16629.0\r\n",
      " E7.17.Ech.lame.2.20150620.T: 16650.0\r\n",
      " E10.Fav.abdi.1.20150629.T: 16677.0\r\n",
      " E3.15.Dip.heli.2.20150123.M: 16697.0\r\n",
      " E3.15.Dip.heli.2.20150119.S: 16707.0\r\n",
      " E3.3.Por.cyli.1.20150123.M: 16743.0\r\n",
      " E10.Sty.pist.1.20150628.T: 16760.0\r\n",
      " E10.10.Poc.dami.1.20150630.T: 16791.0\r\n",
      " E1.7.Pav.veno.1.20140811.S: 16884.0\r\n",
      " E3.17.Ech.mamm.1.20150129.M: 16913.0\r\n",
      " E1.11.Psa.cont.1.20140730.S: 16932.0\r\n",
      " E1.17.Ech.lame.1.20140905.T: 16973.0\r\n",
      " E7.17.Fav.abdi.1.20150623.M: 16982.0\r\n",
      " E1.5.Gal.astr.1.20140814.S: 16996.0\r\n",
      " E3.15.Dip.heli.2.20150121.M: 17017.0\r\n",
      " E7.19.Lob.hemp.1.20150621.T: 17030.0\r\n",
      " E10.17.Gon.favu.1.20150630.T: 17038.0\r\n",
      " E3.15.Dip.heli.3.20150105.T: 17105.0\r\n",
      " E10.Water.1.20150629: 17107.0\r\n",
      " E10.Mon.curt.1.20150629.S: 17126.0\r\n",
      " E3.4.Pac.spec.1.20150129.S: 17135.0\r\n",
      " E7.3.Por.loba.3.20150619.M: 17158.0\r\n",
      " E7.17.Pla.sine.2.20150622.S: 17182.0\r\n",
      " E1.17.Cau.furc.1.20140801.M: 17185.0\r\n",
      " E1.17.Fav.sp.1.20140811.S: 17261.0\r\n",
      " E3.6.Acr.hyac.2.20150105.S: 17279.0\r\n",
      " E3.3.Por.cyli.1.20150127.T: 17409.0\r\n",
      " E10.17.Gon.favu.2.20150630.T: 17422.0\r\n",
      " E7.7.Pav.vari.1.20150622.M: 17476.0\r\n",
      " E3.7.Pav.vari.1.20150124.M: 17487.0\r\n",
      " E1.2.Tur.reni.1.20140814.T: 17488.0\r\n",
      " E7.6.Mon.aequ.1.20150620.T: 17498.0\r\n",
      " E7.19.Lob.hemp.1.20150619.S: 17500.0\r\n",
      " E3.6.Acr.hyac.2.20150121.M: 17505.0\r\n",
      " E1.11.Psa.cont.1.20140730.T: 17506.0\r\n",
      " E10.Poc.dami.1.20150628.M: 17559.0\r\n",
      " E1.5.Gal.astr.1.20140814.T: 17574.0\r\n",
      " E3.6.Acr.hyac.4.20150105.T: 17590.0\r\n",
      " E10.Acr.pali.2.20150627.M: 17591.0\r\n",
      " E1.6.Acr.sp.1.20140905.M: 17681.0\r\n",
      " E7.6.Acr.hyac.3.20150622.M: 17687.0\r\n",
      " E1.3.Por.cyli.1.20140815.M: 17691.0\r\n",
      " E10.Acr.pali.3.20150627.T: 17708.0\r\n",
      " E10.Water.2.20150628: 17728.0\r\n",
      " E3.4.Pac.rugo.1.20150128.M: 17770.0\r\n",
      " E3.6.Acr.hyac.3.20150105.T: 17793.0\r\n",
      " E3.15.Dip.heli.2.20150105.S: 17815.0\r\n",
      " E10.Aca.hill.1.20150629.M: 17816.0\r\n",
      " E3.4.Pac.spec.1.20150129.M: 17829.0\r\n",
      " E1.6.Iso.pali.1.20140803.T: 17851.0\r\n",
      " E3.3.Por.cyli.1.20150127.M: 17871.0\r\n",
      " E10.Sed.1.20150628: 17898.0\r\n",
      " E7.3.Por.loba.2.20150619.T: 17986.0\r\n",
      " E7.17.Ech.lame.1.20150620.T: 18042.0\r\n",
      " E3.15.Dip.heli.3.20150105.S: 18042.0\r\n",
      " E10.Por.hero.2.20150628.M: 18085.0\r\n",
      " E7.17.Myc.elep.1.20150622.M: 18110.0\r\n",
      " E1.2.Tur.stel.1.20140805.S: 18125.0\r\n",
      " E3.15.Dip.heli.1.20150121.S: 18126.0\r\n",
      " E3.11.Fun.sp.2.20150125.M: 18141.0\r\n",
      " E3.15.Dip.heli.1.20150123.M: 18160.0\r\n",
      " E3.15.Dip.heli.3.20150106.S: 18171.0\r\n",
      " E3.4.Pac.rugo.1.20150127.M: 18194.0\r\n",
      " E1.7.Pav.veno.1.20140809.S: 18197.0\r\n",
      " E3.2.Tur.reni.2.20150124.M: 18213.0\r\n",
      " E1.17.Fav.sp.1.20140811.M: 18228.0\r\n",
      " E1.6.Iso.pali.1.20140815.S: 18233.0\r\n",
      " E7.17.Gon.edwa.1.20150623.T: 18255.0\r\n",
      " E1.15.Dip.heli.1.20140815.T: 18283.0\r\n",
      " E1.19.Sym.sp.1.20140815.M: 18286.0\r\n",
      " E3.15.Dip.heli.3.20150120.M: 18290.0\r\n",
      " E1.17.Hyd.exes.1.20140803.T: 18291.0\r\n",
      " E3.19.Sym.sp.1.20150127.M: 18350.0\r\n",
      " E7.17.Hyd.exes.1.20150622.T: 18385.0\r\n",
      " E1.19.Sym.sp.1.20140724.S: 18388.0\r\n",
      " E1.3.Por.cyli.1.20140801.M: 18430.0\r\n",
      " E1.6.Iso.pali.1.20140815.T: 18495.0\r\n",
      " E7.17.Gon.edwa.1.20150623.M: 18509.0\r\n",
      " E1.Out.Mil.sp.1.20140814.M: 18582.0\r\n",
      " E10.Alv.alli.3.20150628.T: 18588.0\r\n",
      " E7.17.Pla.sine.2.20150622.M: 18614.0\r\n",
      " E1.6.Iso.pali.1.20140803.S.2: 18622.0\r\n",
      " E3.15.Dip.heli.1.20150119.M: 18647.0\r\n",
      " E3.5.Gal.astr.1.20150124.T: 18651.0\r\n",
      " E1.15.Dip.heli.1.20140905.T: 18664.0\r\n",
      " E3.2.Tur.reni.1.20150124.M: 18704.0\r\n",
      " E3.7.Pav.vari.1.20150127.M: 18711.0\r\n",
      " E10.Mon.curt.1.20150629.T: 18712.0\r\n",
      " E3.6.Acr.hyac.1.20150123.T: 18730.0\r\n",
      " E7.19.Lob.hemp.2.20150619.M: 18732.0\r\n",
      " E1.5.Gal.astr.1.20140905.T: 18738.0\r\n",
      " E1.17.Ech.lame.1.20140905.S: 18768.0\r\n",
      " E1.4.Pac.spec.1.20140811.S: 18807.0\r\n",
      " E3.15.Dip.heli.3.20150121.M: 18820.0\r\n",
      " E7.10.Poc.dami.2.20150621.M: 18844.0\r\n",
      " E1.3.Por.cyli.1.20140801.T: 18858.0\r\n",
      " E3.6.Acr.hyac.3.20150105.M: 18872.0\r\n",
      " E7.17.Fav.abdi.1.20150623.S: 18892.0\r\n",
      " E7.6.Mon.aequ.3.20150620.T: 18900.0\r\n",
      " E3.14.Phy.lich.1.20150125.M: 19007.0\r\n",
      " E1.19.Lob.sp.1.20140809.S: 19114.0\r\n",
      " E3.6.Acr.hyac.2.20150123.T: 19142.0\r\n",
      " E3.6.Acr.hyac.2.20150106.S: 19219.0\r\n",
      " E3.6.Acr.hyac.1.20150106.T: 19265.0\r\n",
      " E7.19.Lob.hemp.1.20150621.S: 19320.0\r\n",
      " E7.17.Pla.sine.3.20150622.S: 19321.0\r\n",
      " E7.5.Gal.fasc.1.20150622.M: 19420.0\r\n",
      " E1.17.Fav.sp.1.20140811.T: 19436.0\r\n",
      " E1.5.Gal.astr.1.20140905.S: 19441.0\r\n",
      " E1.17.Fav.sp.2.20140805.S: 19459.0\r\n",
      " E3.6.Acr.hyac.2.20150123.S: 19493.0\r\n",
      " E7.17.Mer.ampl.3.20150621.T: 19502.0\r\n",
      " E3.15.Dip.heli.4.20150105.S: 19552.0\r\n",
      " E10.Mon.curt.1.20150629.M: 19590.0\r\n",
      " E10.10.Poc.dami.1.20150630.M: 19604.0\r\n",
      " E1.15.Dip.heli.1.20140809.S: 19656.0\r\n",
      " E7.17.Hyd.exes.2.20150622.M: 19656.0\r\n",
      " E3.15.Dip.heli.1.20150105.S: 19728.0\r\n",
      " E7.17.Hyd.exce.1.20150620.M: 19779.0\r\n",
      " E7.17.Hyd.exce.1.20150620.T: 19804.0\r\n",
      " E7.17.Ech.lame.2.20150620.M: 19849.0\r\n",
      " E7.17.Cyp.micr.3.20150621.M: 19867.0\r\n",
      " E1.3.Por.cyli.1.20140815.S: 19883.0\r\n",
      " E1.Out.Mil.sp.1.20140814.S: 19950.0\r\n",
      " E1.19.Lob.sp.1.20140809.M: 20084.0\r\n",
      " E3.15.Dip.heli.1.20150123.S: 20096.0\r\n",
      " E3.2.Tur.reni.1.20150127.M: 20125.0\r\n",
      " E1.6.Acr.sp.1.20140905.T: 20168.0\r\n",
      " E1.Out.Hel.coer.1.20140811.T: 20222.0\r\n",
      " E7.17.Fav.spA.2.20150623.S: 20268.0\r\n",
      " E10.17.Mon.curt.3.20150629.T: 20356.0\r\n",
      " E10.17.Gon.favu.1.20150630.S: 20382.0\r\n",
      " E3.6.Acr.hyac.1.20150105.S: 20410.0\r\n",
      " E3.19.Sym.sp.1.20150125.M: 20425.0\r\n",
      " E7.19.Lob.hemp.1.20150619.M: 20461.0\r\n",
      " E1.7.Pav.sp.1.20140905.S: 20517.0\r\n",
      " E7.3.Por.loba.2.20150619.M: 20540.0\r\n",
      " E1.6.Iso.pali.1.20140803.S: 20546.0\r\n",
      " E3.Out.Pal.caes.1.20150125.M: 20587.0\r\n",
      " E1.5.Gal.astr.1.20140814.M: 20640.0\r\n",
      " E7.5.Gal.astr.1.20150622.S: 20676.0\r\n",
      " E1.14.Phy.lich.1.20140811.T: 20731.0\r\n",
      " E1.17.Ech.lame.1.20140905.M: 20731.0\r\n",
      " E10.Cyp.micr.1.20150627.T: 20749.0\r\n",
      " E7.6.Mon.aequ.3.20150620.S: 20798.0\r\n",
      " E7.17.Ech.lame.3.20150620.M: 20835.0\r\n",
      " E1.11.Psa.cont.1.20140730.TS: 20837.0\r\n",
      " E1.Out.Pal.caes.1.20140815.T: 20853.0\r\n",
      " E7.5.Gal.fasc.1.20150621.M: 20962.0\r\n",
      " E3.15.Dip.heli.3.20150120.S: 21088.0\r\n",
      " E10.17.Gon.favu.2.20150630.M: 21147.0\r\n",
      " E10.Acr.cyth.3.20150627.S: 21149.0\r\n",
      " E1.6.Iso.pali.1.20140815.M: 21159.0\r\n",
      " E3.15.Dip.heli.4.20150105.M: 21188.0\r\n",
      " E7.17.Ech.lame.2.20150620.S: 21203.0\r\n",
      " E1.11.Psa.digi.1.20140807.M: 21342.0\r\n",
      " E3.6.Acr.hyac.1.20150123.M: 21371.0\r\n",
      " E3.7.Pav.vari.2.20150124.M: 21392.0\r\n",
      " E3.5.Gal.astr.2.20150127.M: 21434.0\r\n",
      " E3.15.Dip.heli.4.20150106.S: 21561.0\r\n",
      " E1.15.Dip.heli.1.20140905.S: 21633.0\r\n",
      " E7.17.Fav.spA.1.20150623.S: 21653.0\r\n",
      " E1.10.Sty.pist.1.20140815.T: 21705.0\r\n",
      " E1.4.Pac.rugo.1.20140814.T: 21754.0\r\n",
      " E10.Mon.curt.2.20150629.T: 21799.0\r\n",
      " E3.5.Gal.astr.2.20150124.M: 21831.0\r\n",
      " E7.17.Cyp.micr.1.20150621.S: 21929.0\r\n",
      " E1.19.Lob.sp.1.20140809.T: 21933.0\r\n",
      " E1.Out.Mil.sp.1.20140731.T: 21952.0\r\n",
      " E3.19.Sym.sp.2.20150125.S: 21981.0\r\n",
      " E1.6.Iso.pali.1.20140803.M.2: 21994.0\r\n",
      " E3.15.Dip.heli.1.20150106.S: 22035.0\r\n",
      " E3.15.Dip.heli.3.20150123.S: 22173.0\r\n",
      " E7.6.Mon.aequ.1.20150620.M: 22187.0\r\n",
      " E1.19.Sym.sp.1.20140815.S: 22229.0\r\n",
      " E7.17.Pla.sine.3.20150622.M: 22250.0\r\n",
      " E7.17.Hyd.exes.2.20150622.T: 22399.0\r\n",
      " E3.5.Gal.astr.2.20150124.S: 22464.0\r\n",
      " E7.5.Gal.astr.1.20150622.T: 22597.0\r\n",
      " E7.6.Mon.aequ.3.20150620.M: 22610.0\r\n",
      " E3.6.Acr.hyac.2.20150120.M: 22656.0\r\n",
      " E7.17.Fav.abdi.2.20150623.M: 22667.0\r\n",
      " E3.15.Dip.heli.1.20150105.M: 22674.0\r\n",
      " E7.10.Poc.dami.3.20150621.M: 22687.0\r\n",
      " E10.Acr.cyth.1.20150627.S: 22724.0\r\n",
      " E3.15.Dip.heli.2.20150123.S: 22874.0\r\n",
      " E7.10.Ser.cali.3.20150622.M: 22919.0\r\n",
      " E3.15.Dip.heli.3.20150105.M: 22931.0\r\n",
      " E1.15.Dip.heli.1.20140809.T: 22945.0\r\n",
      " E10.Mon.curt.2.20150629.M: 23048.0\r\n",
      " E10.Acr.pali.1.20150627.M: 23067.0\r\n",
      " E3.3.Por.cyli.2.20150123.M: 23076.0\r\n",
      " E10.Ser.cali.1.20150627.M: 23096.0\r\n",
      " E10.Por.hero.1.20150628.M: 23099.0\r\n",
      " E3.17.Ech.mamm.1.20150127.M: 23107.0\r\n",
      " E1.Out.Pal.caes.1.20140815.S: 23128.0\r\n",
      " E7.17.Cyp.micr.2.20150621.M: 23141.0\r\n",
      " E3.6.Acr.hyac.3.20150105.S: 23211.0\r\n",
      " E10.17.Fav.abdi.2.20150630.M: 23282.0\r\n",
      " E1.6.Iso.pali.1.20140803.T.2: 23341.0\r\n",
      " E3.15.Dip.heli.2.20150120.M: 23362.0\r\n",
      " E3.6.Acr.hyac.2.20150106.T: 23388.0\r\n",
      " E3.15.Dip.heli.2.20150106.S: 23400.0\r\n",
      " E7.17.Gon.edwa.2.20150623.M: 23443.0\r\n",
      " E3.6.Acr.hyac.1.20150120.M: 23484.0\r\n",
      " E7.17.Cyp.micr.1.20150621.M: 23530.0\r\n",
      " E10.Acr.cyth.3.20150627.M: 23539.0\r\n",
      " E1.11.Psa.digi.1.20140807.S: 23543.0\r\n",
      " E7.6.Acr.hyac.2.20150622.T: 23583.0\r\n",
      " E1.Out.Hel.coer.1.20140811.S: 23588.0\r\n",
      " E3.15.Dip.heli.3.20150123.M: 23655.0\r\n",
      " E1.5.Gal.astr.1.20140905.M: 23776.0\r\n",
      " E1.Out.Pal.caes.1.20140815.M: 23859.0\r\n",
      " E7.17.Hyd.exes.1.20150622.M: 24010.0\r\n",
      " E10.Fav.abdi.1.20150629.M: 24124.0\r\n",
      " E7.10.Ser.cali.2.20150622.M: 24222.0\r\n",
      " E10.Acr.cyth.2.20150627.S: 24267.0\r\n",
      " E7.5.Gal.fasc.1.20150621.T: 24385.0\r\n",
      " E7.3.Por.loba.4.20150619.M: 24448.0\r\n",
      " E10.Cyp.micr.1.20150627.M: 24528.0\r\n",
      " E3.6.Acr.hyac.1.20150105.M: 24644.0\r\n",
      " E10.17.Gon.favu.1.20150629.M: 24699.0\r\n",
      " E1.Out.Mil.sp.1.20140814.T: 24730.0\r\n",
      " E10.Acr.pali.3.20150627.M: 24768.0\r\n",
      " E1.14.Phy.lich.1.20140811.S: 24846.0\r\n",
      " E7.17.Fav.spB.1.20150623.M: 25081.0\r\n",
      " E3.15.Dip.heli.1.20150120.S: 25191.0\r\n",
      " E3.6.Acr.hyac.2.20150123.M: 25208.0\r\n",
      " E1.11.Psa.digi.1.20140807.T: 25567.0\r\n",
      " E3.6.Acr.hyac.3.20150123.M: 25664.0\r\n",
      " E10.Sed.1.20150627: 25668.0\r\n",
      " E7.17.Mer.ampl.3.20150621.M: 25690.0\r\n",
      " E7.3.Por.loba.1.20150619.M: 25879.0\r\n",
      " E3.6.Acr.hyac.1.20150106.M: 25937.0\r\n",
      " E3.Out.Pal.caes.2.20150125.M: 26230.0\r\n",
      " E10.Acr.cyth.2.20150627.T: 26327.0\r\n",
      " E10.17.Fav.abdi.1.20150630.M: 26462.0\r\n",
      " E1.4.Pac.rugo.1.20140814.S: 26511.0\r\n",
      " E7.17.Ech.lame.1.20150620.M: 26583.0\r\n",
      " E7.6.Mon.aequ.4.20150620.M: 26690.0\r\n",
      " E7.6.Mon.aequ.2.20150620.M: 26877.0\r\n",
      " E7.6.Acr.hyac.2.20150622.M: 26934.0\r\n",
      " E7.5.Gal.astr.1.20150622.M: 27075.0\r\n",
      " E7.17.Mer.ampl.1.20150621.M: 27128.0\r\n",
      " E10.Acr.cyth.2.20150627.M: 27180.0\r\n",
      " E3.15.Dip.heli.2.20150121.S: 27242.0\r\n",
      " E3.15.Dip.heli.3.20150121.S: 27387.0\r\n",
      " E1.3.Por.loba.1.20140905.S: 27482.0\r\n",
      " E7.6.Acr.hyac.1.20150622.M: 27595.0\r\n",
      " E1.11.Fun.sp.1.20140811.T: 27603.0\r\n",
      " E3.Out.Mil.sp.1.20150125.T: 27894.0\r\n",
      " E7.10.Ser.cali.1.20150622.M: 28052.0\r\n",
      " E10.Aca.hill.2.20150629.M: 28263.0\r\n",
      " E3.15.Dip.heli.1.20150121.M: 28304.0\r\n",
      " E10.Alv.alli.1.20150628.M: 28318.0\r\n",
      " E3.19.Sym.sp.2.20150125.M: 28347.0\r\n",
      " E3.6.Acr.hyac.2.20150105.M: 28794.0\r\n",
      " E3.Out.Mil.sp.2.20150125.T: 28870.0\r\n",
      " E3.17.Ech.mamm.1.20150123.M: 28896.0\r\n",
      " E7.19.Lob.hemp.1.20150621.M: 29541.0\r\n",
      " E3.11.Fun.sp.1.20150125.M: 30168.0\r\n",
      " E10.Alv.alli.2.20150628.M: 31070.0\r\n",
      " E3.Out.Mil.sp.2.20150125.S: 31091.0\r\n",
      " E1.11.Fun.sp.1.20140811.M: 32780.0\r\n",
      " E3.Out.Mil.sp.1.20150125.S: 33141.0\r\n",
      " E1.11.Fun.sp.1.20140811.S: 35581.0\r\n",
      " E10.Acr.cyth.1.20150627.M: 38523.0\r\n"
     ]
    }
   ],
   "source": [
    "#Summarize BIOM\n",
    "!biom summarize-table -i $otu_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
