{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Deep rarefaction analysis\n",
    "Reviewers suggested we try alternatives to the 1k rarefaction depth. Also there is evidence that you get significant info from reads until at least 50k in many systems. The tradeoff, of course, is that this results in fewer samples.\n",
    "\n",
    "So in this script we will test different rarefaction depths, and verify whether e.g. the pattern of skeleton > tissue > mucus diversity holds.\n",
    "\n",
    "The choice of depths is a little arbitrary, but going with 1k (original study), 5k (preserves many samples), 10k (same as Apprill et al, mentioned by reviewers), 15k (keeps about half of the samples),20k (about highest to preserve reasonable sample numbers)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bdiv_bc_prefs.txt\r\n",
      "gcmp16S_map_r25.txt\r\n",
      "gg_constrained_rep_set_fastttree.tre\r\n",
      "host_tree_from_step_11.newick\r\n",
      "otu_table_mc2_wtax_no_pynast_failures_no_organelles_coral_tissue_mucus_skeleton_only.biom\r\n",
      "readme.txt\r\n",
      "table_summary.txt\r\n"
     ]
    }
   ],
   "source": [
    "from os import listdir\n",
    "from os.path import join,abspath,splitext\n",
    "!ls ../input\n",
    "input_dir = \"../input/\"\n",
    "output_dir = \"../output/\"\n",
    "otu_table = join(input_dir,\"otu_table_mc2_wtax_no_pynast_failures_no_organelles_coral_tissue_mucus_skeleton_only.biom\")\n",
    "mapping = join(input_dir,\"gcmp16S_map_r25.txt\")\n",
    "tree = join(input_dir,\"gg_constrained_rep_set_fastttree.tre\")\n",
    "rarefaction_depths = [1000,5000,10000,15000,20000]\n",
    "bdiv_prefs_fp = join(input_dir,\"bdiv_bc_prefs.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Prefiltering\n",
    "#Table summaries suggest that there are are a few outgroup samples \n",
    "#still in the analysis.\n",
    "otu_table_no_outgroups = join(output_dir,\"otu_table_no_outgroups.biom\")\n",
    "!filter_samples_from_otu_table.py -i $otu_table -m $mapping -s \"outgroup:n\" \\\n",
    "  -o $otu_table_no_outgroups\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Rarefying at depth 1000\n",
      "Error in multiple_rarefactions_even_depth.py: option -i: file does not exist: '-o'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Rarefying at depth 5000\n",
      "Error in multiple_rarefactions_even_depth.py: option -i: file does not exist: '-o'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Rarefying at depth 10000\n",
      "Error in multiple_rarefactions_even_depth.py: option -i: file does not exist: '-o'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Rarefying at depth 15000\n",
      "Error in multiple_rarefactions_even_depth.py: option -i: file does not exist: '-o'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Rarefying at depth 20000\n",
      "Error in multiple_rarefactions_even_depth.py: option -i: file does not exist: '-o'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n"
     ]
    }
   ],
   "source": [
    "\n",
    "rarefied_data = {}\n",
    "n_reps = 10 #for the records, this is the default value\n",
    "for depth in rarefaction_depths:\n",
    "    \n",
    "    print(\"Rarefying at depth %i\" % depth)\n",
    "    \n",
    "    curr_output_dir = join(output_dir,\"rarified_to_%i/\" %depth)\n",
    "    rarefied_data[depth] = curr_output_dir\n",
    "    !multiple_rarefactions_even_depth.py -i $otu_table_no_outgroups \\\n",
    "      -o $curr_output_dir --lineages_included -d $depth -n 10\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define rarifaction directories \n",
    "\n",
    "Do this without running analytical steps to allow easier checkpointing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1000, 5000, 10000, 15000, 20000]\n",
      "{1000: '../output/adiv_1000/', 5000: '../output/adiv_5000/', 15000: '../output/adiv_15000/', 10000: '../output/adiv_10000/', 20000: '../output/adiv_20000/'}\n"
     ]
    }
   ],
   "source": [
    "#Separately inputing just the informational part of the rarifaction so you can\n",
    "#rariefied filepaths without actually rerunning the analysis.\n",
    "rarefaction_depths = [1000,5000,10000,15000,20000]\n",
    "print(rarefaction_depths)\n",
    "rarefied_data = {}\n",
    "collated_alpha_dirs = {}\n",
    "for depth in rarefaction_depths: \n",
    "    curr_output_dir = join(output_dir,\"rarified_to_%i/\" %depth)\n",
    "    rarefied_data[depth] = curr_output_dir\n",
    "    collated_alpha_dir = join(output_dir,\"adiv_%i/\" %depth)\n",
    "    collated_alpha_dirs[depth] = collated_alpha_dir\n",
    "    \n",
    "print(collated_alpha_dirs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Alpha-diveristy Compartment comparison\n",
    "\n",
    "Run alpha diversity for ALL samples (NOT split by compartment)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing depth 1000\n"
     ]
    }
   ],
   "source": [
    "adiv_methods = \"equitability,PD_whole_tree,chao1,observed_otus\"\n",
    "comparison_categories = \"BiologicalMatter,functional_group_sensu_darling\"\n",
    "for depth in rarefaction_depths:\n",
    "    print (\"Analyzing depth %i\" %depth)\n",
    "    otu_table_dir = rarefied_data[depth]\n",
    "    curr_outdir = join(output_dir,\"adiv_%i\") %depth\n",
    "    \n",
    "    !alpha_diversity.py -i $otu_table_dir -o $curr_outdir -t $tree -m $adiv_methods\n",
    "    \n",
    "    print (\"Collating alpha diversity...\")\n",
    "    collated_alpha_dir = join(output_dir,\"collated_alpha_%i/\" %depth)\n",
    "    print (\"Collated alpha dir: %s\" %collated_alpha_dir)\n",
    "    print (\"alpha diversity dir: %s\" %curr_outdir)\n",
    "    !collate_alpha.py -i $curr_outdir -o $collated_alpha_dir\n",
    "    print (\"Running statistical comparisons...\")\n",
    "    compare_alpha_results = join(output_dir,\"compare_alpha_%i\" %depth)\n",
    "    for m in adiv_methods.split(\",\"):\n",
    "        collated_alpha_file = join(collated_alpha_dir,\"%s.txt\" %m)\n",
    "        !compare_alpha_diversity.py -i $collated_alpha_file -o $compare_alpha_results \\\n",
    "          -m $mapping -c $comparison_categories -t nonparametric"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "### Add per-sample alpha-diversity to mapping file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "add_alpha_to_mapping_file.py -i ../output/adiv_1000/alpha_rarefaction_1000_0.txt -o ../output/mapping_with_alpha_values.txt -m ../input/gcmp16S_map_r25.txt --depth 1000 --binning_method quantile -b 4\n"
     ]
    }
   ],
   "source": [
    "depth = 1000\n",
    "alpha_dir = collated_alpha_dirs[depth]\n",
    "#alpha_files = \",\".join([join(alpha_dir,m) for m in (\"observed_otus.txt\",\"equitability.txt\",\"PD_whole_tree.txt\",\"chao1.txt\")])\n",
    "alpha_file = join(alpha_dir,\"alpha_rarefaction_1000_0.txt\")\n",
    "output_file = join(output_dir,\"mapping_with_alpha_values.txt\")\n",
    "cmd_str = \"add_alpha_to_mapping_file.py -i %s -o %s -m %s --depth %i --binning_method quantile -b 4\" %(alpha_file,output_file,mapping,depth)\n",
    "print(cmd_str)\n",
    "!$cmd_str"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Per compartment alpha-diversity comparisons\n",
    "\n",
    "To run compare_alpha_diversity I need seperate alpha-diversity results for each compartment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing depth 1000\n",
      "curr compartment filter:'BiologicalMatter:Coral Tissue'\n",
      "curr outdir: ../output/rarified_to_1000_Coral_Tissue\n",
      "mkdir: ../output/rarified_to_1000_Coral_Tissue: File exists\n",
      "Rarified OTU table: rarefaction_1000_0.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_1000_0.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Tissue/Coral_Tissue_rarefaction_1000_0.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_0.biom\n",
      "Rarified OTU table: rarefaction_1000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_1000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_1000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_1000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_1000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_1000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_1000_1.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_1000_1.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Tissue/Coral_Tissue_rarefaction_1000_1.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_1.biom\n",
      "Rarified OTU table: rarefaction_1000_2.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_1000_2.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Tissue/Coral_Tissue_rarefaction_1000_2.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_2.biom\n",
      "Rarified OTU table: rarefaction_1000_3.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_1000_3.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Tissue/Coral_Tissue_rarefaction_1000_3.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_3.biom\n",
      "Rarified OTU table: rarefaction_1000_4.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_1000_4.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Tissue/Coral_Tissue_rarefaction_1000_4.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_4.biom\n",
      "Rarified OTU table: rarefaction_1000_5.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_1000_5.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Tissue/Coral_Tissue_rarefaction_1000_5.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_5.biom\n",
      "Rarified OTU table: rarefaction_1000_6.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_1000_6.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Tissue/Coral_Tissue_rarefaction_1000_6.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_6.biom\n",
      "Rarified OTU table: rarefaction_1000_7.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_1000_7.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Tissue/Coral_Tissue_rarefaction_1000_7.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_7.biom\n",
      "Rarified OTU table: rarefaction_1000_8.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_1000_8.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Tissue/Coral_Tissue_rarefaction_1000_8.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_8.biom\n",
      "Rarified OTU table: rarefaction_1000_9.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_1000_9.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Tissue/Coral_Tissue_rarefaction_1000_9.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_9.biom\n",
      "curr compartment filter:'BiologicalMatter:Coral Mucus'\n",
      "curr outdir: ../output/rarified_to_1000_Coral_Mucus\n",
      "Rarified OTU table: rarefaction_1000_0.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_1000_0.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Mucus/Coral_Mucus_rarefaction_1000_0.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_0.biom\n",
      "Rarified OTU table: rarefaction_1000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_1000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_1000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_1000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_1000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_1000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_1000_1.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_1000_1.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Mucus/Coral_Mucus_rarefaction_1000_1.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_1.biom\n",
      "Rarified OTU table: rarefaction_1000_2.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_1000_2.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Mucus/Coral_Mucus_rarefaction_1000_2.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_2.biom\n",
      "Rarified OTU table: rarefaction_1000_3.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_1000_3.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Mucus/Coral_Mucus_rarefaction_1000_3.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_3.biom\n",
      "Rarified OTU table: rarefaction_1000_4.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_1000_4.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Mucus/Coral_Mucus_rarefaction_1000_4.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_4.biom\n",
      "Rarified OTU table: rarefaction_1000_5.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_1000_5.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Mucus/Coral_Mucus_rarefaction_1000_5.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_5.biom\n",
      "Rarified OTU table: rarefaction_1000_6.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_1000_6.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Mucus/Coral_Mucus_rarefaction_1000_6.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_6.biom\n",
      "Rarified OTU table: rarefaction_1000_7.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_1000_7.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Mucus/Coral_Mucus_rarefaction_1000_7.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_7.biom\n",
      "Rarified OTU table: rarefaction_1000_8.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_1000_8.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Mucus/Coral_Mucus_rarefaction_1000_8.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_8.biom\n",
      "Rarified OTU table: rarefaction_1000_9.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_1000_9.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Mucus/Coral_Mucus_rarefaction_1000_9.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_9.biom\n",
      "curr compartment filter:'BiologicalMatter:Coral Skeleton'\n",
      "curr outdir: ../output/rarified_to_1000_Coral_Skeleton\n",
      "Rarified OTU table: rarefaction_1000_0.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_1000_0.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Skeleton/Coral_Skeleton_rarefaction_1000_0.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_0.biom\n",
      "Rarified OTU table: rarefaction_1000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_1000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_1000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_1000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_1000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_1000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_1000_1.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_1000_1.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Skeleton/Coral_Skeleton_rarefaction_1000_1.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_1.biom\n",
      "Rarified OTU table: rarefaction_1000_2.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_1000_2.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Skeleton/Coral_Skeleton_rarefaction_1000_2.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_2.biom\n",
      "Rarified OTU table: rarefaction_1000_3.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_1000_3.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Skeleton/Coral_Skeleton_rarefaction_1000_3.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_3.biom\n",
      "Rarified OTU table: rarefaction_1000_4.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_1000_4.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Skeleton/Coral_Skeleton_rarefaction_1000_4.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_4.biom\n",
      "Rarified OTU table: rarefaction_1000_5.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_1000_5.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Skeleton/Coral_Skeleton_rarefaction_1000_5.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_5.biom\n",
      "Rarified OTU table: rarefaction_1000_6.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_1000_6.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Skeleton/Coral_Skeleton_rarefaction_1000_6.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_6.biom\n",
      "Rarified OTU table: rarefaction_1000_7.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_1000_7.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Skeleton/Coral_Skeleton_rarefaction_1000_7.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_7.biom\n",
      "Rarified OTU table: rarefaction_1000_8.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_1000_8.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Skeleton/Coral_Skeleton_rarefaction_1000_8.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_8.biom\n",
      "Rarified OTU table: rarefaction_1000_9.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_1000_9.biom\n",
      "Output filepath:../output/rarified_to_1000_Coral_Skeleton/Coral_Skeleton_rarefaction_1000_9.biom\n",
      "Table to filter:../output/rarified_to_1000/rarefaction_1000_9.biom\n",
      "Analyzing depth 5000\n",
      "curr compartment filter:'BiologicalMatter:Coral Tissue'\n",
      "curr outdir: ../output/rarified_to_5000_Coral_Tissue\n",
      "mkdir: ../output/rarified_to_5000_Coral_Tissue: File exists\n",
      "Rarified OTU table: rarefaction_5000_0.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_5000_0.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Tissue/Coral_Tissue_rarefaction_5000_0.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_0.biom\n",
      "Rarified OTU table: rarefaction_5000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_5000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_5000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_5000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_5000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_5000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_5000_1.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_5000_1.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Tissue/Coral_Tissue_rarefaction_5000_1.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_1.biom\n",
      "Rarified OTU table: rarefaction_5000_2.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_5000_2.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Tissue/Coral_Tissue_rarefaction_5000_2.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_2.biom\n",
      "Rarified OTU table: rarefaction_5000_3.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_5000_3.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Tissue/Coral_Tissue_rarefaction_5000_3.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_3.biom\n",
      "Rarified OTU table: rarefaction_5000_4.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_5000_4.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Tissue/Coral_Tissue_rarefaction_5000_4.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_4.biom\n",
      "Rarified OTU table: rarefaction_5000_5.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_5000_5.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Tissue/Coral_Tissue_rarefaction_5000_5.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_5.biom\n",
      "Rarified OTU table: rarefaction_5000_6.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_5000_6.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Tissue/Coral_Tissue_rarefaction_5000_6.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_6.biom\n",
      "Rarified OTU table: rarefaction_5000_7.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_5000_7.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Tissue/Coral_Tissue_rarefaction_5000_7.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_7.biom\n",
      "Rarified OTU table: rarefaction_5000_8.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_5000_8.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Tissue/Coral_Tissue_rarefaction_5000_8.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_8.biom\n",
      "Rarified OTU table: rarefaction_5000_9.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_5000_9.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Tissue/Coral_Tissue_rarefaction_5000_9.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_9.biom\n",
      "curr compartment filter:'BiologicalMatter:Coral Mucus'\n",
      "curr outdir: ../output/rarified_to_5000_Coral_Mucus\n",
      "mkdir: ../output/rarified_to_5000_Coral_Mucus: File exists\n",
      "Rarified OTU table: rarefaction_5000_0.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_5000_0.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Mucus/Coral_Mucus_rarefaction_5000_0.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_0.biom\n",
      "Rarified OTU table: rarefaction_5000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_5000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_5000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_5000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_5000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_5000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_5000_1.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_5000_1.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Mucus/Coral_Mucus_rarefaction_5000_1.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_1.biom\n",
      "Rarified OTU table: rarefaction_5000_2.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_5000_2.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Mucus/Coral_Mucus_rarefaction_5000_2.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_2.biom\n",
      "Rarified OTU table: rarefaction_5000_3.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_5000_3.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Mucus/Coral_Mucus_rarefaction_5000_3.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_3.biom\n",
      "Rarified OTU table: rarefaction_5000_4.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_5000_4.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Mucus/Coral_Mucus_rarefaction_5000_4.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_4.biom\n",
      "Rarified OTU table: rarefaction_5000_5.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_5000_5.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Mucus/Coral_Mucus_rarefaction_5000_5.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_5.biom\n",
      "Rarified OTU table: rarefaction_5000_6.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_5000_6.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Mucus/Coral_Mucus_rarefaction_5000_6.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_6.biom\n",
      "Rarified OTU table: rarefaction_5000_7.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_5000_7.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Mucus/Coral_Mucus_rarefaction_5000_7.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_7.biom\n",
      "Rarified OTU table: rarefaction_5000_8.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_5000_8.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Mucus/Coral_Mucus_rarefaction_5000_8.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_8.biom\n",
      "Rarified OTU table: rarefaction_5000_9.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_5000_9.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Mucus/Coral_Mucus_rarefaction_5000_9.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_9.biom\n",
      "curr compartment filter:'BiologicalMatter:Coral Skeleton'\n",
      "curr outdir: ../output/rarified_to_5000_Coral_Skeleton\n",
      "mkdir: ../output/rarified_to_5000_Coral_Skeleton: File exists\n",
      "Rarified OTU table: rarefaction_5000_0.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_5000_0.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Skeleton/Coral_Skeleton_rarefaction_5000_0.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_0.biom\n",
      "Rarified OTU table: rarefaction_5000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_5000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_5000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_5000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_5000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_5000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_5000_1.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_5000_1.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Skeleton/Coral_Skeleton_rarefaction_5000_1.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_1.biom\n",
      "Rarified OTU table: rarefaction_5000_2.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_5000_2.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Skeleton/Coral_Skeleton_rarefaction_5000_2.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_2.biom\n",
      "Rarified OTU table: rarefaction_5000_3.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_5000_3.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Skeleton/Coral_Skeleton_rarefaction_5000_3.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_3.biom\n",
      "Rarified OTU table: rarefaction_5000_4.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_5000_4.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Skeleton/Coral_Skeleton_rarefaction_5000_4.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_4.biom\n",
      "Rarified OTU table: rarefaction_5000_5.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_5000_5.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Skeleton/Coral_Skeleton_rarefaction_5000_5.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_5.biom\n",
      "Rarified OTU table: rarefaction_5000_6.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_5000_6.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Skeleton/Coral_Skeleton_rarefaction_5000_6.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_6.biom\n",
      "Rarified OTU table: rarefaction_5000_7.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_5000_7.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Skeleton/Coral_Skeleton_rarefaction_5000_7.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_7.biom\n",
      "Rarified OTU table: rarefaction_5000_8.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_5000_8.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Skeleton/Coral_Skeleton_rarefaction_5000_8.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_8.biom\n",
      "Rarified OTU table: rarefaction_5000_9.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_5000_9.biom\n",
      "Output filepath:../output/rarified_to_5000_Coral_Skeleton/Coral_Skeleton_rarefaction_5000_9.biom\n",
      "Table to filter:../output/rarified_to_5000/rarefaction_5000_9.biom\n",
      "Analyzing depth 10000\n",
      "curr compartment filter:'BiologicalMatter:Coral Tissue'\n",
      "curr outdir: ../output/rarified_to_10000_Coral_Tissue\n",
      "mkdir: ../output/rarified_to_10000_Coral_Tissue: File exists\n",
      "Rarified OTU table: rarefaction_10000_0.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_10000_0.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Tissue/Coral_Tissue_rarefaction_10000_0.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_0.biom\n",
      "Rarified OTU table: rarefaction_10000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_10000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_10000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_10000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_10000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_10000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_10000_1.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_10000_1.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Tissue/Coral_Tissue_rarefaction_10000_1.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_1.biom\n",
      "Rarified OTU table: rarefaction_10000_2.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_10000_2.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Tissue/Coral_Tissue_rarefaction_10000_2.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_2.biom\n",
      "Rarified OTU table: rarefaction_10000_3.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_10000_3.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Tissue/Coral_Tissue_rarefaction_10000_3.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_3.biom\n",
      "Rarified OTU table: rarefaction_10000_4.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_10000_4.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Tissue/Coral_Tissue_rarefaction_10000_4.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_4.biom\n",
      "Rarified OTU table: rarefaction_10000_5.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_10000_5.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Tissue/Coral_Tissue_rarefaction_10000_5.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_5.biom\n",
      "Rarified OTU table: rarefaction_10000_6.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_10000_6.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Tissue/Coral_Tissue_rarefaction_10000_6.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_6.biom\n",
      "Rarified OTU table: rarefaction_10000_7.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_10000_7.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Tissue/Coral_Tissue_rarefaction_10000_7.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_7.biom\n",
      "Rarified OTU table: rarefaction_10000_8.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_10000_8.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Tissue/Coral_Tissue_rarefaction_10000_8.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_8.biom\n",
      "Rarified OTU table: rarefaction_10000_9.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_10000_9.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Tissue/Coral_Tissue_rarefaction_10000_9.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_9.biom\n",
      "curr compartment filter:'BiologicalMatter:Coral Mucus'\n",
      "curr outdir: ../output/rarified_to_10000_Coral_Mucus\n",
      "mkdir: ../output/rarified_to_10000_Coral_Mucus: File exists\n",
      "Rarified OTU table: rarefaction_10000_0.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_10000_0.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Mucus/Coral_Mucus_rarefaction_10000_0.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_0.biom\n",
      "Rarified OTU table: rarefaction_10000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_10000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_10000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_10000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_10000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_10000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_10000_1.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_10000_1.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Mucus/Coral_Mucus_rarefaction_10000_1.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_1.biom\n",
      "Rarified OTU table: rarefaction_10000_2.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_10000_2.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Mucus/Coral_Mucus_rarefaction_10000_2.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_2.biom\n",
      "Rarified OTU table: rarefaction_10000_3.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_10000_3.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Mucus/Coral_Mucus_rarefaction_10000_3.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_3.biom\n",
      "Rarified OTU table: rarefaction_10000_4.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_10000_4.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Mucus/Coral_Mucus_rarefaction_10000_4.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_4.biom\n",
      "Rarified OTU table: rarefaction_10000_5.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_10000_5.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Mucus/Coral_Mucus_rarefaction_10000_5.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_5.biom\n",
      "Rarified OTU table: rarefaction_10000_6.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_10000_6.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Mucus/Coral_Mucus_rarefaction_10000_6.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_6.biom\n",
      "Rarified OTU table: rarefaction_10000_7.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_10000_7.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Mucus/Coral_Mucus_rarefaction_10000_7.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_7.biom\n",
      "Rarified OTU table: rarefaction_10000_8.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_10000_8.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Mucus/Coral_Mucus_rarefaction_10000_8.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_8.biom\n",
      "Rarified OTU table: rarefaction_10000_9.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_10000_9.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Mucus/Coral_Mucus_rarefaction_10000_9.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_9.biom\n",
      "curr compartment filter:'BiologicalMatter:Coral Skeleton'\n",
      "curr outdir: ../output/rarified_to_10000_Coral_Skeleton\n",
      "mkdir: ../output/rarified_to_10000_Coral_Skeleton: File exists\n",
      "Rarified OTU table: rarefaction_10000_0.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_10000_0.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Skeleton/Coral_Skeleton_rarefaction_10000_0.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_0.biom\n",
      "Rarified OTU table: rarefaction_10000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_10000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_10000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_10000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_10000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_10000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_10000_1.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_10000_1.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Skeleton/Coral_Skeleton_rarefaction_10000_1.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_1.biom\n",
      "Rarified OTU table: rarefaction_10000_2.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_10000_2.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Skeleton/Coral_Skeleton_rarefaction_10000_2.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_2.biom\n",
      "Rarified OTU table: rarefaction_10000_3.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_10000_3.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Skeleton/Coral_Skeleton_rarefaction_10000_3.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_3.biom\n",
      "Rarified OTU table: rarefaction_10000_4.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_10000_4.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Skeleton/Coral_Skeleton_rarefaction_10000_4.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_4.biom\n",
      "Rarified OTU table: rarefaction_10000_5.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_10000_5.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Skeleton/Coral_Skeleton_rarefaction_10000_5.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_5.biom\n",
      "Rarified OTU table: rarefaction_10000_6.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_10000_6.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Skeleton/Coral_Skeleton_rarefaction_10000_6.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_6.biom\n",
      "Rarified OTU table: rarefaction_10000_7.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_10000_7.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Skeleton/Coral_Skeleton_rarefaction_10000_7.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_7.biom\n",
      "Rarified OTU table: rarefaction_10000_8.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_10000_8.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Skeleton/Coral_Skeleton_rarefaction_10000_8.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_8.biom\n",
      "Rarified OTU table: rarefaction_10000_9.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_10000_9.biom\n",
      "Output filepath:../output/rarified_to_10000_Coral_Skeleton/Coral_Skeleton_rarefaction_10000_9.biom\n",
      "Table to filter:../output/rarified_to_10000/rarefaction_10000_9.biom\n",
      "Analyzing depth 15000\n",
      "curr compartment filter:'BiologicalMatter:Coral Tissue'\n",
      "curr outdir: ../output/rarified_to_15000_Coral_Tissue\n",
      "mkdir: ../output/rarified_to_15000_Coral_Tissue: File exists\n",
      "Rarified OTU table: rarefaction_15000_0.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_15000_0.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Tissue/Coral_Tissue_rarefaction_15000_0.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_0.biom\n",
      "Rarified OTU table: rarefaction_15000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_15000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_15000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_15000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_15000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_15000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_15000_1.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_15000_1.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Tissue/Coral_Tissue_rarefaction_15000_1.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_1.biom\n",
      "Rarified OTU table: rarefaction_15000_2.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_15000_2.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Tissue/Coral_Tissue_rarefaction_15000_2.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_2.biom\n",
      "Rarified OTU table: rarefaction_15000_3.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_15000_3.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Tissue/Coral_Tissue_rarefaction_15000_3.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_3.biom\n",
      "Rarified OTU table: rarefaction_15000_4.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_15000_4.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Tissue/Coral_Tissue_rarefaction_15000_4.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_4.biom\n",
      "Rarified OTU table: rarefaction_15000_5.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_15000_5.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Tissue/Coral_Tissue_rarefaction_15000_5.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_5.biom\n",
      "Rarified OTU table: rarefaction_15000_6.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_15000_6.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Tissue/Coral_Tissue_rarefaction_15000_6.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_6.biom\n",
      "Rarified OTU table: rarefaction_15000_7.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_15000_7.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Tissue/Coral_Tissue_rarefaction_15000_7.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_7.biom\n",
      "Rarified OTU table: rarefaction_15000_8.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_15000_8.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Tissue/Coral_Tissue_rarefaction_15000_8.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_8.biom\n",
      "Rarified OTU table: rarefaction_15000_9.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_15000_9.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Tissue/Coral_Tissue_rarefaction_15000_9.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_9.biom\n",
      "curr compartment filter:'BiologicalMatter:Coral Mucus'\n",
      "curr outdir: ../output/rarified_to_15000_Coral_Mucus\n",
      "mkdir: ../output/rarified_to_15000_Coral_Mucus: File exists\n",
      "Rarified OTU table: rarefaction_15000_0.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_15000_0.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Mucus/Coral_Mucus_rarefaction_15000_0.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_0.biom\n",
      "Rarified OTU table: rarefaction_15000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_15000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_15000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_15000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_15000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_15000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_15000_1.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_15000_1.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Mucus/Coral_Mucus_rarefaction_15000_1.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_1.biom\n",
      "Rarified OTU table: rarefaction_15000_2.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_15000_2.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Mucus/Coral_Mucus_rarefaction_15000_2.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_2.biom\n",
      "Rarified OTU table: rarefaction_15000_3.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_15000_3.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Mucus/Coral_Mucus_rarefaction_15000_3.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_3.biom\n",
      "Rarified OTU table: rarefaction_15000_4.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_15000_4.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Mucus/Coral_Mucus_rarefaction_15000_4.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_4.biom\n",
      "Rarified OTU table: rarefaction_15000_5.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_15000_5.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Mucus/Coral_Mucus_rarefaction_15000_5.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_5.biom\n",
      "Rarified OTU table: rarefaction_15000_6.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_15000_6.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Mucus/Coral_Mucus_rarefaction_15000_6.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_6.biom\n",
      "Rarified OTU table: rarefaction_15000_7.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_15000_7.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Mucus/Coral_Mucus_rarefaction_15000_7.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_7.biom\n",
      "Rarified OTU table: rarefaction_15000_8.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_15000_8.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Mucus/Coral_Mucus_rarefaction_15000_8.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_8.biom\n",
      "Rarified OTU table: rarefaction_15000_9.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_15000_9.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Mucus/Coral_Mucus_rarefaction_15000_9.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_9.biom\n",
      "curr compartment filter:'BiologicalMatter:Coral Skeleton'\n",
      "curr outdir: ../output/rarified_to_15000_Coral_Skeleton\n",
      "mkdir: ../output/rarified_to_15000_Coral_Skeleton: File exists\n",
      "Rarified OTU table: rarefaction_15000_0.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_15000_0.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Skeleton/Coral_Skeleton_rarefaction_15000_0.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_0.biom\n",
      "Rarified OTU table: rarefaction_15000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_15000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_15000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_15000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_15000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_15000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_15000_1.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_15000_1.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Skeleton/Coral_Skeleton_rarefaction_15000_1.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_1.biom\n",
      "Rarified OTU table: rarefaction_15000_2.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_15000_2.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Skeleton/Coral_Skeleton_rarefaction_15000_2.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_2.biom\n",
      "Rarified OTU table: rarefaction_15000_3.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_15000_3.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Skeleton/Coral_Skeleton_rarefaction_15000_3.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_3.biom\n",
      "Rarified OTU table: rarefaction_15000_4.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_15000_4.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Skeleton/Coral_Skeleton_rarefaction_15000_4.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_4.biom\n",
      "Rarified OTU table: rarefaction_15000_5.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_15000_5.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Skeleton/Coral_Skeleton_rarefaction_15000_5.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_5.biom\n",
      "Rarified OTU table: rarefaction_15000_6.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_15000_6.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Skeleton/Coral_Skeleton_rarefaction_15000_6.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_6.biom\n",
      "Rarified OTU table: rarefaction_15000_7.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_15000_7.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Skeleton/Coral_Skeleton_rarefaction_15000_7.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_7.biom\n",
      "Rarified OTU table: rarefaction_15000_8.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_15000_8.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Skeleton/Coral_Skeleton_rarefaction_15000_8.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_8.biom\n",
      "Rarified OTU table: rarefaction_15000_9.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_15000_9.biom\n",
      "Output filepath:../output/rarified_to_15000_Coral_Skeleton/Coral_Skeleton_rarefaction_15000_9.biom\n",
      "Table to filter:../output/rarified_to_15000/rarefaction_15000_9.biom\n",
      "Analyzing depth 20000\n",
      "curr compartment filter:'BiologicalMatter:Coral Tissue'\n",
      "curr outdir: ../output/rarified_to_20000_Coral_Tissue\n",
      "mkdir: ../output/rarified_to_20000_Coral_Tissue: File exists\n",
      "Rarified OTU table: rarefaction_20000_0.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_20000_0.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Tissue/Coral_Tissue_rarefaction_20000_0.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_0.biom\n",
      "Rarified OTU table: rarefaction_20000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_20000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_20000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_20000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_20000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_20000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_20000_1.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_20000_1.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Tissue/Coral_Tissue_rarefaction_20000_1.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_1.biom\n",
      "Rarified OTU table: rarefaction_20000_2.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_20000_2.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Tissue/Coral_Tissue_rarefaction_20000_2.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_2.biom\n",
      "Rarified OTU table: rarefaction_20000_3.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_20000_3.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Tissue/Coral_Tissue_rarefaction_20000_3.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_3.biom\n",
      "Rarified OTU table: rarefaction_20000_4.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_20000_4.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Tissue/Coral_Tissue_rarefaction_20000_4.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_4.biom\n",
      "Rarified OTU table: rarefaction_20000_5.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_20000_5.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Tissue/Coral_Tissue_rarefaction_20000_5.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_5.biom\n",
      "Rarified OTU table: rarefaction_20000_6.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_20000_6.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Tissue/Coral_Tissue_rarefaction_20000_6.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_6.biom\n",
      "Rarified OTU table: rarefaction_20000_7.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_20000_7.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Tissue/Coral_Tissue_rarefaction_20000_7.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_7.biom\n",
      "Rarified OTU table: rarefaction_20000_8.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_20000_8.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Tissue/Coral_Tissue_rarefaction_20000_8.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_8.biom\n",
      "Rarified OTU table: rarefaction_20000_9.biom\n",
      "Filtered filename: Coral_Tissue_rarefaction_20000_9.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Tissue/Coral_Tissue_rarefaction_20000_9.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_9.biom\n",
      "curr compartment filter:'BiologicalMatter:Coral Mucus'\n",
      "curr outdir: ../output/rarified_to_20000_Coral_Mucus\n",
      "mkdir: ../output/rarified_to_20000_Coral_Mucus: File exists\n",
      "Rarified OTU table: rarefaction_20000_0.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_20000_0.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Mucus/Coral_Mucus_rarefaction_20000_0.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_0.biom\n",
      "Rarified OTU table: rarefaction_20000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_20000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_20000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_20000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_20000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_20000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_20000_1.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_20000_1.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Mucus/Coral_Mucus_rarefaction_20000_1.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_1.biom\n",
      "Rarified OTU table: rarefaction_20000_2.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_20000_2.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Mucus/Coral_Mucus_rarefaction_20000_2.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_2.biom\n",
      "Rarified OTU table: rarefaction_20000_3.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_20000_3.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Mucus/Coral_Mucus_rarefaction_20000_3.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_3.biom\n",
      "Rarified OTU table: rarefaction_20000_4.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_20000_4.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Mucus/Coral_Mucus_rarefaction_20000_4.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_4.biom\n",
      "Rarified OTU table: rarefaction_20000_5.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_20000_5.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Mucus/Coral_Mucus_rarefaction_20000_5.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_5.biom\n",
      "Rarified OTU table: rarefaction_20000_6.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_20000_6.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Mucus/Coral_Mucus_rarefaction_20000_6.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_6.biom\n",
      "Rarified OTU table: rarefaction_20000_7.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_20000_7.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Mucus/Coral_Mucus_rarefaction_20000_7.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_7.biom\n",
      "Rarified OTU table: rarefaction_20000_8.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_20000_8.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Mucus/Coral_Mucus_rarefaction_20000_8.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_8.biom\n",
      "Rarified OTU table: rarefaction_20000_9.biom\n",
      "Filtered filename: Coral_Mucus_rarefaction_20000_9.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Mucus/Coral_Mucus_rarefaction_20000_9.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_9.biom\n",
      "curr compartment filter:'BiologicalMatter:Coral Skeleton'\n",
      "curr outdir: ../output/rarified_to_20000_Coral_Skeleton\n",
      "mkdir: ../output/rarified_to_20000_Coral_Skeleton: File exists\n",
      "Rarified OTU table: rarefaction_20000_0.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_20000_0.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Skeleton/Coral_Skeleton_rarefaction_20000_0.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_0.biom\n",
      "Rarified OTU table: rarefaction_20000_0_Coral_Mucus.biom\n",
      "Skipping file: rarefaction_20000_0_Coral_Mucus.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_20000_0_Coral_Skeleton.biom\n",
      "Skipping file: rarefaction_20000_0_Coral_Skeleton.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_20000_0_Coral_Tissue.biom\n",
      "Skipping file: rarefaction_20000_0_Coral_Tissue.biom ....doesn't look like a rarified OTU table\n",
      "Rarified OTU table: rarefaction_20000_1.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_20000_1.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Skeleton/Coral_Skeleton_rarefaction_20000_1.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_1.biom\n",
      "Rarified OTU table: rarefaction_20000_2.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_20000_2.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Skeleton/Coral_Skeleton_rarefaction_20000_2.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_2.biom\n",
      "Rarified OTU table: rarefaction_20000_3.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_20000_3.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Skeleton/Coral_Skeleton_rarefaction_20000_3.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_3.biom\n",
      "Rarified OTU table: rarefaction_20000_4.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_20000_4.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Skeleton/Coral_Skeleton_rarefaction_20000_4.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_4.biom\n",
      "Rarified OTU table: rarefaction_20000_5.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_20000_5.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Skeleton/Coral_Skeleton_rarefaction_20000_5.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_5.biom\n",
      "Rarified OTU table: rarefaction_20000_6.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_20000_6.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Skeleton/Coral_Skeleton_rarefaction_20000_6.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_6.biom\n",
      "Rarified OTU table: rarefaction_20000_7.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_20000_7.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Skeleton/Coral_Skeleton_rarefaction_20000_7.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_7.biom\n",
      "Rarified OTU table: rarefaction_20000_8.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_20000_8.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Skeleton/Coral_Skeleton_rarefaction_20000_8.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_8.biom\n",
      "Rarified OTU table: rarefaction_20000_9.biom\n",
      "Filtered filename: Coral_Skeleton_rarefaction_20000_9.biom\n",
      "Output filepath:../output/rarified_to_20000_Coral_Skeleton/Coral_Skeleton_rarefaction_20000_9.biom\n",
      "Table to filter:../output/rarified_to_20000/rarefaction_20000_9.biom\n"
     ]
    }
   ],
   "source": [
    "compartments = ['Coral Tissue','Coral Mucus','Coral Skeleton']\n",
    "\n",
    "for depth in rarefaction_depths:\n",
    "    print (\"Analyzing depth %i\" %depth)\n",
    "    otu_table_dir = rarefied_data[depth]\n",
    "    #curr_otu_table = join(otu_table_dir,\"rarefaction_%i_0.biom\" %depth)\n",
    "    #print (\"curr_otu_table:\",curr_otu_table)\n",
    "    for compartment in compartments:\n",
    "        compartment_no_spaces = compartment.replace(\" \",\"_\")\n",
    "        criterion = \"'BiologicalMatter:%s'\" %compartment\n",
    "        print(\"curr compartment filter:%s\" %criterion)\n",
    "        \n",
    "        curr_outdir = join(output_dir,\"rarified_to_%i_%s\" %(depth,compartment_no_spaces))\n",
    "        print(\"curr outdir: %s\" %curr_outdir)\n",
    "        !mkdir $curr_outdir\n",
    "        \n",
    "        for rarified_otu_table in listdir(otu_table_dir):\n",
    "            print(\"Rarified OTU table: %s\" %rarified_otu_table)\n",
    "            base_name,ext = splitext(rarified_otu_table)\n",
    "            if not base_name[-1].isdigit():\n",
    "                print(\"Skipping file: %s ....doesn't look like a rarified OTU table\" %rarified_otu_table)\n",
    "                continue\n",
    "            new_name = \"\".join([\"_\".join([compartment_no_spaces,base_name]),ext])\n",
    "            print(\"Filtered filename: %s\" %new_name)\n",
    "            new_fp = join(curr_outdir,new_name)\n",
    "            print(\"Output filepath:%s\" %new_fp)\n",
    "            \n",
    "            \n",
    "            table_to_filter = join(otu_table_dir,rarified_otu_table)\n",
    "            print(\"Table to filter:%s\" %table_to_filter)\n",
    "            !filter_samples_from_otu_table.py -i $table_to_filter -o $new_fp --valid_states $criterion -m $mapping\n",
    "        \n",
    "        \n",
    "    \n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing depth 1000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_1000_Coral_Tissue\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_1000_Coral_Tissue -o ../output/adiv_Coral_Tissue_1000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Tissue_1000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Tissue_1000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_1000_Coral_Mucus\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_1000_Coral_Mucus -o ../output/adiv_Coral_Mucus_1000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Mucus_1000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Mucus_1000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_1000_Coral_Skeleton\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_1000_Coral_Skeleton -o ../output/adiv_Coral_Skeleton_1000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Skeleton_1000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Skeleton_1000\n",
      "Analyzing depth 5000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_5000_Coral_Tissue\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_5000_Coral_Tissue -o ../output/adiv_Coral_Tissue_5000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Tissue_5000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Tissue_5000\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/collate_alpha.py\", line 130, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/collate_alpha.py\", line 96, in main\n",
      "    file_name_table = map(parse_rarefaction_fname, file_names)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/parse.py\", line 379, in parse_rarefaction_fname\n",
      "    iters = int(root_list.pop())\n",
      "ValueError: invalid literal for int() with base 10: 'Tissue'\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_5000_Coral_Mucus\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_5000_Coral_Mucus -o ../output/adiv_Coral_Mucus_5000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Mucus_5000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Mucus_5000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_5000_Coral_Skeleton\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_5000_Coral_Skeleton -o ../output/adiv_Coral_Skeleton_5000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Skeleton_5000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Skeleton_5000\n",
      "Analyzing depth 10000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_10000_Coral_Tissue\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_10000_Coral_Tissue -o ../output/adiv_Coral_Tissue_10000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/alpha_diversity.py\", line 131, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/alpha_diversity.py\", line 115, in main\n",
      "    opts.tree_path)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/alpha_diversity.py\", line 379, in multiple_file_alpha\n",
      "    output_fp, tree_path)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/alpha_diversity.py\", line 326, in single_file_alpha\n",
      "    result_path=outfilepath, log_path=None)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 259, in __call__\n",
      "    result = self.getResult(*args, **kwargs)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/alpha_diversity.py\", line 197, in getResult\n",
      "    taxon_names=otu_table.ids(axis='observation'),\n",
      "AttributeError: 'NoneType' object has no attribute 'ids'\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Tissue_10000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Tissue_10000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_10000_Coral_Mucus\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_10000_Coral_Mucus -o ../output/adiv_Coral_Mucus_10000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Mucus_10000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Mucus_10000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_10000_Coral_Skeleton\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_10000_Coral_Skeleton -o ../output/adiv_Coral_Skeleton_10000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Skeleton_10000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Skeleton_10000\n",
      "Analyzing depth 15000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_15000_Coral_Tissue\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_15000_Coral_Tissue -o ../output/adiv_Coral_Tissue_15000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Tissue_15000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Tissue_15000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_15000_Coral_Mucus\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_15000_Coral_Mucus -o ../output/adiv_Coral_Mucus_15000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Mucus_15000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Mucus_15000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_15000_Coral_Skeleton\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_15000_Coral_Skeleton -o ../output/adiv_Coral_Skeleton_15000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Skeleton_15000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Skeleton_15000\n",
      "Analyzing depth 20000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_20000_Coral_Tissue\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_20000_Coral_Tissue -o ../output/adiv_Coral_Tissue_20000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Tissue_20000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Tissue_20000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_20000_Coral_Mucus\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_20000_Coral_Mucus -o ../output/adiv_Coral_Mucus_20000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Mucus_20000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Mucus_20000\n",
      "Calculating alpha diversity on directory: ../output/rarified_to_20000_Coral_Skeleton\n",
      "adiv cmd:alpha_diversity.py -i ../output/rarified_to_20000_Coral_Skeleton -o ../output/adiv_Coral_Skeleton_20000 -t ../input/gg_constrained_rep_set_fastttree.tre -m equitability,PD_whole_tree,chao1,observed_otus\n",
      "Done. alpha diversity output dir: ../output/adiv_Coral_Skeleton_20000\n",
      "Collating alpha diversity to output dir: ../output/collated_alpha_Coral_Skeleton_20000\n"
     ]
    }
   ],
   "source": [
    "compartments = ['Coral Tissue','Coral Mucus','Coral Skeleton']\n",
    "adiv_methods = \"equitability,PD_whole_tree,chao1,observed_otus\"\n",
    "\n",
    "for depth in rarefaction_depths:\n",
    "    print (\"Analyzing depth %i\" %depth)\n",
    "    \n",
    "    #here's the issue\n",
    "    \n",
    "    #curr_otu_table = join(otu_table_dir,\"rarefaction_%i_0.biom\" %depth)\n",
    "    #print (\"curr_otu_table:\",curr_otu_table)\n",
    "    for compartment in compartments:\n",
    "        compartment_no_spaces = compartment.replace(\" \",\"_\")\n",
    "        otu_table_dir =  join(output_dir,\"rarified_to_%i_%s\" %(depth,compartment_no_spaces))\n",
    "        curr_outdir = join(output_dir,\"adiv_%s_%i\" %(compartment_no_spaces,depth))\n",
    "       \n",
    "        \n",
    "        print(\"Calculating alpha diversity on directory: %s\" %otu_table_dir)\n",
    "        adiv_cmd = \"alpha_diversity.py -i %s -o %s -t %s -m %s\" %(otu_table_dir,curr_outdir,tree,adiv_methods)\n",
    "        print(\"adiv cmd:%s\"%adiv_cmd)\n",
    "        !$adiv_cmd\n",
    "        print (\"Done. alpha diversity output dir: %s\" %curr_outdir)\n",
    "        collated_alpha_dir = join(output_dir,\"collated_alpha_%s_%i\" %(compartment_no_spaces,depth))\n",
    "        print (\"Collating alpha diversity to output dir: %s\" %collated_alpha_dir)\n",
    "        \n",
    "        !collate_alpha.py -i $curr_outdir -o $collated_alpha_dir\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing depth 1000\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Analyzing depth 5000\n",
      "Running alpha diversity comparisons comparisons...\n",
      "Error in compare_alpha_diversity.py: option -i: file does not exist: '../output/collated_alpha_Coral_Tissue_5000/equitability.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in compare_alpha_diversity.py: option -i: file does not exist: '../output/collated_alpha_Coral_Tissue_5000/PD_whole_tree.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in compare_alpha_diversity.py: option -i: file does not exist: '../output/collated_alpha_Coral_Tissue_5000/chao1.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Error in compare_alpha_diversity.py: option -i: file does not exist: '../output/collated_alpha_Coral_Tissue_5000/observed_otus.txt'\n",
      "\n",
      "If you need help with QIIME, see:\n",
      "http://help.qiime.org\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Analyzing depth 10000\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Analyzing depth 15000\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Analyzing depth 20000\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "Running alpha diversity comparisons comparisons...\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/tight_layout.py:225: UserWarning: tight_layout : falling back to Agg renderer\n",
      "  warnings.warn(\"tight_layout : falling back to Agg renderer\")\n"
     ]
    }
   ],
   "source": [
    "adiv_methods = \"equitability,PD_whole_tree,chao1,observed_otus\"\n",
    "comparison_categories = \"functional_group_sensu_darling\"\n",
    "compartments = ['Coral Tissue','Coral Mucus','Coral Skeleton']\n",
    "\n",
    "for depth in rarefaction_depths:\n",
    "    print (\"Analyzing depth %i\" %depth)\n",
    "    otu_table_dir = rarefied_data[depth]\n",
    "    curr_outdir = join(output_dir,\"rarified_to_%i_%s\" %(depth,compartment_no_spaces))\n",
    "    for compartment in compartments:\n",
    "        compartment_no_spaces = compartment.replace(\" \",\"_\")\n",
    "        otu_table_dir = curr_outdir\n",
    "        curr_outdir = join(output_dir,\"adiv_%s_%i\" %(compartment_no_spaces,depth))  \n",
    "        collated_alpha_dir = join(output_dir,\"collated_alpha_%s_%i\" %(compartment_no_spaces,depth))\n",
    "\n",
    "        print (\"Running alpha diversity comparisons comparisons...\")\n",
    "        \n",
    "        for m in adiv_methods.split(\",\"):\n",
    "            compare_alpha_results = join(output_dir,\"compare_alpha_%s_%i_%s\" %(compartment_no_spaces,depth,m))\n",
    "            collated_alpha_file = join(collated_alpha_dir,\"%s.txt\" %m)\n",
    "            !compare_alpha_diversity.py -i $collated_alpha_file -o $compare_alpha_results \\\n",
    "              -m $mapping -c $comparison_categories -t nonparametric"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Core microbiome analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing depth 1000\n",
      "curr_compartment:BiologicalMatter:Coral Tissue\n",
      "../output/rarified_to_1000/rarefaction_1000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_1000/rarefaction_1000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Tissue\" -o ../output/core_Coral_Tissue_1000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "curr_compartment:BiologicalMatter:Coral Mucus\n",
      "../output/rarified_to_1000/rarefaction_1000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_1000/rarefaction_1000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Mucus\" -o ../output/core_Coral_Mucus_1000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "curr_compartment:BiologicalMatter:Coral Skeleton\n",
      "../output/rarified_to_1000/rarefaction_1000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_1000/rarefaction_1000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Skeleton\" -o ../output/core_Coral_Skeleton_1000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "Analyzing depth 5000\n",
      "curr_compartment:BiologicalMatter:Coral Tissue\n",
      "../output/rarified_to_5000/rarefaction_5000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_5000/rarefaction_5000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Tissue\" -o ../output/core_Coral_Tissue_5000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "curr_compartment:BiologicalMatter:Coral Mucus\n",
      "../output/rarified_to_5000/rarefaction_5000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_5000/rarefaction_5000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Mucus\" -o ../output/core_Coral_Mucus_5000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "curr_compartment:BiologicalMatter:Coral Skeleton\n",
      "../output/rarified_to_5000/rarefaction_5000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_5000/rarefaction_5000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Skeleton\" -o ../output/core_Coral_Skeleton_5000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "Analyzing depth 10000\n",
      "curr_compartment:BiologicalMatter:Coral Tissue\n",
      "../output/rarified_to_10000/rarefaction_10000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_10000/rarefaction_10000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Tissue\" -o ../output/core_Coral_Tissue_10000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "curr_compartment:BiologicalMatter:Coral Mucus\n",
      "../output/rarified_to_10000/rarefaction_10000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_10000/rarefaction_10000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Mucus\" -o ../output/core_Coral_Mucus_10000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "curr_compartment:BiologicalMatter:Coral Skeleton\n",
      "../output/rarified_to_10000/rarefaction_10000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_10000/rarefaction_10000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Skeleton\" -o ../output/core_Coral_Skeleton_10000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "Analyzing depth 15000\n",
      "curr_compartment:BiologicalMatter:Coral Tissue\n",
      "../output/rarified_to_15000/rarefaction_15000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_15000/rarefaction_15000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Tissue\" -o ../output/core_Coral_Tissue_15000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "curr_compartment:BiologicalMatter:Coral Mucus\n",
      "../output/rarified_to_15000/rarefaction_15000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_15000/rarefaction_15000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Mucus\" -o ../output/core_Coral_Mucus_15000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "curr_compartment:BiologicalMatter:Coral Skeleton\n",
      "../output/rarified_to_15000/rarefaction_15000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_15000/rarefaction_15000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Skeleton\" -o ../output/core_Coral_Skeleton_15000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "Analyzing depth 20000\n",
      "curr_compartment:BiologicalMatter:Coral Tissue\n",
      "../output/rarified_to_20000/rarefaction_20000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_20000/rarefaction_20000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Tissue\" -o ../output/core_Coral_Tissue_20000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "curr_compartment:BiologicalMatter:Coral Mucus\n",
      "../output/rarified_to_20000/rarefaction_20000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_20000/rarefaction_20000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Mucus\" -o ../output/core_Coral_Mucus_20000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n",
      "curr_compartment:BiologicalMatter:Coral Skeleton\n",
      "../output/rarified_to_20000/rarefaction_20000_0.biom\n",
      "rarefaction_1000_0.biom rarefaction_1000_2.biom rarefaction_1000_4.biom rarefaction_1000_6.biom rarefaction_1000_8.biom\n",
      "rarefaction_1000_1.biom rarefaction_1000_3.biom rarefaction_1000_5.biom rarefaction_1000_7.biom rarefaction_1000_9.biom\n",
      "compute_core_microbiome.py -i ../output/rarified_to_20000/rarefaction_20000_0.biom --mapping_fp ../input/gcmp16S_map_r25.txt --valid_states \"BiologicalMatter:Coral Skeleton\" -o ../output/core_Coral_Skeleton_20000 --num_fraction_for_core_steps 21\n",
      "Traceback (most recent call last):\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 171, in <module>\n",
      "    main()\n",
      "  File \"/macqiime/anaconda/bin/compute_core_microbiome.py\", line 156, in main\n",
      "    write_biom_table(core_table, output_table_fp)\n",
      "  File \"/macqiime/anaconda/lib/python2.7/site-packages/qiime/util.py\", line 569, in write_biom_table\n",
      "    \"Attempting to write an empty BIOM table to disk. \"\n",
      "qiime.util.EmptyBIOMTableError: Attempting to write an empty BIOM table to disk. QIIME doesn't support writing empty BIOM output files.\n"
     ]
    }
   ],
   "source": [
    "## Pick the 0th rarefaction at each depth,\n",
    "# calculate core microbiomes for each tissue compartment\n",
    "\n",
    "for depth in rarefaction_depths:\n",
    "    print (\"Analyzing depth %i\" %depth)\n",
    "    otu_table_dir = rarefied_data[depth]\n",
    "    curr_otu_table = join(otu_table_dir,\"rarefaction_%i_0.biom\" %depth)\n",
    "    #print (\"curr_otu_table:\",curr_otu_table)\n",
    "    for compartment in ['Coral Tissue','Coral Mucus','Coral Skeleton']:\n",
    "        curr_outdir = join(output_dir,\"core_%s_%i\" %(compartment.replace(\" \",\"_\"),depth))\n",
    "        curr_compartment = \"BiologicalMatter:%s\" %compartment\n",
    "        print(\"curr_compartment:%s\" %curr_compartment)\n",
    "        !echo $curr_otu_table\n",
    "        !ls ../output/rarified_to_1000/\n",
    "        \n",
    "        \n",
    "        #NOTE this shouldn't be necessary\n",
    "        #but was getting an arcane error when substituting directly using !/$ magic\n",
    "        cmd_template = 'compute_core_microbiome.py -i %s --mapping_fp %s --valid_states \"%s\" -o %s --num_fraction_for_core_steps 11'\n",
    "        cmd_str = cmd_template %(curr_otu_table,mapping,curr_compartment,curr_outdir)\n",
    "        print(cmd_str)\n",
    "        !$cmd_str\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Beta-diversity analysis across rarefactions\n",
    "\n",
    "As a next step, we'd like to test the senstitivity of our conclusions about beta-diversity\n",
    "(mostly conducted at 1000 seqs/sample), at additional depths. \n",
    "\n",
    "Across all of our analyses, the factors we test are:\n",
    "Factors we analyze:\n",
    "prop_Colony_maximum_diameter_universal\n",
    "field_host_name\n",
    "latitude\n",
    "field_host_genus_id\n",
    "reef_name\n",
    "oz_disease_mean\n",
    "visibility\n",
    "turf_contact_percent\n",
    "host_clade_sensu_fukami\n",
    "binary_turf_contact\n",
    "photosynthetically_active_radiation\n",
    "cyanobacteria_percent\n",
    "functional_group_sensu_darling\n",
    "cyanobacteria_contact\n",
    "n_macroalgal_contacts\n",
    "binary_macroalgal_contact\n",
    "\n",
    "additional factors we should include for consistency or by reference to reviewer comments:\n",
    "temperature \n",
    "depth\n",
    "geographic_area\n",
    "maximum_corallite_width\n",
    "symbiodinium_sp_in_propagules\n",
    "growth_form_typical\n",
    "complex_robust\n",
    "\n",
    "\n",
    "###Step 1: infer beta-diversity distances at each depth. \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing depth 1000\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_1000/rarefaction_1000_0.biom -o ../output/rarified_to_1000/rarefaction_1000_0_Coral_Tissue.biom --valid_states 'BiologicalMatter:Coral Tissue' -m ../input/gcmp16S_map_r25.txt\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_1000/rarefaction_1000_0.biom -o ../output/rarified_to_1000/rarefaction_1000_0_Coral_Mucus.biom --valid_states 'BiologicalMatter:Coral Mucus' -m ../input/gcmp16S_map_r25.txt\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_1000/rarefaction_1000_0.biom -o ../output/rarified_to_1000/rarefaction_1000_0_Coral_Skeleton.biom --valid_states 'BiologicalMatter:Coral Skeleton' -m ../input/gcmp16S_map_r25.txt\n",
      "Analyzing depth 5000\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_5000/rarefaction_5000_0.biom -o ../output/rarified_to_5000/rarefaction_5000_0_Coral_Tissue.biom --valid_states 'BiologicalMatter:Coral Tissue' -m ../input/gcmp16S_map_r25.txt\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_5000/rarefaction_5000_0.biom -o ../output/rarified_to_5000/rarefaction_5000_0_Coral_Mucus.biom --valid_states 'BiologicalMatter:Coral Mucus' -m ../input/gcmp16S_map_r25.txt\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_5000/rarefaction_5000_0.biom -o ../output/rarified_to_5000/rarefaction_5000_0_Coral_Skeleton.biom --valid_states 'BiologicalMatter:Coral Skeleton' -m ../input/gcmp16S_map_r25.txt\n",
      "Analyzing depth 10000\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_10000/rarefaction_10000_0.biom -o ../output/rarified_to_10000/rarefaction_10000_0_Coral_Tissue.biom --valid_states 'BiologicalMatter:Coral Tissue' -m ../input/gcmp16S_map_r25.txt\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_10000/rarefaction_10000_0.biom -o ../output/rarified_to_10000/rarefaction_10000_0_Coral_Mucus.biom --valid_states 'BiologicalMatter:Coral Mucus' -m ../input/gcmp16S_map_r25.txt\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_10000/rarefaction_10000_0.biom -o ../output/rarified_to_10000/rarefaction_10000_0_Coral_Skeleton.biom --valid_states 'BiologicalMatter:Coral Skeleton' -m ../input/gcmp16S_map_r25.txt\n",
      "Analyzing depth 15000\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_15000/rarefaction_15000_0.biom -o ../output/rarified_to_15000/rarefaction_15000_0_Coral_Tissue.biom --valid_states 'BiologicalMatter:Coral Tissue' -m ../input/gcmp16S_map_r25.txt\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_15000/rarefaction_15000_0.biom -o ../output/rarified_to_15000/rarefaction_15000_0_Coral_Mucus.biom --valid_states 'BiologicalMatter:Coral Mucus' -m ../input/gcmp16S_map_r25.txt\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_15000/rarefaction_15000_0.biom -o ../output/rarified_to_15000/rarefaction_15000_0_Coral_Skeleton.biom --valid_states 'BiologicalMatter:Coral Skeleton' -m ../input/gcmp16S_map_r25.txt\n",
      "Analyzing depth 20000\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_20000/rarefaction_20000_0.biom -o ../output/rarified_to_20000/rarefaction_20000_0_Coral_Tissue.biom --valid_states 'BiologicalMatter:Coral Tissue' -m ../input/gcmp16S_map_r25.txt\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_20000/rarefaction_20000_0.biom -o ../output/rarified_to_20000/rarefaction_20000_0_Coral_Mucus.biom --valid_states 'BiologicalMatter:Coral Mucus' -m ../input/gcmp16S_map_r25.txt\n",
      "filter_samples_from_otu_table.py -i ../output/rarified_to_20000/rarefaction_20000_0.biom -o ../output/rarified_to_20000/rarefaction_20000_0_Coral_Skeleton.biom --valid_states 'BiologicalMatter:Coral Skeleton' -m ../input/gcmp16S_map_r25.txt\n"
     ]
    }
   ],
   "source": [
    "#Beta diversity across depths\n",
    "#First, generate compartment tables at each rarefaction depth\n",
    "for depth in rarefaction_depths:\n",
    "    print (\"Analyzing depth %i\" %depth)\n",
    "    otu_table_dir = rarefied_data[depth]\n",
    "    curr_otu_table = join(otu_table_dir,\"rarefaction_%i_0.biom\" %depth)\n",
    "    #print (\"curr_otu_table:\",curr_otu_table)\n",
    "    for compartment in ['Coral Tissue','Coral Mucus','Coral Skeleton']:\n",
    "        curr_outdir = join(output_dir,\"bdiv_%s_%i\" %(compartment.replace(\" \",\"_\"),depth))\n",
    "        curr_compartment = \"BiologicalMatter:%s\" %compartment\n",
    "        #print(\"curr_compartment:%s\" %curr_compartment)\n",
    "        compartment_table = join(otu_table_dir,\"rarefaction_%i_0_%s.biom\" %(depth,compartment.replace(\" \",\"_\"))) \n",
    "        cmd_str = \"filter_samples_from_otu_table.py -i %s -o %s --valid_states '%s' -m %s\"\n",
    "        cmd_str = cmd_str %(curr_otu_table,compartment_table,curr_compartment,mapping)\n",
    "        print(cmd_str)\n",
    "        !$cmd_str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing depth 1000\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_1000/rarefaction_1000_0_Coral_Tissue.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Tissue_1000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0906508978078 and the largest is 7.85722779812.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.468480011747 and the largest is 9.85686706579.\n",
      "  RuntimeWarning\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_1000/rarefaction_1000_0_Coral_Mucus.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Mucus_1000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.103362265619 and the largest is 9.53854433753.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.226329406238 and the largest is 9.76416875165.\n",
      "  RuntimeWarning\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_1000/rarefaction_1000_0_Coral_Skeleton.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Skeleton_1000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.108502288103 and the largest is 5.83355785548.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.186908535579 and the largest is 10.952176102.\n",
      "  RuntimeWarning\n",
      "Analyzing depth 5000\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_5000/rarefaction_5000_0_Coral_Tissue.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Tissue_5000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0417069795628 and the largest is 5.57210498002.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.346190448085 and the largest is 7.72196850813.\n",
      "  RuntimeWarning\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_5000/rarefaction_5000_0_Coral_Mucus.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Mucus_5000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0999010554251 and the largest is 9.50574306702.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.214869669958 and the largest is 9.35353069331.\n",
      "  RuntimeWarning\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_5000/rarefaction_5000_0_Coral_Skeleton.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Skeleton_5000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.102410683627 and the largest is 4.76831285506.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.154669722703 and the largest is 9.60593153469.\n",
      "  RuntimeWarning\n",
      "Analyzing depth 10000\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_10000/rarefaction_10000_0_Coral_Tissue.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Tissue_10000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0393014960353 and the largest is 4.70012392481.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.234150350819 and the largest is 6.11570299078.\n",
      "  RuntimeWarning\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_10000/rarefaction_10000_0_Coral_Mucus.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Mucus_10000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0868095056358 and the largest is 8.83304610374.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.198539020378 and the largest is 8.64307329326.\n",
      "  RuntimeWarning\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_10000/rarefaction_10000_0_Coral_Skeleton.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Skeleton_10000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0296882133245 and the largest is 3.72507384002.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0869277164592 and the largest is 7.43019242522.\n",
      "  RuntimeWarning\n",
      "Analyzing depth 15000\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_15000/rarefaction_15000_0_Coral_Tissue.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Tissue_15000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0282773704449 and the largest is 3.2185517484.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0636247484442 and the largest is 3.46185165536.\n",
      "  RuntimeWarning\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_15000/rarefaction_15000_0_Coral_Mucus.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Mucus_15000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0720130664232 and the largest is 7.64609790414.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.144873543761 and the largest is 7.29743499003.\n",
      "  RuntimeWarning\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_15000/rarefaction_15000_0_Coral_Skeleton.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Skeleton_15000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0514108940653 and the largest is 5.11407556054.\n",
      "  RuntimeWarning\n",
      "Analyzing depth 20000\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_20000/rarefaction_20000_0_Coral_Tissue.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Tissue_20000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.000390202272056 and the largest is 0.836076609084.\n",
      "  RuntimeWarning\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_20000/rarefaction_20000_0_Coral_Mucus.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Mucus_20000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.00140459510831 and the largest is 4.50794345619.\n",
      "  RuntimeWarning\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/skbio/stats/ordination/_principal_coordinate_analysis.py:107: RuntimeWarning: The result contains negative eigenvalues. Please compare their magnitude with the magnitude of some of the largest positive eigenvalues. If the negative ones are smaller, it's probably safe to ignore them, but if they are large in magnitude, the results won't be useful. See the Notes section for more details. The smallest eigenvalue is -0.0752142858861 and the largest is 4.31971412094.\n",
      "  RuntimeWarning\n",
      "beta_diversity_through_plots.py -i ../output/rarified_to_20000/rarefaction_20000_0_Coral_Skeleton.biom -t ../input/gg_constrained_rep_set_fastttree.tre --mapping_fp ../input/gcmp16S_map_r25.txt -o ../output/bdiv_Coral_Skeleton_20000 -p ../input/bdiv_bc_prefs.txt\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/numpy/core/fromnumeric.py:2652: VisibleDeprecationWarning: `rank` is deprecated; use the `ndim` attribute or function instead. To find the rank of a matrix see `numpy.linalg.matrix_rank`.\n",
      "  VisibleDeprecationWarning)\n"
     ]
    }
   ],
   "source": [
    "#Beta diversity across depths\n",
    "#Second, run beta_diversity_through_plots.py\n",
    "for depth in rarefaction_depths:\n",
    "    print (\"Analyzing depth %i\" %depth)\n",
    "    otu_table_dir = rarefied_data[depth]\n",
    "    curr_otu_table = join(otu_table_dir,\"rarefaction_%i_0.biom\" %depth)\n",
    "    #print (\"curr_otu_table:\",curr_otu_table)\n",
    "    for compartment in ['Coral Tissue','Coral Mucus','Coral Skeleton']:\n",
    "        curr_outdir = join(output_dir,\"bdiv_%s_%i\" %(compartment.replace(\" \",\"_\"),depth))\n",
    "        curr_compartment = \"BiologicalMatter:%s\" %compartment \n",
    "        compartment_table = join(otu_table_dir,\"rarefaction_%i_0_%s.biom\" %(depth,compartment.replace(\" \",\"_\")))\n",
    "        #NOTE this shouldn't be necessary\n",
    "        #but was getting an arcane error when substituting directly using !/$ magic\n",
    "        cmd_template = 'beta_diversity_through_plots.py -i %s -t %s --mapping_fp %s -o %s -p %s'\n",
    "        cmd_str = cmd_template %(compartment_table,tree,mapping,curr_outdir,bdiv_prefs_fp)\n",
    "        print(cmd_str)\n",
    "        !$cmd_str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing depth 1000\n",
      "../output/bdiv_Coral_Tissue_1000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_1000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_1000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/1000/Coral_Tissue/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Tissue_1000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_1000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_1000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/1000/Coral_Tissue/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Tissue_1000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_1000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_1000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/1000/Coral_Tissue/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_1000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_1000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_1000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/1000/Coral_Mucus/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_1000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_1000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_1000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/1000/Coral_Mucus/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_1000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_1000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_1000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/1000/Coral_Mucus/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_1000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_1000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_1000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/1000/Coral_Skeleton/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_1000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_1000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_1000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/1000/Coral_Skeleton/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_1000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_1000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_1000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/1000/Coral_Skeleton/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "Analyzing depth 5000\n",
      "../output/bdiv_Coral_Tissue_5000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_5000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_5000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/5000/Coral_Tissue/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Tissue_5000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_5000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_5000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/5000/Coral_Tissue/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Tissue_5000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_5000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_5000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/5000/Coral_Tissue/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_5000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_5000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_5000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/5000/Coral_Mucus/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_5000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_5000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_5000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/5000/Coral_Mucus/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_5000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_5000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_5000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/5000/Coral_Mucus/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_5000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_5000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_5000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/5000/Coral_Skeleton/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_5000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_5000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_5000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/5000/Coral_Skeleton/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_5000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_5000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_5000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/5000/Coral_Skeleton/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "Analyzing depth 10000\n",
      "../output/bdiv_Coral_Tissue_10000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_10000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_10000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/10000/Coral_Tissue/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Tissue_10000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_10000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_10000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/10000/Coral_Tissue/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Tissue_10000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_10000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_10000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/10000/Coral_Tissue/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_10000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_10000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_10000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/10000/Coral_Mucus/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_10000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_10000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_10000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/10000/Coral_Mucus/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_10000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_10000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_10000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/10000/Coral_Mucus/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_10000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_10000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_10000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/10000/Coral_Skeleton/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_10000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_10000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_10000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/10000/Coral_Skeleton/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_10000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_10000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_10000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/10000/Coral_Skeleton/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "Analyzing depth 15000\n",
      "../output/bdiv_Coral_Tissue_15000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_15000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_15000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/15000/Coral_Tissue/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Tissue_15000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_15000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_15000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/15000/Coral_Tissue/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Tissue_15000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_15000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_15000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/15000/Coral_Tissue/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_15000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_15000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_15000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/15000/Coral_Mucus/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_15000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_15000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_15000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/15000/Coral_Mucus/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_15000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_15000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_15000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/15000/Coral_Mucus/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_15000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_15000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_15000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/15000/Coral_Skeleton/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_15000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_15000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_15000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/15000/Coral_Skeleton/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_15000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_15000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_15000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/15000/Coral_Skeleton/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "Analyzing depth 20000\n",
      "../output/bdiv_Coral_Tissue_20000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_20000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_20000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/20000/Coral_Tissue/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Tissue_20000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_20000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_20000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/20000/Coral_Tissue/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Tissue_20000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Tissue_20000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Tissue_20000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/20000/Coral_Tissue/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_20000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_20000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_20000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/20000/Coral_Mucus/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_20000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_20000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_20000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/20000/Coral_Mucus/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Mucus_20000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Mucus_20000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Mucus_20000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/20000/Coral_Mucus/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_20000/weighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_20000/weighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_20000/weighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/20000/Coral_Skeleton/compare_categories_depth_weighted_unifrac -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_20000/bray_curtis_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_20000/bray_curtis_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_20000/bray_curtis_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/20000/Coral_Skeleton/compare_categories_depth_bray_curtis -c 'depth' -n 9999\n",
      "../output/bdiv_Coral_Skeleton_20000/unweighted_unifrac_dm.txt\n",
      "compare_categories.py -i ../output/bdiv_Coral_Skeleton_20000/unweighted_unifrac_dm_filtered_for_depth.txt  -m ../output/bdiv_Coral_Skeleton_20000/unweighted_unifrac_mapping_filtered_for_depth.txt --method adonis -o ../output/compare_categories_r2/20000/Coral_Skeleton/compare_categories_depth_unweighted_unifrac -c 'depth' -n 9999\n"
     ]
    }
   ],
   "source": [
    "categories = ['Range_size','Colony_maximum_diameter','prop_Colony_maximum_GCMP_recorded',\\\n",
    "            'max_dimension','enclosed_area','IUCN_Red_List_categoy','16S_tree_name','outgroup',\\\n",
    "  'Oocyte_size_at_maturity','temperature','depth','geographic_area',\\\n",
    "  'Corallite_width_maximum','Corallite_width_minimum','Symbiodinium_sp_in_propagules',\\\n",
    " 'Growth_form_typical','Sexual_system','Mode_of_larval_development','complex_robust',\\\n",
    "'prop_Colony_maximum_diameter_universal','host_name','latitude','longitude','host_genus',\\\n",
    " 'reef_name','oz_disease_mean','visibility','Skeletal_density','turf_contact_percent',\\\n",
    "  'host_clade_sensu_fukami','binary_turf_contact',\\\n",
    "  'photosynthetically_active_radiation','cyanobacteria_percent',\\\n",
    "    'functional_group_sensu_darling','cyanobacteria_contact',\\\n",
    "  'binary_CCA_contact','cca_contact_percent','n_macroalgal_contacts',\\\n",
    "    'binary_macroalgal_contact']\n",
    "#categories = ['depth']\n",
    "\n",
    "#Beta diversity across depths\n",
    "#Second, run beta_diversity_through_plots.py\n",
    "#this is going to be annoying to report at a gazillion depths. Maybe just the most reasonable\n",
    "#ones (probably 1k for comparison to Zaneveld et al., 2016 and 10k for comparison to Apprill et al)\n",
    "#similarly for now just doing weighted \n",
    "\n",
    "#for similar reasons I am setting this up to be general, \n",
    "#but just calculating weighted UniFrac for now\n",
    "metrics = ['weighted_unifrac','bray_curtis','unweighted_unifrac']\n",
    "#metrics = ['weighted_unifrac']\n",
    "rarefaction_depths = [1000,5000,10000,15000,20000]\n",
    "#rarefaction_depths = [1000]\n",
    "for depth in rarefaction_depths:\n",
    "    print (\"Analyzing depth %i\" %depth)\n",
    "    otu_table_dir = rarefied_data[depth]\n",
    "    curr_otu_table = join(otu_table_dir,\"rarefaction_%i_0.biom\" %depth)\n",
    "    #print (\"curr_otu_table:\",curr_otu_table)\n",
    "    for compartment in ['Coral Tissue','Coral Mucus','Coral Skeleton']:\n",
    "        for metric in metrics:\n",
    "            for category in categories:\n",
    "                curr_dm = join(output_dir,\"bdiv_%s_%i\"%(compartment.replace(\" \",\"_\"),depth),\"%s_dm.txt\"%(metric))\n",
    "                print(curr_dm)\n",
    "                curr_outdir = join(output_dir,\"compare_categories_r2\",\"%i\"%depth,\\\n",
    "                  \"%s\"%(compartment.replace(\" \",\"_\")),\"compare_categories_%s_%s\" %(category,metric))\n",
    "                curr_compartment = \"BiologicalMatter:%s\" %compartment \n",
    "                relevant_dm = join(output_dir,\"bdiv_%s_%i\"%(compartment.replace(\" \",\"_\"),depth),\"%s_dm_filtered_for_%s.txt\"%(metric,category))\n",
    "                negative_filter = \"'%s:Unknown'\" %category\n",
    "                !filter_distance_matrix.py -i $curr_dm -o $relevant_dm -m $mapping -s $negative_filter --negate\n",
    "                \n",
    "                filtered_otu_table = join(output_dir,\"bdiv_%s_%i\"%(compartment.replace(\" \",\"_\"),depth),\"%s_otu_table_filtered_for_%s.biom\"%(metric,category))\n",
    "                filtered_mapping = join(output_dir,\"bdiv_%s_%i\"%(compartment.replace(\" \",\"_\"),depth),\"%s_mapping_filtered_for_%s.txt\"%(metric,category))\n",
    "                \n",
    "                #NOTE: I only really want to filter samples with 'Unknown' values from the mapping file\n",
    "                #but there is no separate script for this. So I have to filter the OTU table too.\n",
    "                \n",
    "                positive_filter = \"'%s:*,!Unknown'\" %category\n",
    "                !filter_samples_from_otu_table.py -i $curr_otu_table -o $filtered_otu_table --output_mapping_fp $filtered_mapping -m $mapping -s $positive_filter\n",
    "                #NOTE this shouldn't be necessary\n",
    "                #but was getting an arcane error when substituting directly using !/$ magic\n",
    "                cmd_template = \"compare_categories.py -i %s  -m %s --method adonis -o %s -c '%s' -n 9999\"\n",
    "                cmd_str = cmd_template %(relevant_dm,filtered_mapping,curr_outdir,category)\n",
    "                print(cmd_str)\n",
    "                !$cmd_str"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Compile Adonis results\n",
    "\n",
    "To generate heatmaps, we now need to compile all these Adonis results into a single CSV file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analyzing depth 1000\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "Analyzing depth 5000\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "Analyzing depth 10000\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "Analyzing depth 15000\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "Analyzing depth 20000\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n",
      "/macqiime/anaconda/lib/python2.7/site-packages/matplotlib/collections.py:590: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  if self._edgecolors == str('face'):\n"
     ]
    }
   ],
   "source": [
    "rarefaction_depths = [1000,5000,10000,15000,20000]\n",
    "\n",
    "for depth in rarefaction_depths:\n",
    "    print (\"Analyzing depth %i\" %depth)   \n",
    "    for compartment in ['Coral Tissue','Coral Mucus','Coral Skeleton']:\n",
    "            for category in categories:\n",
    "                curr_outdir = join(output_dir,\"compare_categories_r2\",\"%i\"%depth)\n",
    "                curr_output = join(output_dir,\"compare_categories_r2\",\"adonis_results_%i.tsv\" %depth)\n",
    "                !python csv_from_adonis_results_deep_rarefaction.py $curr_outdir $curr_output\n",
    "                !python make_heatmap.py $curr_output\n",
    "                 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Summarize taxa for ASR\n",
    "\n",
    "Summarize taxonomy at 1000 seqs/sample and add to the mapping file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Resulting heatmaps:"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
